#################
## Get files
#################
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/AA_GWAS_AX_HbA1cadjFGlu_BioMe_SNPTEST2_20032014_CS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/AA_GWAS_AX_HbA1cadjFGlu_HANDLS_MACH_24122014_SMT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/AA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_AGES_PROBABEL_23092014_AVS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_BioMe_SNPTEST2_20032014_CS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_CRC_MACH_09102014_TH.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_DES_SNPTEST_22042014_CL.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_DGI_SNPTEST_23122014_CL.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_Fenland_SNPTEST_14052014_SW.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_Fenland-OMICS_SNPTEST_28022015_PZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_FHS_MACH_01092015_AP.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_GOYA-Male-CASES_QUICKTEST_13112014_TSA.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_GOYA-Male-CONTROLS_QUICKTEST_13112014_TSA.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_KORA_SNPTEST_15012015_CM.SW.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_LifeLines_PLINK_27102014_PJM.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_LOLIPOP-EW610_SNPTEST_27052014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_LOLIPOP-EWA_SNPTEST_08052014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_LOLIPOP-EWP_SNPTEST_08052014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_LURIC_SNPTEST_27112014_MK.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_METSIM_EPACTSEMMAX_10042014_XS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_NTR_PLINK_08042014_JJH.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_XF_HbA1cadjFGlu_NTR_PLINK_08052014_JJH.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_XM_HbA1cadjFGlu_NTR_PLINK_08052014_JJH.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_ORCADES_PROBABEL_29052014_KES.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_A_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_X_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_TRAILSpop_SNPTEST_12062014_PJ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EA_GWAS_AX_HbA1cadjFGlu_TWINGENE_EMMAX_02062014_SG.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_AX_HbA1cadjFGlu_AASC_MINIMACH_25102014_YT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_A_HbA1cadjFGlu_CAGE-KING-Omni_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_XF_HbA1cadjFGlu_CAGE-KING-Omni_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_XM_HbA1cadjFGlu_CAGE-KING-Omni_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_A_HbA1cadjFGlu_CAGE-KING-Quad_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_XF_HbA1cadjFGlu_CAGE-KING-Quad_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_XM_HbA1cadjFGlu_CAGE-KING-Quad_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_A_HbA1cadjFGlu_CAGEGWAS1_PLINK_14102014_FT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_AX_HbA1cadjFGlu_KARE_MACH2QTL_31102014_YJK.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_AX_HbA1cadjFGlu_NAGAHAMA_MINIMACH_12262014_YT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_AX_HbA1cadjFGlu_NHAPC_SNPTEST_19092014_YQ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_A_HbA1cadjFGlu_SP2-1M_SNPTEST_15092014_EB.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_A_HbA1cadjFGlu_SP2-610_SNPTEST_13092014_EB.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_A_HbA1cadjFGlu_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/HA_GWAS_AX_HbA1cadjFGlu_BioMe_SNPTEST2_20032014_CS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/HA_GWAS_A_HbA1cadjFGlu_IRAS_SNPTEST_02032015_JTZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/HA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/HA_GWAS_AX_HbA1cadjFGlu_SOL_MIXEDMODEL_20022015_TL.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/IAA_GWAS_A_HbA1cadjFGlu_LOLIPOP-IA317_SNPTEST_16062014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/IAA_GWAS_A_HbA1cadjFGlu_LOLIPOP-IA610_SNPTEST_08052014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/IAA_GWAS_A_HbA1cadjFGlu_LOLIPOP-IAP_SNPTEST_16062014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/IAA_GWAS_A_HbA1cadjFGlu_LOLIPOP-OmniEE_SNPTEST_16062014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/IAA_GWAS_A_HbA1cadjFGlu_RHS_SNPTEST_04012015_FT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/Raw/IAA_GWAS_A_HbA1cadjFGlu_RHS-610K_SNPTEST_04012015_FT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EA_MCHIP_A_HbA1cadjFGlu_D2D2007.DPS_EPACTSEMMAX_10042014_XS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EA_MCHIP_A_HbA1cadjFGlu_DIAGEN_EPACTSEMMAX_10042014_XS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EA_MCHIP_A_HbA1cadjFGlu_Ely_SNPTEST_15122014_JAL.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EA_MCHIP_A_HbA1cadjFGlu_GoDARTS_IMPUTE_06102014_APM.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EA_MCHIP_A_HbA1cadjFGlu_Health2006_SNPTEST_16122014_NG.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EA_MCHIP_A_HbA1cadjFGlu_Inter99_SNPTEST_16122014_NG.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EA_MCHIP_A_HbA1cadjFGlu_LEIPZIG-adults_SNPTEST_31112014_AMah.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EA_MCHIP_A_HbA1cadjFGlu_LEIPZIG-kids_SNPTEST_31112014_AMah.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/Raw/EAA_MCHIP_A_HbA1cadjFGlu_TAICHI_SNPTEST_11022015_YFC.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/AA_GWAS_AX_HbA1cadjFGlu_InvNorm_BioMe_SNPTEST2_20032014_CS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/AA_GWAS_AX_HbA1cadjFGlu_InvNorm_HANDLS_MACH_24122014_SMT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/AA_GWAS_AX_HbA1cadjFGlu_InvNorm_MESA_SNPTEST_29122014_JY.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_AGES_PROBABEL_23092014_AVS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_BioMe_SNPTEST2_20032014_CS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_CRC_MACH_09102014_TH.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_DES_SNPTEST_22042014_CL.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_DGI_SNPTEST_23122014_CL.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_Fenland_SNPTEST_14052014_SW.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_Fenland-OMICS_SNPTEST_28022015_PZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_FHS_MACH_01092015_AP.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_GOYA-Male-CASES_QUICKTEST_13112014_TSA.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_GOYA-Male-CONTROLS_QUICKTEST_13112014_TSA.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_KORA_SNPTEST_15012015_CM.SW.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LifeLines_PLINK_27102014_PJM.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-EW610_SNPTEST_27052014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-EWA_SNPTEST_08052014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-EWP_SNPTEST_08052014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LURIC_SNPTEST_07112014_MK.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_MESA_SNPTEST_29122014_JY.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_METSIM_EPACTSEMMAX_10042014_XS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_NTR_PLINK_08042014_JJH.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_XF_HbA1cadjFGlu_InvNorm_NTR_PLINK_08052014_JJH.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_XM_HbA1cadjFGlu_InvNorm_NTR_PLINK_08052014_JJH.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_ORCADES_PROBABEL_29052014_KES.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_A_HbA1cadjFGlu_InvNorm_SardiNIA_EPACTS_22092014_GP.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_X_HbA1cadjFGlu_InvNorm_SardiNIA_EPACTS_22092014_GP.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_TRAILSpop_SNPTEST_12062014_PJ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_TWINGENE_EMMAX_02062014_SG.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_AASC_MINIMACH_25102014_YT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_CAGE-KING-Omni_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_XF_HbA1cadjFGlu_InvNorm_CAGE-KING-Omni_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_XM_HbA1cadjFGlu_InvNorm_CAGE-KING-Omni_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_CAGE-KING-Quad_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_XF_HbA1cadjFGlu_InvNorm_CAGE-KING-Quad_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_XM_HbA1cadjFGlu_InvNorm_CAGE-KING-Quad_MACH_03102014_MN.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_CAGEGWAS1_PLINK_14102014_FT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_KARE_MACH2QTL_31102014_YJK.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_MESA_SNPTEST_29122014_JY.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_NAGAHAMA_MINIMACH_12262014_YT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_NHAPC_SNPTEST_19092014_YQ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_SP2-1M_SNPTEST_11092014_EB.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_SP2-610_SNPTEST_06092014_EB.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/HA_GWAS_AX_HbA1cadjFGlu_InvNorm_BioMe_SNPTEST2_20032014_CS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/HA_GWAS_A_HbA1cadjFGlu_InvNorm_IRAS_SNPTEST_02032015_JTZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/HA_GWAS_AX_HbA1cadjFGlu_InvNorm_MESA_SNPTEST_29122014_JY.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/HA_GWAS_AX_HbA1cadjFGlu_InvNorm_SOL_MIXEDMODEL_20022015_TL.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-IA317_SNPTEST_16062014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-IA610_SNPTEST_08052014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-IAP_SNPTEST_16062014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-OmniEE_SNPTEST_16062014_WZ.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_RHS_SNPTEST_04012015_FT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/GWAS-1000G/HbA1cadjFGlu/InvNorm/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_RHS-610K_SNPTEST_04012015_FT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_D2D2007.DPS_EPACTSEMMAX_10042014_XS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_DIAGEN_EPACTSEMMAX_10042014_XS.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_Ely_SNPTEST_15122014_JAL.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_GoDARTS_IMPUTE_06102014_APM.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_Health2006_SNPTEST_16122014_NG.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_Inter99_SNPTEST_16122014_NG.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_LEIPZIG-adults_SNPTEST_31112014_AMah.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_LEIPZIG-kids_SNPTEST_31112014_AMah.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/
#get /cleaned_gluinsrelatedtraits/Harmonized/Mchip-1000G/HbA1cadjFGlu/InvNorm/EAA_MCHIP_A_HbA1cadjFGlu_InvNorm_TAICHI_SNPTEST_11022015_YFC.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/


## Deal with the unknown values from PROCARDIS before EasyQC
zcat QC2_download/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz | head -n10000000 | grep unknown | head

QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz
QC2_download/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz

mv QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz QC2_originalPROCARDIS/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz
mv QC2_download/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz         QC2_originalPROCARDIS/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz

zcat QC2_originalPROCARDIS/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz | sed "s/unknown/./g" | bgzip -c > QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz
zcat QC2_originalPROCARDIS/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz         | sed "s/unknown/./g" | bgzip -c > QC2_download/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz

zcat QC2_download/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz | head -n10000000 | grep unknown | head
zcat QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz | head -n10000000 | grep unknown | head



## The SP2-1M files have NA values!!
mv QC2_download/*SP2-1* QC2_originalPROCARDIS/

zcat QC2_originalPROCARDIS/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_SP2-1M_SNPTEST_11092014_EB.txt.gz | sed "s/\tNA\t/\t.\t/g" | bgzip -c > QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_SP2-1M_SNPTEST_11092014_EB.txt.gz
zcat QC2_originalPROCARDIS/EAA_GWAS_A_HbA1cadjFGlu_SP2-1M_SNPTEST_15092014_EB.txt.gz         | sed "s/\tNA\t/\t.\t/g" | bgzip -c > QC2_download/EAA_GWAS_A_HbA1cadjFGlu_SP2-1M_SNPTEST_15092014_EB.txt.gz

## The Vnaderbilt files have INFO as ".-000" or ".-001"
mv QC2_download/*Vanderbilt* QC2_originalPROCARDIS/

zcat QC2_originalPROCARDIS/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz | awk '{ if($19==".-000" || $19==".-001") { print $1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7"\t"$8"\t"$9"\t"$10"\t"$11"\t"$12"\t"$13"\t"$14"\t"$15"\t"$16"\t"$17"\t"$18"\t0"} else {print $0} }' | bgzip -c > QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz
zcat QC2_originalPROCARDIS/EAA_GWAS_A_HbA1cadjFGlu_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz         | awk '{ if($19==".-000" || $19==".-001") { print $1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7"\t"$8"\t"$9"\t"$10"\t"$11"\t"$12"\t"$13"\t"$14"\t"$15"\t"$16"\t"$17"\t"$18"\t0"} else {print $0} }' | bgzip -c > QC2_download/EAA_GWAS_A_HbA1cadjFGlu_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz


zcat QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz | cut -f19 | grep -F ".-" | wc
zcat QC2_download/EAA_GWAS_A_HbA1cadjFGlu_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz | cut -f19 | grep -F ".-" | wc

#################
## Scripts and try EasyQC
#################


#CALCULATE --rcdCalc sum(CALL_RATE_QTL<0.95,na.rm=T) --strCalcName Harmo.N_lowCALLRAT
#STRSPLITCOL --colSplit MarkerName --strSplit : --numSplitIdx 1 --colOut CHR2

####### /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/QC2_EasyQC_test.ecf
####### /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/QC2_EasyQC_test.R
### Set parameters and input files
DEFINE --strMissing . --strSeparator TAB --pathOut /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned --acolIn MarkerName;SNP;CHR;POS;EFFECT_ALLELE;NON_EFFECT_ALLELE;N_QTL;N0_QTL;N1_QTL;N2_QTL;EAF_QTL;HWE_P_QTL;CALL_RATE_QTL;BETA;SE;PVAL;IMPUTED;INFO_TYPE;INFO --acolInClasses character;character;character;integer;character;character;numeric;numeric;numeric;numeric;numeric;numeric;numeric;numeric;numeric;numeric;numeric;numeric;numeric
### Start EasyQC
START EASYQC
# add/edit some columns
ADDCOL  --rcdAddCol  pmin(EAF_QTL*N_QTL*2,(1-EAF_QTL)*N_QTL*2,na.rm=TRUE)                      --colOut MAC
ADDCOL  --rcdAddCol  sapply( MarkerName , FUN=function(x){ return(strsplit(x,":")[[1]][1]) } ) --colOut CHR2
EDITCOL --rcdEditCol ifelse(SE==(-1),NA,SE)     --colEdit SE
EDITCOL --rcdEditCol ifelse(PVAL==(-1),NA,PVAL) --colEdit PVAL
# general counts
GETNUM --rcdGetNum (grepl("SNP",MarkerName))   --strGetNumName Harmo.N_SNPs
GETNUM --rcdGetNum (grepl("INDEL",MarkerName) | grepl("MERGED_DEL",MarkerName)) --strGetNumName Harmo.N_indels
GETNUM --rcdGetNum (CHR2 %in% 1:22)            --strGetNumName Harmo.N_A
GETNUM --rcdGetNum (CHR2=="X")                 --strGetNumName Harmo.N_X
GETNUM --rcdGetNum (CHR2=="XY")                --strGetNumName Harmo.N_XY
GETNUM --rcdGetNum (IMPUTED==1)                --strGetNumName Harmo.N_imputed
GETNUM --rcdGetNum (IMPUTED==0)                --strGetNumName Harmo.N_genotyped
GETNUM --rcdGetNum (is.na(IMPUTED))            --strGetNumName Harmo.N_unknownImpute
# counts of cleaning criteria
GETNUM --rcdGetNum (MAC<3)                           --strGetNumName Harmo.N_lowMAC
GETNUM --rcdGetNum (SE>=10)                          --strGetNumName Harmo.N_highSE
GETNUM --rcdGetNum (INFO<0.4)                        --strGetNumName Harmo.N_lowINFO
GETNUM --rcdGetNum (is.na(BETA) | is.na(SE))         --strGetNumName Harmo.N_missingBETAorSE
GETNUM --rcdGetNum (HWE_P_QTL<0.000001 & IMPUTED==0) --strGetNumName Harmo.N_genotypedLowHWE
GETNUM --rcdGetNum (CALL_RATE_QTL<0.95 & IMPUTED==0) --strGetNumName Harmo.N_genotypedLowCALLRATE
GETNUM --rcdGetNum (IMPUTED==1 & is.na(INFO))        --strGetNumName Harmo.N_imputedWithNoInfo
GETNUM --rcdGetNum (is.na(IMPUTED) & is.na(INFO))    --strGetNumName Harmo.N_unknownImputeWithNoInfo
# counts for further checks
GETNUM --rcdGetNum (EAF_QTL==0 | EAF_QTL==1)           --strGetNumName Harmo.N_mono
GETNUM --rcdGetNum (IMPUTED==0 & is.na(HWE_P_QTL))     --strGetNumName Harmo.N_genotypedWithNoHWE
GETNUM --rcdGetNum (IMPUTED==0 & is.na(CALL_RATE_QTL)) --strGetNumName Harmo.N_genotypedWithNoCALLRATE
GETNUM --rcdGetNum (CALL_RATE_QTL<0.95 & IMPUTED==1)   --strGetNumName Harmo.N_imputedLowCALLRATE
GETNUM --rcdGetNum (is.na(N_QTL))                      --strGetNumName Harmo.N_QTL_NA
# counts for further checks of mine not in the guidelines
GETNUM --rcdGetNum (is.na(EAF_QTL))      --strGetNumName Harmo.EAF_QTL_NA_getnum
GETNUM --rcdGetNum (CALL_RATE_QTL<0.95)  --strGetNumName Harmo.N_lowCALLRATE
GETNUM --rcdGetNum (HWE_P_QTL<0.000001)  --strGetNumName Harmo.N_lowHWE
# stats
EVALSTAT --colStat HWE_P_QTL     --strTag Harmo
EVALSTAT --colStat CALL_RATE_QTL --strTag Harmo
EVALSTAT --colStat MAC           --strTag Harmo
EVALSTAT --colStat INFO          --strTag Harmo
EVALSTAT --colStat BETA          --strTag Harmo
EVALSTAT --colStat SE            --strTag Harmo
EVALSTAT --colStat PVAL          --strTag Harmo
EVALSTAT --colStat EAF_QTL       --strTag Harmo
# cleaning
CLEAN --rcdClean (MAC<3)                           --strCleanName Harmo.N_lowMAC_SeqClean
CLEAN --rcdClean (SE>=10)                          --strCleanName Harmo.N_highSE_SeqClean
CLEAN --rcdClean (INFO<0.4)                        --strCleanName Harmo.N_lowINFO_SeqClean
CLEAN --rcdClean (is.na(BETA) | is.na(SE))         --strCleanName Harmo.N_missingBETAorSE_SeqClean
CLEAN --rcdClean (IMPUTED==0 & CALL_RATE_QTL<0.95) --strCleanName Harmo.N_genotypedLowCALLRATE_SeqClean
CLEAN --rcdClean (IMPUTED==0 & HWE_P_QTL<0.000001) --strCleanName Harmo.N_genotypedLowHWE_SeqClean
CLEAN --rcdClean (IMPUTED==1 & is.na(INFO))        --strCleanName Harmo.N_imputedWithNoInfo_SeqClean
CLEAN --rcdClean (is.na(IMPUTED) & is.na(INFO))    --strCleanName Harmo.N_unknownImputeWithNoInfo_SeqClean
#writing
GETCOLS --acolOut MarkerName;EFFECT_ALLELE;NON_EFFECT_ALLELE;N_QTL;EAF_QTL;BETA;SE;PVAL;IMPUTED;INFO
WRITE --strMode gz --strSep TAB --strMissing .
### Stop EasyQC
STOP EASYQC
####################################################################################

## To check AASC files
ADDCOL  --rcdAddCol  ifelse(HWE_P_QTL<0.000001,yes=NA,no=HWE_P_QTL)      --colOut HWE_P_QTL_high
ADDCOL  --rcdAddCol  ifelse(HWE_P_QTL<=0.000001,yes=HWE_P_QTL,no=NA)     --colOut HWE_P_QTL_low

GETNUM --rcdGetNum (IMPUTED==1)                --strGetNumName Harmo.N_imputed
GETNUM --rcdGetNum (IMPUTED==0)                --strGetNumName Harmo.N_genotyped
GETNUM --rcdGetNum (is.na(IMPUTED))            --strGetNumName Harmo.N_unknownImpute

EVALSTAT --colStat HWE_P_QTL       --strTag Harmo
EVALSTAT --colStat HWE_P_QTL_high  --strTag Harmo
EVALSTAT --colStat HWE_P_QTL_low   --strTag Harmo

GETNUM --rcdGetNum (HWE_P_QTL<0.000001)  --strGetNumName Harmo.N_lowHWE
GETNUM --rcdGetNum (HWE_P_QTL>0.000001)  --strGetNumName Harmo.N_highHWE
GETNUM --rcdGetNum (is.na(HWE_P_QTL))    --strGetNumName Harmo.N_NoHWE

GETNUM --rcdGetNum (HWE_P_QTL<0.000001 & IMPUTED==0) --strGetNumName Harmo.N_genotypedLowHWE
GETNUM --rcdGetNum (HWE_P_QTL>0.000001 & IMPUTED==0) --strGetNumName Harmo.N_genotypedHighHWE
GETNUM --rcdGetNum (is.na(HWE_P_QTL)   & IMPUTED==0) --strGetNumName Harmo.N_genotypedNoHWE

GETNUM --rcdGetNum (HWE_P_QTL<0.000001 & IMPUTED==1) --strGetNumName Harmo.N_imputedLowHWE
GETNUM --rcdGetNum (HWE_P_QTL>0.000001 & IMPUTED==1) --strGetNumName Harmo.N_imputedHighHWE
GETNUM --rcdGetNum (is.na(HWE_P_QTL)   & IMPUTED==1) --strGetNumName Harmo.N_imputedNoHWE

CLEAN --rcdClean (HWE_P_QTL<0.000001) --strCleanName Harmo.N_genotypedLowHWE_SeqClean





library(EasyQC)
EasyQC("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/QC2_EasyQC_test.ecf")
# bsub -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_EasyQC_test.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_EasyQC_test.err -R"select[mem>12000] rusage[mem=12000]" -M12000 "/software/R-3.0.0/bin/R --no-save --args QC2_EasyQC_test.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"


rr = read.table("QC2_cleaned/QC2_EasyQC_test.rep",h=T,as.is=T)

ff=read.table(gzfile("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_X_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.txt.gz"),h=T,as.is=T,na.strings=".")
ff$MAC = pmin(ff$EAF_QTL*ff$N_QTL*2,(1-ff$EAF_QTL)*ff$N_QTL*2,na.rm=TRUE)
ff$CHR2 = sapply( ff$MarkerName , FUN=function(x){ return(strsplit(x,":")[[1]][1]) } )
se  =read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/EA_GWAS_X_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.Harmo.N_highSE.txt",h=T,as.is=T,na.strings=".")
mac =read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/EA_GWAS_X_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.Harmo.N_lowMAC.txt",h=T,as.is=T,na.strings=".")
info=read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/EA_GWAS_X_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.Harmo.N_lowINFO.txt",h=T,as.is=T,na.strings=".")

dim(ff)
dim(se)
dim(mac)
dim(info)

ff$ExcSE=ifelse( ff$MarkerName %in% se$MarkerName , yes=1 , no=0 )
ff$ExcMAC=ifelse( ff$MarkerName %in% mac$MarkerName , yes=1 , no=0 )
ff$ExcINFO=ifelse( ff$MarkerName %in% info$MarkerName , yes=1 , no=0 )

table(ff$ExcSE)
table(ff$ExcMAC)
table(ff$ExcINFO)

max(mac$MAC)
min(se$SE)
max(info$INFO)

table( ff$ExcMAC , ff$MAC<3 )
table( ff$ExcSE , ff$SE>=10 , ff$ExcMAC )
table( ff$ExcINFO , ff$INFO<0.4 , ff$ExcMAC==0 & ff$ExcSE==0 )

table( ff$ExcSE==1 | ff$ExcMAC==1 | ff$ExcINFO==1 )

max(mac$MAC)
min(se$SE)
max(info$INFO)

vv = ff$HWE_P_QTL
sum(is.na(vv))
min(vv,na.rm=T)
max(vv,na.rm=T)
mean(vv,na.rm=T)
median(vv,na.rm=T)
sd(vv,na.rm=T)
quantile(vv,p=0.25,na.rm=T)
quantile(vv,p=0.75,na.rm=T)
rm(vv)


# check Katie's code

tt=read.table(gzfile("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_XF_HbA1cadjFGlu_NTR_PLINK_08052014_JJH.txt.gz"),as.is=T,h=T,na.strings=".")
tt$MAC=pmin(tt$EAF_QTL*tt$N_QTL*2,(1-tt$EAF_QTL)*tt$N_QTL*2,na.rm=TRUE)

table(tt$MAC>=3 & tt$SE<10 & !is.na(tt$BETA) & !is.na(tt$SE) & tt$INFO>=0.4 & (tt$IMPUTED==1 | is.na(tt$IMPUTED) | tt$HWE_P_QTL>=0.000001 & tt$IMPUTED==0) & (tt$IMPUTED==0 | (tt$IMPUTED==1 | is.na(tt$IMPUTED)) & !is.na(tt$INFO))  , grepl("SNP",tt$MarkerName) )



#cleaning:
clean1=subset(tt, as.numeric(tt$MAC)>=3, na.rm=FALSE)
clean2=subset(clean1, as.numeric(clean1$SE)<10)
clean3=subset(clean2, as.numeric(clean2$INFO)>=0.4, na.rm=FALSE)
clean4=subset(clean3, is.na(clean3$BETA)==FALSE & is.na(clean3$SE)==FALSE)

table( grepl("SNP",clean1$MarkerName) )
table( tt$MAC>=3 , grepl("SNP",tt$MarkerName) )

table( grepl("SNP",clean2$MarkerName) )
table( tt$MAC>=3 & tt$SE<10 , grepl("SNP",tt$MarkerName) )

table( grepl("SNP",clean3$MarkerName) )
table( tt$MAC>=3 & tt$SE<10  & tt$INFO>=0.4 , grepl("SNP",tt$MarkerName) )

table( grepl("SNP",clean4$MarkerName) )
table( tt$MAC>=3 & tt$SE<10  & tt$INFO>=0.4 & !is.na(tt$BETA) & !is.na(tt$SE) , grepl("SNP",tt$MarkerName) )



#genotype variants:
clean4_geno=subset(clean4, clean4$IMPUTED==0)
#guideline: CALL_RATE_QTL >=0.95 for genotyped variants
if (!all(is.na(clean4_geno$CALL_RATE_QT)))
{
  clean5_geno=subset(clean4_geno, as.numeric(clean4_geno$CALL_RATE_QTL)>=0.95)
}
if (all(is.na(clean4_geno$CALL_RATE_QT)))
{
        clean5_geno=clean4_geno
}
#guideline: HWE_P_QTL >=0.95 for genotyped variants
#if (all(is.na(clean4_geno$HWE_P_QTL))) ### ---> ERROR
if (!all(is.na(clean4_geno$HWE_P_QTL)))
{
        clean6_geno=subset(clean5_geno, as.numeric(clean5_geno$clean5$HWE_P_QTL)>=1e-6)
}
#if (!all(is.na(clean4_geno$HWE_P_QTL)))  ### ---> ERROR
if (all(is.na(clean4_geno$HWE_P_QTL)))
{
        clean6_geno=clean5_geno
}
#imputed variants: (guideline: for imputed variants, INFO has to be available)
clean4_imputed=subset(clean4, clean4$IMPUTED==1)
clean5_imputed=subset(clean4_imputed, is.na(clean4_imputed$INFO)==FALSE)
#unknown imputed variants: (guideline: for variants unknown to be imputed or genotyped, INFO has to be available)
clean4_unkimputed=subset(clean4, is.na(clean4$IMPUTED)==TRUE)
clean5_unkimputed=subset(clean4_unkimputed, is.na(clean4_unkimputed$INFO)==FALSE)
#combine the data:
clean5and6=rbind(clean6_geno, clean5_imputed, clean5_unkimputed)



## AASC discrepancy with Katie

tt = read.table(gzfile("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_AASC_MINIMACH_25102014_YT.txt.gz"),as.is=T,h=T,na.strings=".")
table( tt$HWE_P_QTL<0.000001 , tt$IMPUTED , exclude=c() )
#              0       1    <NA>
#  FALSE  518553 7168173       0
#  TRUE     3038   62851       0  # EasyQC found 64763 with low HWE & 2983 genotyped with low HWE
#  <NA>        9     831       0


tt = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_AASC/EAA_GWAS_AX_HbA1cadjFGlu_AASC_MINIMACH_25102014_YT.Harmo.N_genotypedLowHWE_SeqClean.txt",as.is=T,h=T,na.strings=".")
summary(tt$HWE_P_QTL)
table( tt$HWE_P_QTL<0.000001 , tt$IMPUTED , exclude=c() )


tt = read.table(gzfile("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_AASC/EAA_GWAS_AX_HbA1cadjFGlu_AASC_MINIMACH_25102014_YT.gz"),as.is=T,h=T,na.strings=".")
table( tt$HWE_P_QTL<0.000001 , tt$IMPUTED , exclude=c() )
wrongMARKER = tt$MarkerName[which(!is.na(tt$HWE_P_QTL_low))]


table( ttC$HWE_P_QTL<0.000001 , ttC$IMPUTED , exclude=c() )
table( tt00$HWE_P_QTL<0.000001 , tt00$IMPUTED , exclude=c() )
tt00$clean=ifelse(tt00$MarkerName %in% ttC$MarkerName,yes=1,no=0)
table(tt00$MarkerName[which(tt00$clean==1)]==ttC$MarkerName)
table(tt00$HWE_P_QTL[which(tt00$clean==1)]==ttC$HWE_P_QTL)



zcat QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_AASC_MINIMACH_25102014_YT.txt.gz | awk 'BEGIN{OFS="\t"}{ if ($12==".") {sub(/./,"NA",$12)} ; {print$0} }' | bgzip -c > QC2_AASC/O_EAA_GWAS_AX_HbA1cadjFGlu_AASC_MINIMACH_25102014_YT.txt.gz




# Prepare QT SNP file for MCHIP lambda
tt=read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/Metabochip_QT_SNPs.txt",h=T,as.is=T)
dim(tt)
head(tt)
table(tt$trait)
table(substr(tt$snp,1,2))
table(substr(tt$chrpos_b37,1,5))
tt$MarkerName = paste(sub("chr","",tt$chrpos_b37),":SNP",sep="")
head(tt)
write.table(tt,"/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/Metabochip_QT_SNPs_withMarkerName.txt",quote=F,col.names=T,row.names=F,sep="\t")


zcat QC2_download/AA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz | awk '{split($1,a,":") ; print a[3]}' | sort | uniq
zcat QC2_download/EA_MCHIP_A_HbA1cadjFGlu_Ely_SNPTEST_15122014_JAL.txt.gz | awk '{split($1,a,":") ; print a[3]}' | sort | uniq

####### /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/QC2_EasyQC_test_Post.ecf
####### !!! the input files are the same than for PreQC/Cleaning script EXCEPT than the folder is now QC2_cleaned/ and the file are xxx.gz (not xxx.txt.gz!)
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/EA_GWAS_X_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/EA_GWAS_X_HbA1cadjFGlu_InvNorm_SardiNIA_EPACTS_22092014_GP.gz
### Set parameters and input files
DEFINE --strMissing . --strSeparator TAB --pathOut /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned --acolIn MarkerName;EFFECT_ALLELE;NON_EFFECT_ALLELE;N_QTL;EAF_QTL;BETA;SE;PVAL;IMPUTED;INFO --acolInClasses character;character;character;numeric;numeric;numeric;numeric;numeric;numeric;numeric
### Start EasyQC
START EASYQC
# add/edit some columns
ADDCOL  --rcdAddCol  sapply( MarkerName , FUN=function(x){ return(strsplit(x,":")[[1]][1]) } ) --colOut CHR2
# summary on harmonised files
GETNUM --rcdGetNum (grepl("SNP",MarkerName))   --strGetNumName Clean.N_SNPs
GETNUM --rcdGetNum (grepl("INDEL",MarkerName) | grepl("MERGED_DEL",MarkerName)) --strGetNumName Clean.N_indels
GETNUM --rcdGetNum (CHR2 %in% 1:22)            --strGetNumName Clean.N_A
GETNUM --rcdGetNum (CHR2=="X")                 --strGetNumName Clean.N_X
GETNUM --rcdGetNum (CHR2=="XY")                --strGetNumName Clean.N_XY
GETNUM --rcdGetNum (IMPUTED==1)                --strGetNumName Clean.N_imputed
GETNUM --rcdGetNum (IMPUTED==0)                --strGetNumName Clean.N_genotyped
GETNUM --rcdGetNum (is.na(IMPUTED))            --strGetNumName Clean.N_unknownImpute
GETNUM --rcdGetNum (is.na(N_QTL))              --strGetNumName Clean.N_QTL_NA
GETNUM --rcdGetNum (is.na(EAF_QTL))            --strGetNumName Clean.EAF_QTL_NA
GETNUM --rcdGetNum (abs(BETA)>5)               --strGetNumName Clean.N_highBETA5
GETNUM --rcdGetNum (abs(BETA)>10)              --strGetNumName Clean.N_highBETA10
# More
EVALSTAT --colStat BETA  --strTag Clean
CALCULATE --rcdCalc min(N_QTL,na.rm=T)   --strCalcName Clean.MinN
CALCULATE --rcdCalc max(N_QTL,na.rm=T)   --strCalcName Clean.SampleSize
CALCULATE --rcdCalc median(SE)           --strCalcName Clean.MedianSE
# Lambda (Mine and EasyQC one)
CALCULATE --rcdCalc median(qchisq(PVAL,df=1,lower.tail=F),na.rm=T)/0.456 --strCalcName Clean.Lambda
GC --colPval PVAL --blnSuppressCorrection 1
#!! CAUTION for Lambda: for MCHIP  --fileGcSnps /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/Metabochip_QT_SNPs_withMarkerName.txt --colGcSnpsMarker MarkerName --colInMarker MarkerName
### Stop EasyQC
STOP EASYQC
####################################################################################
#PVAL2 = pchisq((BETA/SE)^2,df=1,lower.tail=F)
#sum( abs(PVAL2-PVAL)>0.05 )
#sum( abs(PVAL2-PVAL)>0.02 )


library(EasyQC)
EasyQC("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/QC2_EasyQC_test_Post.ecf")
# bsub -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_EasyQC_test_Post.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_EasyQC_test_Post.err -R"select[mem>1000] rusage[mem=1000]" -M1000 "/software/R-3.0.0/bin/R --no-save --args QC2_EasyQC_test_Post.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"


###################
## Inputs for EasyQC
#################

### GWAS_EA_R
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_AGES_PROBABEL_23092014_AVS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_BioMe_SNPTEST2_20032014_CS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_CRC_MACH_09102014_TH.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_DES_SNPTEST_22042014_CL.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_DGI_SNPTEST_23122014_CL.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_Fenland_SNPTEST_14052014_SW.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_Fenland-OMICS_SNPTEST_28022015_PZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_FHS_MACH_01092015_AP.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_GOYA-Male-CASES_QUICKTEST_13112014_TSA.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_GOYA-Male-CONTROLS_QUICKTEST_13112014_TSA.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_KORA_SNPTEST_15012015_CM.SW.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_LifeLines_PLINK_27102014_PJM.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_LOLIPOP-EW610_SNPTEST_27052014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_LOLIPOP-EWA_SNPTEST_08052014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_LOLIPOP-EWP_SNPTEST_08052014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_LURIC_SNPTEST_27112014_MK.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_METSIM_EPACTSEMMAX_10042014_XS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_NTR_PLINK_08042014_JJH.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_XF_HbA1cadjFGlu_NTR_PLINK_08052014_JJH.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_XM_HbA1cadjFGlu_NTR_PLINK_08052014_JJH.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_ORCADES_PROBABEL_29052014_KES.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_X_HbA1cadjFGlu_SardiNIA_EPACTS_22092014_GP.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_TRAILSpop_SNPTEST_12062014_PJ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_TWINGENE_EMMAX_02062014_SG.txt.gz
### GWAS_EA_I
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_AGES_PROBABEL_23092014_AVS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_BioMe_SNPTEST2_20032014_CS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_CRC_MACH_09102014_TH.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_DES_SNPTEST_22042014_CL.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_DGI_SNPTEST_23122014_CL.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_Fenland_SNPTEST_14052014_SW.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_Fenland-OMICS_SNPTEST_28022015_PZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_FHS_MACH_01092015_AP.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_GOYA-Male-CASES_QUICKTEST_13112014_TSA.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_GOYA-Male-CONTROLS_QUICKTEST_13112014_TSA.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_KORA_SNPTEST_15012015_CM.SW.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LifeLines_PLINK_27102014_PJM.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-EW610_SNPTEST_27052014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-EWA_SNPTEST_08052014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-EWP_SNPTEST_08052014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_LURIC_SNPTEST_07112014_MK.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_MESA_SNPTEST_29122014_JY.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_METSIM_EPACTSEMMAX_10042014_XS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_NTR_PLINK_08042014_JJH.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_XF_HbA1cadjFGlu_InvNorm_NTR_PLINK_08052014_JJH.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_XM_HbA1cadjFGlu_InvNorm_NTR_PLINK_08052014_JJH.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_ORCADES_PROBABEL_29052014_KES.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_PROCARDIS_NOTGIVEN_14062014_OM.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_A_HbA1cadjFGlu_InvNorm_SardiNIA_EPACTS_22092014_GP.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_X_HbA1cadjFGlu_InvNorm_SardiNIA_EPACTS_22092014_GP.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_TRAILSpop_SNPTEST_12062014_PJ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_GWAS_AX_HbA1cadjFGlu_InvNorm_TWINGENE_EMMAX_02062014_SG.txt.gz
### GWAS_nEA_R
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/AA_GWAS_AX_HbA1cadjFGlu_BioMe_SNPTEST2_20032014_CS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/AA_GWAS_AX_HbA1cadjFGlu_HANDLS_MACH_24122014_SMT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/AA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_AASC_MINIMACH_25102014_YT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_CAGE-KING-Omni_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_XF_HbA1cadjFGlu_CAGE-KING-Omni_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_XM_HbA1cadjFGlu_CAGE-KING-Omni_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_CAGE-KING-Quad_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_XF_HbA1cadjFGlu_CAGE-KING-Quad_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_XM_HbA1cadjFGlu_CAGE-KING-Quad_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_CAGEGWAS1_PLINK_14102014_FT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_KARE_MACH2QTL_31102014_YJK.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_NAGAHAMA_MINIMACH_12262014_YT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_NHAPC_SNPTEST_19092014_YQ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_SP2-1M_SNPTEST_15092014_EB.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_SP2-610_SNPTEST_13092014_EB.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/HA_GWAS_AX_HbA1cadjFGlu_BioMe_SNPTEST2_20032014_CS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/HA_GWAS_A_HbA1cadjFGlu_IRAS_SNPTEST_02032015_JTZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/HA_GWAS_AX_HbA1cadjFGlu_MESA_SNPTEST_29122014_JY.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/HA_GWAS_AX_HbA1cadjFGlu_SOL_MIXEDMODEL_20022015_TL.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_LOLIPOP-IA317_SNPTEST_16062014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_LOLIPOP-IA610_SNPTEST_08052014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_LOLIPOP-IAP_SNPTEST_16062014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_LOLIPOP-OmniEE_SNPTEST_16062014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_RHS_SNPTEST_04012015_FT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_RHS-610K_SNPTEST_04012015_FT.txt.gz
### GWAS_nEA_I
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/AA_GWAS_AX_HbA1cadjFGlu_InvNorm_BioMe_SNPTEST2_20032014_CS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/AA_GWAS_AX_HbA1cadjFGlu_InvNorm_HANDLS_MACH_24122014_SMT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/AA_GWAS_AX_HbA1cadjFGlu_InvNorm_MESA_SNPTEST_29122014_JY.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_AASC_MINIMACH_25102014_YT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_CAGE-KING-Omni_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_XF_HbA1cadjFGlu_InvNorm_CAGE-KING-Omni_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_XM_HbA1cadjFGlu_InvNorm_CAGE-KING-Omni_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_CAGE-KING-Quad_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_XF_HbA1cadjFGlu_InvNorm_CAGE-KING-Quad_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_XM_HbA1cadjFGlu_InvNorm_CAGE-KING-Quad_MACH_03102014_MN.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_CAGEGWAS1_PLINK_14102014_FT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_KARE_MACH2QTL_31102014_YJK.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_MESA_SNPTEST_29122014_JY.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_NAGAHAMA_MINIMACH_12262014_YT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_NHAPC_SNPTEST_19092014_YQ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_SP2-1M_SNPTEST_11092014_EB.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_SP2-610_SNPTEST_06092014_EB.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_GWAS_A_HbA1cadjFGlu_InvNorm_VanderbiltGWAS6548_MACH2QTL_11092014_HMR.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/HA_GWAS_AX_HbA1cadjFGlu_InvNorm_BioMe_SNPTEST2_20032014_CS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/HA_GWAS_A_HbA1cadjFGlu_InvNorm_IRAS_SNPTEST_02032015_JTZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/HA_GWAS_AX_HbA1cadjFGlu_InvNorm_MESA_SNPTEST_29122014_JY.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/HA_GWAS_AX_HbA1cadjFGlu_InvNorm_SOL_MIXEDMODEL_20022015_TL.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-IA317_SNPTEST_16062014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-IA610_SNPTEST_08052014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-IAP_SNPTEST_16062014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_LOLIPOP-OmniEE_SNPTEST_16062014_WZ.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_RHS_SNPTEST_04012015_FT.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/IAA_GWAS_A_HbA1cadjFGlu_InvNorm_RHS-610K_SNPTEST_04012015_FT.txt.gz
### MCHIP
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_D2D2007.DPS_EPACTSEMMAX_10042014_XS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_DIAGEN_EPACTSEMMAX_10042014_XS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_Ely_SNPTEST_15122014_JAL.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_GoDARTS_IMPUTE_06102014_APM.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_Health2006_SNPTEST_16122014_NG.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_Inter99_SNPTEST_16122014_NG.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_LEIPZIG-adults_SNPTEST_31112014_AMah.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_LEIPZIG-kids_SNPTEST_31112014_AMah.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_MCHIP_A_HbA1cadjFGlu_TAICHI_SNPTEST_11022015_YFC.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_D2D2007.DPS_EPACTSEMMAX_10042014_XS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_DIAGEN_EPACTSEMMAX_10042014_XS.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_Ely_SNPTEST_15122014_JAL.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_GoDARTS_IMPUTE_06102014_APM.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_Health2006_SNPTEST_16122014_NG.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_Inter99_SNPTEST_16122014_NG.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_LEIPZIG-adults_SNPTEST_31112014_AMah.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EA_MCHIP_A_HbA1cadjFGlu_InvNorm_LEIPZIG-kids_SNPTEST_31112014_AMah.txt.gz
EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_download/EAA_MCHIP_A_HbA1cadjFGlu_InvNorm_TAICHI_SNPTEST_11022015_YFC.txt.gz


###################
## RUN EasyQC
#################

# Pre and cleaning

vi QC2_EasyQC_scripts/GWAS_EA_R_Pre.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Pre.ecf QC2_EasyQC_scripts/GWAS_EA_I_Pre.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Pre.ecf QC2_EasyQC_scripts/GWAS_nEA_R_Pre.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Pre.ecf QC2_EasyQC_scripts/GWAS_nEA_I_Pre.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Pre.ecf QC2_EasyQC_scripts/MCHIP_Pre.ecf


bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_R.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_R.err  -R"select[mem>40000] rusage[mem=40000]" -M40000 "/software/R-3.0.0/bin/R --no-save --args GWAS_EA_R_Pre.ecf  < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_I.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_I.err  -R"select[mem>40000] rusage[mem=40000]" -M40000 "/software/R-3.0.0/bin/R --no-save --args GWAS_EA_I_Pre.ecf  < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_R.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_R.err -R"select[mem>45000] rusage[mem=45000]" -M45000 "/software/R-3.0.0/bin/R --no-save --args GWAS_nEA_R_Pre.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_I.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_I.err -R"select[mem>45000] rusage[mem=45000]" -M45000 "/software/R-3.0.0/bin/R --no-save --args GWAS_nEA_I_Pre.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_MCHIP.out      -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_MCHIP.err      -R"select[mem>28000] rusage[mem=28000]" -M28000 "/software/R-3.0.0/bin/R --no-save --args MCHIP_Pre.ecf      < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_Vanderbilt.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_Vanderbilt.err -R"select[mem>35000] rusage[mem=35000]" -M35000 "/software/R-3.0.0/bin/R --no-save --args GWAS_Vanderbilt_Pre.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"

bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_AASC_O.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_AASC_O.err -R"select[mem>9000] rusage[mem=9000]" -M9000 "/software/R-3.0.0/bin/R --no-save --args GWAS_AASC_O.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"


bjobs -l | grep "MAX MEM"
grep Success QC2_EasyQC_scripts/bsub_Pre*

grep Processing QC2_EasyQC_scripts/bsub_Pre_MCHIP.out | wc
grep -i Processing QC2_EasyQC_scripts/bsub_Pre_MCHIP.out | wc
grep "Max Memory" QC2_EasyQC_scripts/bsub_Pre*

grep EASYIN QC2_EasyQC_scripts/GWAS_EA_R_Pre.ecf | wc
grep EASYIN QC2_EasyQC_scripts/GWAS_EA_I_Pre.ecf | wc
grep EASYIN QC2_EasyQC_scripts/GWAS_nEA_R_Pre.ecf | wc
grep EASYIN QC2_EasyQC_scripts/GWAS_nEA_I_Pre.ecf | wc
grep EASYIN QC2_EasyQC_scripts/MCHIP_Pre.ecf | wc

grep Processing /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_R.out | wc
grep Processing /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_I.out | wc
grep Processing /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_R.out | wc
grep Processing /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_I.out | wc
grep Processing /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_MCHIP.out | wc
grep Processing /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_Vanderbilt.out | wc

grep -i processing QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_R.out
grep -i processing QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_I.out
grep -i processing QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_R.out
grep -i processing QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_I.out


grep "Elapsed time" QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_R.out
grep "Elapsed time" QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_I.out
grep "Elapsed time" QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_R.out
grep "Elapsed time" QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_I.out
grep "Elapsed time" QC2_EasyQC_scripts/bsub_Pre_MCHIP.out

grep "Max Memory" QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_R.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_I.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_R.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_I.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_Pre_MCHIP.out


#cp QC2_cleaned/* ESSAI/
#bsub -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CleaningAfterPre_MCHIP.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CleaningAfterPre_MCHIP.err -R"select[mem>500] rusage[mem=500]" -M500 "for f in QC2_cleaned/*MCHIP*.gz; do f1=\${f/.gz/}; cat \$f1.*.txt | awk '{ if(NR==1 || \$1!=\"MarkerName\" ) {print \$0} }' | bgzip -c > \$f1.EXCLUDE.txt.gz ; rm \$f1.*.txt ; done"


# check the files with columns transformed to character because EasyQC found "."!!

bsub = c( "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_R.out" ,
          "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_EA_I.out" ,
          "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_R.out" ,
          "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_GWAS_nEA_I.out" ,
          "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Pre_MCHIP.out" )

colS=c("MarkerName","SNP","CHR","POS","EFFECT_ALLELE","NON_EFFECT_ALLELE",
       "N_QTL","N0_QTL","N1_QTL","N2_QTL","EAF_QTL","HWE_P_QTL","CALL_RATE_QTL",
       "BETA","SE","PVAL","IMPUTED","INFO_TYPE","INFO")

FF=NULL
for (oo in bsub)
{
  tt=system( paste("cat",oo),intern=T)
  ll=grep("rowsBumped",tt)
  if ( length(ll)>1 )
  {
   dd = (ll-1) %in% ll # whether are for the same file than the previous one
   cc=as.numeric(sapply( tt[ll] , FUN=function(x){x1=strsplit(x," ",fixed=T)[[1]]; y=grep("column",x1)[1] ; return(x1[y+1]) } )) # which column of the file has the PB
   ff=NULL
   for ( i in 1:length(ll) )
   {
    if ( !(dd[i]) )
    {
     ff=c(ff,tt[ll[i]-1])
    } else { ww=max(which(dd[1:i]==F))
             ff=c(ff,tt[ll[ww]-1])
             rm(ww)
           }
   } ; rm(i)
   FF = rbind(FF,cbind( ff , colS[cc] ))
   rm(dd,cc,ff)
  }
  rm(tt,ll)
} ; rm(oo)


#rename/zip the files produced by EasyQC

cd QC2_cleaned/
for f in `ls *gz` ; do f2=${f/.gz/.txt.gz} ; mv $f $f2 ; done
for f in `ls *.Harmo*` ; do bgzip $f ; done
cd ../
mkdir QC2_ListRemovedVariants/
mv QC2_cleaned/*.Harmo* QC2_ListRemovedVariants/

#Summary Post cleaning

vi QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf QC2_EasyQC_scripts/GWAS_EA_I_Post0.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf QC2_EasyQC_scripts/GWAS_nEA_R_Post0.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf QC2_EasyQC_scripts/GWAS_nEA_I_Post0.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf QC2_EasyQC_scripts/MCHIP_Post0.ecf

grep EASYIN QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf | wc
grep EASYIN QC2_EasyQC_scripts/GWAS_EA_I_Post0.ecf | wc
grep EASYIN QC2_EasyQC_scripts/GWAS_nEA_R_Post0.ecf | wc
grep EASYIN QC2_EasyQC_scripts/GWAS_nEA_I_Post0.ecf | wc
grep EASYIN QC2_EasyQC_scripts/MCHIP_Post0.ecf | wc

grep QC2_download QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf | wc
grep QC2_download QC2_EasyQC_scripts/GWAS_EA_I_Post0.ecf | wc
grep QC2_download QC2_EasyQC_scripts/GWAS_nEA_R_Post0.ecf | wc
grep QC2_download QC2_EasyQC_scripts/GWAS_nEA_I_Post0.ecf | wc
grep QC2_download QC2_EasyQC_scripts/MCHIP_Post0.ecf | wc
grep .txt.gz QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf | wc
grep .txt.gz QC2_EasyQC_scripts/GWAS_EA_I_Post0.ecf | wc
grep .txt.gz QC2_EasyQC_scripts/GWAS_nEA_R_Post0.ecf | wc
grep .txt.gz QC2_EasyQC_scripts/GWAS_nEA_I_Post0.ecf | wc
grep .txt.gz QC2_EasyQC_scripts/MCHIP_Post0.ecf | wc

sed "s/QC2_download/QC2_cleaned/g" QC2_EasyQC_scripts/GWAS_EA_R_Post0.ecf | sed "s/.txt.gz/.gz/g" > QC2_EasyQC_scripts/GWAS_EA_R_Post.ecf
sed "s/QC2_download/QC2_cleaned/g" QC2_EasyQC_scripts/GWAS_EA_I_Post0.ecf | sed "s/.txt.gz/.gz/g" > QC2_EasyQC_scripts/GWAS_EA_I_Post.ecf
sed "s/QC2_download/QC2_cleaned/g" QC2_EasyQC_scripts/GWAS_nEA_R_Post0.ecf | sed "s/.txt.gz/.gz/g" > QC2_EasyQC_scripts/GWAS_nEA_R_Post.ecf
sed "s/QC2_download/QC2_cleaned/g" QC2_EasyQC_scripts/GWAS_nEA_I_Post0.ecf | sed "s/.txt.gz/.gz/g" > QC2_EasyQC_scripts/GWAS_nEA_I_Post.ecf
sed "s/QC2_download/QC2_cleaned/g" QC2_EasyQC_scripts/MCHIP_Post0.ecf | sed "s/.txt.gz/.gz/g" > QC2_EasyQC_scripts/MCHIP_Post.ecf

rm QC2_EasyQC_scripts/*Post0.ecf

bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_GWAS_EA_R.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_GWAS_EA_R.err  -R"select[mem>15000] rusage[mem=15000]" -M15000 "/software/R-3.0.0/bin/R --no-save --args GWAS_EA_R_Post.ecf  < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_GWAS_EA_I.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_GWAS_EA_I.err  -R"select[mem>15000] rusage[mem=15000]" -M15000 "/software/R-3.0.0/bin/R --no-save --args GWAS_EA_I_Post.ecf  < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_R.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_R.err -R"select[mem>15000] rusage[mem=15000]" -M15000 "/software/R-3.0.0/bin/R --no-save --args GWAS_nEA_R_Post.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_I.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_I.err -R"select[mem>15000] rusage[mem=15000]" -M15000 "/software/R-3.0.0/bin/R --no-save --args GWAS_nEA_I_Post.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_MCHIP.out      -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_MCHIP.err      -R"select[mem>8000]  rusage[mem=8000]"  -M8000  "/software/R-3.0.0/bin/R --no-save --args MCHIP_Post.ecf      < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_Vanderbilt.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_Post_Vanderbilt.err -R"select[mem>10000] rusage[mem=10000]" -M10000 "/software/R-3.0.0/bin/R --no-save --args GWAS_Vanderbilt_Post.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"

bjobs -l | grep "MAX MEM"
grep Success QC2_EasyQC_scripts/bsub_Post*


grep Processing QC2_EasyQC_scripts/bsub_Post_MCHIP.out | wc
grep -i Processing QC2_EasyQC_scripts/bsub_Post_MCHIP.out | wc


grep Processing QC2_EasyQC_scripts/bsub_Post_GWAS_EA_R.out | wc
grep Processing QC2_EasyQC_scripts/bsub_Post_GWAS_EA_I.out | wc
grep Processing QC2_EasyQC_scripts/bsub_Post_MCHIP.out | wc
grep Processing QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_R.out | wc
grep Processing QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_I.out | wc

ls -lh QC2_EasyQC_scripts/bsub_Post*


grep "Elapsed time" QC2_EasyQC_scripts/bsub_Post_GWAS_EA_R.out
grep "Elapsed time" QC2_EasyQC_scripts/bsub_Post_GWAS_EA_I.out
grep "Elapsed time" QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_R.out
grep "Elapsed time" QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_I.out
grep "Elapsed time" QC2_EasyQC_scripts/bsub_Post_MCHIP.out

grep "Max Memory" QC2_EasyQC_scripts/bsub_Post_GWAS_EA_R.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_Post_GWAS_EA_I.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_R.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_Post_GWAS_nEA_I.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_Post_MCHIP.out


## how many variants with abs(beta)>2.4
cp QC2_EasyQC_scripts/GWAS_EA_I_Post.ecf  QC2_EasyQC_scripts/GWAS_EA_I_check.ecf
cp QC2_EasyQC_scripts/GWAS_EA_R_Post.ecf  QC2_EasyQC_scripts/GWAS_EA_R_check.ecf
cp QC2_EasyQC_scripts/GWAS_nEA_I_Post.ecf QC2_EasyQC_scripts/GWAS_nEA_I_check.ecf
cp QC2_EasyQC_scripts/GWAS_nEA_R_Post.ecf QC2_EasyQC_scripts/GWAS_nEA_R_check.ecf
cp QC2_EasyQC_scripts/MCHIP_Post.ecf      QC2_EasyQC_scripts/MCHIP_check.ecf

bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_GWAS_EA_R.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_GWAS_EA_R.err  -R"select[mem>15000] rusage[mem=15000]" -M15000 "/software/R-3.0.0/bin/R --no-save --args GWAS_EA_R_check.ecf  < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_GWAS_EA_I.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_GWAS_EA_I.err  -R"select[mem>15000] rusage[mem=15000]" -M15000 "/software/R-3.0.0/bin/R --no-save --args GWAS_EA_I_check.ecf  < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_GWAS_nEA_R.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_GWAS_nEA_R.err -R"select[mem>15000] rusage[mem=15000]" -M15000 "/software/R-3.0.0/bin/R --no-save --args GWAS_nEA_R_check.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_GWAS_nEA_I.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_GWAS_nEA_I.err -R"select[mem>15000] rusage[mem=15000]" -M15000 "/software/R-3.0.0/bin/R --no-save --args GWAS_nEA_I_check.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_MCHIP.out      -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_MCHIP.err      -R"select[mem>8000]  rusage[mem=8000]"  -M8000  "/software/R-3.0.0/bin/R --no-save --args MCHIP_check.ecf      < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"

grep Processing QC2_EasyQC_scripts/bsub_check_GWAS_EA_R.out | wc
grep Processing QC2_EasyQC_scripts/bsub_check_GWAS_EA_I.out | wc
grep Processing QC2_EasyQC_scripts/bsub_check_GWAS_nEA_R.out | wc
grep Processing QC2_EasyQC_scripts/bsub_check_GWAS_nEA_I.out | wc
grep Processing QC2_EasyQC_scripts/bsub_check_MCHIP.out | wc

grep "Max Memory" QC2_EasyQC_scripts/bsub_check_GWAS_EA_R.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_check_GWAS_EA_I.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_check_GWAS_nEA_R.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_check_GWAS_nEA_I.out
grep "Max Memory" QC2_EasyQC_scripts/bsub_check_MCHIP.out

bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_IRAS.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_check_IRAS.err  -R"select[mem>5000] rusage[mem=5000]" -M5000 "/software/R-3.0.0/bin/R --no-save --args IRAS_check.ecf  < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"


###################
## EasyQC REPORT
#################

EASYrep_Pre = rbind( read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/MCHIP_Pre.rep"     ,h=T,as.is=T) ,
                     read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_EA_R_Pre.rep"   ,h=T,as.is=T) ,
                     read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_EA_I_Pre.rep"   ,h=T,as.is=T) ,
                     read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_nEA_R_Pre.rep"  ,h=T,as.is=T) ,
                     read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_nEA_I_Pre.rep"  ,h=T,as.is=T) )
EASYrep_Pre = EASYrep_Pre[ -grep("Vanderbilt",EASYrep_Pre$fileInShortName) ,]
EASYrep_Pre = rbind( EASYrep_Pre ,
                     read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_Vanderbilt_Pre.rep" ,h=T,as.is=T) )

EASYrep_Post = rbind( read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/MCHIP_Post.rep"     ,h=T,as.is=T) ,
                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_EA_R_Post.rep" ,h=T,as.is=T) ,
                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_EA_I_Post.rep" ,h=T,as.is=T) ,
                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_nEA_R_Post.rep",h=T,as.is=T) ,
                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_nEA_I_Post.rep",h=T,as.is=T) )
EASYrep_Post = EASYrep_Post[ -grep("Vanderbilt",EASYrep_Post$fileInShortName) ,]
EASYrep_Post = rbind( EASYrep_Post ,
                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_Vanderbilt_Post.rep" ,h=T,as.is=T) )
EASYrep_Post$fileInShortName = sub(".gz",".txt.gz",EASYrep_Post$fileInShortName)

#EASYrep_check = rbind(
#                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/MCHIP_check.rep"     ,h=T,as.is=T) ,
#                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_EA_R_check.rep" ,h=T,as.is=T) ,
#                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_EA_I_check.rep" ,h=T,as.is=T) ,
#                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_nEA_R_check.rep",h=T,as.is=T) ,
#                      read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/GWAS_nEA_I_check.rep",h=T,as.is=T) )
#EASYrep_check$fileInShortName = sub(".gz",".txt.gz",EASYrep_check$fileInShortName)


EASYrep = merge( EASYrep_Pre , EASYrep_Post , by="fileInShortName" )
#EASYrep  = merge( EASYrep0   , EASYrep_check , by="fileInShortName" )
rm(EASYrep_Pre,EASYrep_Post)#,EASYrep_check,EASYrep0)

tp = sapply( EASYrep$fileInShortName , FUN=function(x){ x1=strsplit(x,"_")[[1]]
                                                        if( x1[5]=="InvNorm" ) {tt="InvNorm";ss=x1[6]} else {tt="Raw";ss=x1[5]}
                                                        return(c(Ancestry=x1[1],Chip=x1[2],Chr=x1[3],Trait=x1[4],Transf=tt,Study=ss))
                                                      } )

EASYrep$Ancestry = tp["Ancestry",]
EASYrep$Chip     = tp["Chip",]
EASYrep$Chr      = tp["Chr",]
EASYrep$Trait    = tp["Trait",]
EASYrep$Transf   = tp["Transf",]
EASYrep$Study    = tp["Study",]
rm(tp)

table(EASYrep$Ancestry)
table(EASYrep$Chip)
table(EASYrep$Chr)
table(EASYrep$Trait)
table(EASYrep$Transf)

# Inconsistencies between file name chromosome and actual chromosomes in the file
table( EASYrep$Chr %in% c("A","AX") , EASYrep$Harmo.N_A>0 )
table( EASYrep$Chr %in% c("A") , EASYrep$Harmo.N_X==0 & EASYrep$Harmo.N_XY==0 )
table( EASYrep$Chr %in% c("AX","X","XF","XM") , EASYrep$Harmo.N_X>0 | EASYrep$Harmo.N_XY>0 ) # 4 files / 2 studies
table( EASYrep$Chr %in% c("X","XF","XM") , EASYrep$Harmo.N_A==0 )

EASYrep[which( EASYrep$Chr %in% c("AX") & !(EASYrep$Harmo.N_X>0 | EASYrep$Harmo.N_XY>0) ),c("fileInShortName","numSNPsIn.x","numSNPsOut.x","Harmo.N_SNPs","Harmo.N_indels","Harmo.N_A","Harmo.N_X","Harmo.N_XY")] # 4 files / 2 studies : METSIM and TWINGENE have no X variants, should be named _A_

# No genotyped variants
table( EASYrep$Harmo.N_genotyped==0 )
EASYrep[which( EASYrep$Harmo.N_genotyped==0 ), c("fileInShortName","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute") ]

# Unknown imputed variants
table( EASYrep$Harmo.N_unknownImpute>0 )

table( EASYrep$Harmo.N_imputed==0 & EASYrep$Harmo.N_genotyped==0 , EASYrep$Harmo.N_unknownImpute>0  )
  # 26 files with no any IMPUTED available
  # 2 files with partial IMPUTED
EASYrep[which( !(EASYrep$Harmo.N_imputed==0 & EASYrep$Harmo.N_genotyped==0) & EASYrep$Harmo.N_unknownImpute>0 ), c("fileInShortName","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_A","Harmo.N_X","Harmo.N_XY") ]
  # partial IMPUTED for PROCARDIS
EASYrep[which( EASYrep$Harmo.N_imputed==0 & EASYrep$Harmo.N_genotyped==0 ), c("fileInShortName","Ancestry","Chip","Chr","Trait","Transf","Study",) ]
unique(EASYrep[which( EASYrep$Harmo.N_imputed==0 & EASYrep$Harmo.N_genotyped==0 ), c("Ancestry","Chip","Chr","Trait","Study") ])
  # no any IMPUTED for 13 studies : MESA (4 ancestries), LOLIPOP (7 studies), AGES , CRC
EASYrep[which( EASYrep$Harmo.N_imputed==0 & EASYrep$Harmo.N_genotyped==0 ), c("Study","Ancestry","Chip","Chr","Trait","Transf","Study","Harmo.N_unknownImputeWithNoInfo","Harmo.HWE_P_QTL_NA","Harmo.N_lowHWE","Harmo.CALL_RATE_QTL_NA","Harmo.N_lowCALLRATE","Harmo.INFO_NA","Harmo.N_lowINFO") ]



# Imputed variants with missing INFO
table( EASYrep$Harmo.N_imputedWithNoInfo>0 )
EASYrep[ which(EASYrep$Harmo.N_imputedWithNoInfo>0) , c("fileInShortName","Ancestry","Chip","Chr","Transf","Study","numSNPsIn.x","numSNPsOut.x","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_imputedWithNoInfo","Harmo.INFO_NA") ]
  # TAICHI : some imputed variants with no INFO (all genotyped have no INFO but not matter)
  # DES : all imputed have no INFO (and apparently genotyped have...)
  # VanderbiltGWAS6548: all variants are imputed, 20968 have no INFO --> are they the genotyped ones?

table( EASYrep$Harmo.N_imputedWithNoInfo>0 , EASYrep$Harmo.INFO_NA>0 )




# Unknown IMPUTED status with missing INFO
table( EASYrep$Harmo.N_unknownImputeWithNoInfo>0 )
EASYrep[which( EASYrep$Harmo.N_unknownImputeWithNoInfo>0 ), c("fileInShortName","Ancestry","Chip","Chr","Transf","Study","numSNPsIn.x","numSNPsOut.x","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_A","Harmo.N_X","Harmo.N_XY","Harmo.N_imputedWithNoInfo","Harmo.N_unknownImputeWithNoInfo","Harmo.INFO_NA") ]
  ## this is PROCARDIS


# Genotyped variants with missing HWE_P_QTL and low HWE_P_QTL (<10-6)
table( EASYrep$Harmo.N_genotypedWithNoHWE>0 )
table( EASYrep$Harmo.N_genotypedWithNoHWE>0 & EASYrep$Harmo.N_genotypedWithNoHWE<EASYrep$Harmo.N_genotyped )
table( EASYrep$Harmo.N_genotypedWithNoHWE>0 , EASYrep$Harmo.N_genotypedWithNoHWE<EASYrep$Harmo.N_genotyped )
   ## 42 files where all genotyped variants have missing HWE_P
   ## 6 files where genotyped variants have either available or missing HWE_P
tt1=EASYrep[ which(EASYrep$Harmo.N_genotypedWithNoHWE>0 & EASYrep$Harmo.N_genotypedWithNoHWE<EASYrep$Harmo.N_genotyped)  , c("fileInShortName","Ancestry","Chip","Chr","Transf","Study","numSNPsIn.x","numSNPsOut.x","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedWithNoHWE","Harmo.HWE_P_QTL_NA","Harmo.N_genotypedLowHWE","Harmo.N_lowHWE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)


table( EASYrep$Harmo.N_genotypedLowHWE>0 )
table( EASYrep$Harmo.N_lowHWE>0 )
table( EASYrep$Harmo.N_genotypedLowHWE>0 , EASYrep$Harmo.N_lowHWE>0 )

unique(EASYrep$Study[ which(EASYrep$Harmo.N_genotypedLowHWE>0 & EASYrep$Harmo.N_lowHWE>0) ])
unique(EASYrep$Study[ which(EASYrep$Harmo.N_genotypedLowHWE==0 & EASYrep$Harmo.N_lowHWE>0) ])
unique(EASYrep$Study[ which(EASYrep$Harmo.N_genotypedLowHWE>0 & EASYrep$Harmo.N_genotypedLowHWE<EASYrep$Harmo.N_lowHWE) ])



tt1=EASYrep[ which(EASYrep$Harmo.N_genotypedLowHWE>0)  , c("fileInShortName","Ancestry","Chip","Chr","Transf","Study","numSNPsIn.x","numSNPsOut.x","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedWithNoHWE","Harmo.HWE_P_QTL_NA","Harmo.N_genotypedLowHWE","Harmo.N_lowHWE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)
tt1=EASYrep[ which(EASYrep$Harmo.N_genotypedLowHWE==0 & EASYrep$Harmo.N_lowHWE>0) , c("Study","Ancestry","Chip","Chr","Trait","Transf","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedWithNoHWE","Harmo.HWE_P_QTL_NA","Harmo.N_genotypedLowHWE","Harmo.N_lowHWE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)




# Genotyped variants with missing CALL_RATE_QTL and low CALL RATE (<95%)
table( EASYrep$Harmo.N_genotypedWithNoCALLRATE>0 , EASYrep$Harmo.CALL_RATE_QTL_NA>0 , exclude=c() )
table( EASYrep$Harmo.N_genotypedWithNoCALLRATE>0 & EASYrep$Harmo.N_genotypedWithNoCALLRATE< EASYrep$Harmo.N_genotyped )
table( EASYrep$Harmo.N_genotypedLowCALLRATE>0 )
table( EASYrep$Harmo.N_lowCALLRATE>0 )

tt1=EASYrep[ which(EASYrep$Harmo.N_genotypedLowCALLRATE>0)  , c("fileInShortName","Ancestry","Chip","Chr","Transf","Study","numSNPsIn.x","numSNPsOut.x","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedWithNoCALLRATE","Harmo.CALL_RATE_QTL_NA","Harmo.N_genotypedLowCALLRATE","Harmo.N_lowCALLRATE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)
tt1=EASYrep[ which(EASYrep$Harmo.N_genotypedLowCALLRATE==0 & EASYrep$Harmo.N_lowCALLRATE>0) , c("Study","Ancestry","Chip","Chr","Trait","Transf","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedWithNoCALLRATE","Harmo.CALL_RATE_QTL_NA","Harmo.N_genotypedLowCALLRATE","Harmo.N_lowCALLRATE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)




# Missing MAC (ie missing EAF_QTL or N_QTL)
table( EASYrep$Harmo.MAC_NA>0 )

tt1=EASYrep[ which(EASYrep$Harmo.MAC_NA>0)  , c("Study","Ancestry","Chip","Chr","Trait","Transf","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.MAC_NA") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)

table( EASYrep$Harmo.N_QTL_NA>0 , EASYrep$Harmo.EAF_QTL_NA>0 )
table( EASYrep$Harmo.N_QTL_NA>0 , EASYrep$Clean.N_QTL_NA>0 )
table( EASYrep$Harmo.EAF_QTL_NA>0 , EASYrep$Clean.EAF_QTL_NA>0 )

tt1=EASYrep[ which(EASYrep$Harmo.N_QTL_NA>0 | EASYrep$Harmo.EAF_QTL_NA>0)  , c("fileInShortName","Ancestry","Chip","Chr","Transf","Study","numSNPsIn.x","numSNPsOut.x","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_A","Harmo.N_X","Harmo.N_XY","Harmo.N_QTL_NA","Clean.N_QTL_NA","Harmo.EAF_QTL_NA","Clean.EAF_QTL_NA") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)



# Missing or unexpected PVAL
table( EASYrep$Harmo.PVAL_NA>0 )
table( EASYrep$Harmo.PVAL_min<0 )
table( EASYrep$Harmo.PVAL_max>1 )

tt1=EASYrep[ which(EASYrep$Harmo.PVAL_NA>0 | EASYrep$Harmo.PVAL_min<0 | EASYrep$Harmo.PVAL_max>1)  , c("fileInShortName","Study","Ancestry","Chip","Chr","Trait","Transf","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.PVAL_NA","Harmo.PVAL_min","Harmo.PVAL_max") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)

table( EASYrep$Harmo.SE_min<0 )

tt1=EASYrep[ which(EASYrep$Harmo.SE_min<0 )  , c("fileInShortName","Study","Ancestry","Chip","Chr","Trait","Transf","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.PVAL_NA","Harmo.PVAL_min","Harmo.PVAL_max","Harmo.SE_min") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)


table( EASYrep$Harmo.MAC_min<0 )
table( EASYrep$Harmo.HWE_P_QTL_min<0 )
table( EASYrep$Harmo.HWE_P_QTL_max>1 )
table( EASYrep$Harmo.CALL_RATE_QTL_min<0 )
table( EASYrep$Harmo.CALL_RATE_QTL_max>1 )
table( EASYrep$Harmo.INFO_min<0 )
table( EASYrep$Harmo.INFO_max>1 )
table( EASYrep$Harmo.EAF_QTL_min<0 )
table( EASYrep$Harmo.EAF_QTL_max>1 )
table( EASYrep$Harmo.EAF_QTL_NA>0 , exclude=c() )




# Missing SE
table( EASYrep$Harmo.SE_NA>0 )
# Missing BETA
table( EASYrep$Harmo.BETA_NA>0 )

tt1=EASYrep[ which(EASYrep$Harmo.SE_NA>0 | EASYrep$Harmo.BETA_NA>0)  , c("fileInShortName","Ancestry","Chip","Chr","Transf","Study","numSNPsIn.x","numSNPsOut.x","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.BETA_NA","Harmo.SE_NA") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)




# Unknown imputed variants in cleaned filed
table( EASYrep$Clean.N_unknownImpute>0 )

# High BETS
table( EASYrep$Clean.N_highBETA5>0 )
table( EASYrep$Clean.N_highBETA10>0 )

tt1=EASYrep[ which(EASYrep$Clean.N_highBETA5>0 )  , c("Study","Ancestry","Chip","Chr","Transf","Clean.N_highBETA5","Clean.N_highBETA10","Harmo.BETA_NA","Harmo.BETA_min","Harmo.BETA_max") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)


summary(EASYrep$Clean.N_highBETA2.4)
summary(EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn)
plot(EASYrep$numSNPsIn,EASYrep$Clean.N_highBETA2.4)
boxplot(EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn)
abline(h=0.001,col="red")
abline(h=0.0001,col="red")
table( EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.001 )
table( EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.0001 )
table( EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.0005 )

plot(EASYrep$numSNPsIn,EASYrep$Clean.N_highBETA2.4,
     col=ifelse(EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.0005,yes="red",no="black"),
     pch=ifelse(EASYrep$Trans=="Raw",yes=20,no=4) )

table( EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.0005 , EASYrep$Clean.N_highBETA2.4<10000 )
table( EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.0005 , EASYrep$numSNPsIn<5000000 )

EASYrep[which(EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.0005 & EASYrep$numSNPsIn<5000000),c("fileInShortName","numSNPsIn","Clean.N_highBETA2.4","Clean.SampleSize")]
EASYrep[which(EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.0005),c("fileInShortName","numSNPsIn","Clean.N_highBETA2.4","Clean.SampleSize")]

plot(EASYrep$Clean.SampleSize,EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn,col=ifelse(EASYrep$Clean.N_highBETA2.4/EASYrep$numSNPsIn>0.0005,yes="red",no="black"))

table( EASYrep$Clean.N_highBETA2.4>100 , EASYrep$Transf )
table( EASYrep$Clean.N_highBETA2.4>100 , EASYrep$Chr )



# variants with low sample size
table( EASYrep$Clean.MinN<=30 )
table( EASYrep$Clean.N_missingN>0 )

tt1=EASYrep[ which(EASYrep$Clean.N_missingN>0 )  , c("Study","Ancestry","Chip","Chr","Transf","Clean.N_missingN","Harmo.MAC_NA") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)



# high/low Lambdas
table( EASYrep$Clean.Lambda<0.9 , EASYrep$Lambda.PVAL.GC<0.9 )
table( EASYrep$Clean.Lambda>1.1 , EASYrep$Lambda.PVAL.GC>1.1 )

plot( EASYrep$Clean.Lambda , EASYrep$Lambda.PVAL.GC , xlab="/0.456" , ylab="/qchisq(0.5,df=1)" , xlim=c(min(c(0.5,EASYrep$Clean.Lambda)),max(c(1.5,EASYrep$Clean.Lambda))) , ylim=c(min(c(0.5,EASYrep$Lambda.PVAL.GC)),max(c(1.5,EASYrep$Lambda.PVAL.GC))) )
abline(0,1,col="grey") ; abline( h=c(0.9,1.1) , v=c(0.9,1.1) , col="red" )

tt1=EASYrep[ which(EASYrep$Lambda.PVAL.GC>1.1 | EASYrep$Lambda.PVAL.GC<0.9 )  , c("Study","Ancestry","Chip","Chr","Transf","Clean.Lambda","Lambda.PVAL.GC") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)




## formatting
ee  =c("File", # columns asked in the guidelines
"Harmo.N_SNPs",
"Harmo.N_indels",
"Harmo.N_A",
"Harmo.N_X",
"Harmo.N_XY",
"Harmo.N_imputed",
"Harmo.N_genotyped",
"Harmo.N_unknownImpute",
"Harmo.N_lowMAC",
"Harmo.N_highSE",
"Harmo.N_lowINFO",
"Harmo.N_missingBETAorSE",
"Harmo.N_genotypedLowHWE",
"Harmo.N_genotypedLowCALLRATE",
"Harmo.N_imputedWithNoInfo",
"Harmo.N_unknownImputeWithNoInfo",
"Harmo.N_mono",
"Harmo.N_genotypedWithNoHWE",
"Harmo.N_genotypedWithNoCALLRATE",
"Harmo.N_imputedLowCALLRATE",
"Harmo.N_QTL_NA",
"Harmo.HWE_P_QTL_mean",
"Harmo.HWE_P_QTL_median",
"Harmo.HWE_P_QTL_p25",
"Harmo.HWE_P_QTL_p75",
"Harmo.HWE_P_QTL_min",
"Harmo.HWE_P_QTL_max",
"Harmo.HWE_P_QTL_NA",
"Harmo.CALL_RATE_QTL_mean",
"Harmo.CALL_RATE_QTL_median",
"Harmo.CALL_RATE_QTL_p25",
"Harmo.CALL_RATE_QTL_p75",
"Harmo.CALL_RATE_QTL_min",
"Harmo.CALL_RATE_QTL_max",
"Harmo.CALL_RATE_QTL_NA",
"Harmo.MAC_mean",
"Harmo.MAC_median",
"Harmo.MAC_p25",
"Harmo.MAC_p75",
"Harmo.MAC_min",
"Harmo.MAC_max",
"Harmo.MAC_NA",
"Harmo.INFO_mean",
"Harmo.INFO_median",
"Harmo.INFO_p25",
"Harmo.INFO_p75",
"Harmo.INFO_min",
"Harmo.INFO_max",
"Harmo.INFO_NA",
"Harmo.BETA_mean",
"Harmo.BETA_median",
"Harmo.BETA_p25",
"Harmo.BETA_p75",
"Harmo.BETA_min",
"Harmo.BETA_max",
"Harmo.BETA_NA",
"Harmo.SE_mean",
"Harmo.SE_median",
"Harmo.SE_p25",
"Harmo.SE_p75",
"Harmo.SE_min",
"Harmo.SE_max",
"Harmo.SE_NA",
"Harmo.PVAL_mean",
"Harmo.PVAL_median",
"Harmo.PVAL_p25",
"Harmo.PVAL_p75",
"Harmo.PVAL_min",
"Harmo.PVAL_max",
"Harmo.PVAL_NA",
"Harmo.EAF_QTL_mean",
"Harmo.EAF_QTL_median",
"Harmo.EAF_QTL_p25",
"Harmo.EAF_QTL_p75",
"Harmo.EAF_QTL_min",
"Harmo.EAF_QTL_max",
"Harmo.EAF_QTL_NA",
"Clean.N_SNPs",
"Clean.N_indels",
"Clean.N_A",
"Clean.N_X",
"Clean.N_XY",
"Clean.N_imputed",
"Clean.N_genotyped",
"Clean.N_unknownImpute",
"Clean.N_QTL_NA",
"Clean.EAF_QTL_NA",
"Clean.N_highBETA5",
"Clean.N_highBETA10",
"Clean.BETA_mean",
"Clean.BETA_median",
"Clean.BETA_p25",
"Clean.BETA_p75",
"Clean.BETA_min",
"Clean.BETA_max",
"Clean.MinN",
"Clean.SampleSize",
"Clean.MedianSE",
"Clean.Lambda")

colnames(EASYrep)[1]="File"
colnames(EASYrep)[which(colnames(EASYrep)=="Clean.Lambda")]="Clean.LambdaGaelle"
colnames(EASYrep)[which(colnames(EASYrep)=="Lambda.PVAL.GC")]="Clean.Lambda"

table( colnames(EASYrep) %in% ee )
table( ee %in% colnames(EASYrep) )

EASYrepf = EASYrep[,ee]

EASYrepf$CoVolunteer= ifelse( (EASYrep$Chip=="GWAS" & EASYrep$Ancestry %in% c("AA","EA")) | EASYrep$Study=="AASC" | (EASYrep$Study=="CAGE-KING-Omni" & EASYrep$Chr=="A") , yes="Katie" , no="Laura" )
EASYrep$CoVolunteer= ifelse( (EASYrep$Chip=="GWAS" & EASYrep$Ancestry %in% c("AA","EA")) | EASYrep$Study=="AASC" | (EASYrep$Study=="CAGE-KING-Omni" & EASYrep$Chr=="A") , yes="Katie" , no="Laura" )

EASYrepf = EASYrepf[order(EASYrep$Chip,EASYrep$Ancestry,EASYrep$Study),]

write.table(EASYrepf,"/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Summary_Gaelle.txt",quote=F,row.names=F,sep="\t")

## compare with Laura

gaelle = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Summary_Gaelle.txt",h=T,as.is=T)
#laura  = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Summary_LauraCorbin_for_Gaelle.txt",h=T,as.is=T)
laura  = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Summary_LauraCorbin_v2.txt",h=T,as.is=T)

dim(gaelle)
dim(laura)

table(colnames(gaelle)[1:100]==colnames(laura))

table(gaelle$File==laura$File)
table( gaelle$File %in% laura$File , gaelle$CoVolunteer )
table( laura$File %in% gaelle$File )

mm = merge(gaelle , laura , by="File" )

var=NULL
for ( v in colnames(laura)[-c(1)] )
{
 tt = table(mm[,paste(v,".x",sep="")]==mm[,paste(v,".y",sep="")],exclude=c())
 ttTR = as.numeric(tt["TRUE"] ) ; if(is.na(ttTR)) {ttTR=0}
 ttFA = as.numeric(tt["FALSE"]) ; if(is.na(ttFA)) {ttFA=0}
 ttNA = as.numeric(tt[which(is.na(names(tt)))])
 #if ( ttTR!=64 ) {cat( "\n",v,": " , ttFA ,"FALSE +",ttNA,"NA\n")}
 #if ( ttFA>0   ) {var=c(var,v)}
 if ( ttNA>0   ) { if(sum(is.na(mm[,paste(v,".x",sep="")])!=is.na(mm[,paste(v,".y",sep="")]),na.rm=T)>0) {var=c(var,v)}}
 rm(tt,ttTR,ttFA,ttNA)
} ; rm(v)

mm = mm[,c("File",paste(rep(colnames(laura)[-c(1)],each=2),c(".x",".y"),sep=""))]

write.table(mm,"/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Summary_Gaelle_vs_Laura.txt",quote=F,row.names=F,sep="\t")


mm[ which(mm$Harmo.N_highSE.x!=mm$Harmo.N_highSE.y) , c("File","Harmo.N_highSE.x","Harmo.N_highSE.y") ]
mm[ which(mm$Harmo.N_missingBETAorSE.x!=mm$Harmo.N_missingBETAorSE.y) , c("File","Harmo.N_missingBETAorSE.x","Harmo.N_missingBETAorSE.y") ]
mm[ which(mm$Harmo.N_genotypedLowHWE.x!=mm$Harmo.N_genotypedLowHWE.y) , c("File","Harmo.N_genotypedLowHWE.x","Harmo.N_genotypedLowHWE.y") ]
mm[ which(mm$Harmo.N_genotypedWithNoHWE.x!=mm$Harmo.N_genotypedWithNoHWE.y) , c("File","Harmo.N_genotypedWithNoHWE.x","Harmo.N_genotypedWithNoHWE.y") ]


mm[ which(mm$Harmo.INFO_mean.x!=mm$Harmo.INFO_mean.y) , c("File","Harmo.INFO_mean.x","Harmo.INFO_mean.y") ]

table( mm$Clean.Lambda.x!=mm$Clean.Lambda.y , grepl("MCHIP",mm$File) )
mm[ which(mm$Clean.Lambda.x!=mm$Clean.Lambda.y) , c("File","Clean.Lambda.x","Clean.Lambda.y") ]

mm[ which(mm$Harmo.N_lowINFO.x!=mm$Harmo.N_lowINFO.y) , c("File","Harmo.N_lowINFO.x","Harmo.N_lowINFO.y") ]
mm[ which(mm$Harmo.N_imputedWithNoInfo.x!=mm$Harmo.N_imputedWithNoInfo.y) , c("File","Harmo.N_imputedWithNoInfo.x","Harmo.N_imputedWithNoInfo.y") ]

head(mm[ which(mm$Harmo.HWE_P_QTL_mean.x!=mm$Harmo.HWE_P_QTL_mean.y) , c("File","Harmo.HWE_P_QTL_mean.x","Harmo.HWE_P_QTL_mean.y") ])
head(mm[ which(mm$Harmo.HWE_P_QTL_min.x!=mm$Harmo.HWE_P_QTL_min.y) , c("File","Harmo.HWE_P_QTL_min.x","Harmo.HWE_P_QTL_min.y") ])

head(mm[ which(mm$Clean.Lambda.x!=mm$Clean.Lambda.y) , c("File","Clean.Lambda.x","Clean.Lambda.y") ])
plot( mm$Clean.Lambda.x , mm$Clean.Lambda.y ) ; abline(0,1)

pdf("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/CompPlots.pdf")
for ( v in var )
{
 plot( mm[,paste(v,".x",sep="")] , mm[,paste(v,".y",sep="")] , xlab="gaelle" , ylab="laura" , main=v , col=ifelse(mm[,paste(v,".x",sep="")]==mm[,paste(v,".y",sep="")],yes="black",no="red") ) ; abline(0,1)
} ; rm(v)
dev.off()

mm[ grep("Vanderbilt",mm$File) , c("File","Harmo.N_lowINFO.x","Harmo.N_lowINFO.y","Harmo.N_imputedWithNoInfo.x","Harmo.N_imputedWithNoInfo.y") ]
mm[ which(is.na(mm$Clean.MinN.x)!=is.na(mm$Clean.MinN.y)) , c("File","Clean.MinN.x","Clean.MinN.y","Clean.SampleSize.x","Clean.SampleSize.y") ]
mm[ which(is.na(mm$Clean.SampleSize.x)!=is.na(mm$Clean.SampleSize.y)) , c("File","Clean.MinN.x","Clean.MinN.y","Clean.SampleSize.x","Clean.SampleSize.y") ]



tt=read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/EA_MCHIP_A_HbA1cadjFGlu_GoDARTS_IMPUTE_06102014_APM.gz",h=T,as.is=T)
qtsnp=read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/Metabochip_QT_SNPs_withMarkerName.txt",h=T,as.is=T)
table(qtsnp$MarkerName %in% tt$MarkerName)
table(tt$MarkerName %in% qtsnp$MarkerName)
median(qchisq(tt$PVAL,df=1,lower.tail=F),na.rm=T)/0.456
median(qchisq(tt$PVAL[which(tt$MarkerName %in% qtsnp$MarkerName)],df=1,lower.tail=F),na.rm=T)/0.456
median(qchisq(tt$PVAL,df=1,lower.tail=F),na.rm=T)/qchisq(0.5,df=1)
median(qchisq(tt$PVAL[which(tt$MarkerName %in% qtsnp$MarkerName)],df=1,lower.tail=F),na.rm=T)/qchisq(0.5,df=1)




###################
## Check QC2 summaries from all volunteers and report problematic studies
#################



FG1 = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/SummariesFromSFTP/Summary_YW.txt",as.is=T,h=T)
FI1 = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/SummariesFromSFTP/Summary_Johannes_Ching-Ti.txt",as.is=T,h=T)
FI2 = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/SummariesFromSFTP/SUMMARY_QC2_EW.txt",as.is=T,h=T)
HA1 = read.table(gzfile("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/SummariesFromSFTP/Summary_Katie_HbA1cadjFGlu_v3.txt.gz"),as.is=T,h=T)
HA2 = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/SummariesFromSFTP/Summary_LauraCorbin_HbA1cadjFGlu.txt",as.is=T,h=T)
HA3 = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Summary_Gaelle.txt",as.is=T,h=T)
HG1 = read.table(gzfile("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/SummariesFromSFTP/Summary_Katie_2hrGluadjBMI_final.txt.gz"),as.is=T,h=T)
HG2 = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/SummariesFromSFTP/Summary_LauraCorbin_2hrGluadjBMI.txt",as.is=T,h=T)
HU1 = read.table("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/SummariesFromSFTP/summary_PZ.txt",as.is=T,h=T)

dim(FG1)
dim(FI1)
dim(FI2)
dim(HA1)
dim(HA2)
dim(HA3)
dim(HG1)
dim(HG2)
dim(HU1)

table( colnames(FG1)==ee )
table( colnames(FI1)==ee )
table( colnames(FI2)==ee )
table( colnames(HA1)==ee )
table( colnames(HA2)==ee )
table( colnames(HA3)[1:100]==ee ) # It is mine, I aded Katie/laura in the column 101
table( colnames(HG1)==ee )
table( colnames(HG2)==ee )
table( colnames(HU1)==ee )

colnames(FI2)[which(colnames(FI2)!=ee)]
colnames(FI2)[1]=ee[1]
table( colnames(FI2)==ee )

colnames(FG1)[which(colnames(FG1)!=ee)]
colnames(FG1)[99]=ee[99]
table( colnames(FG1)==ee )

table( colnames(FI1) %in% ee )
colnames(FI1)[which(!(colnames(FI1) %in% ee))]
ee[which(!(ee %in% colnames(FI1)))]
colnames(FI1)[which(colnames(FI1)=="fileInShortName")]="File"
colnames(FI1)[which(colnames(FI1)=="Clean_N_highBETA5")]="Clean.N_highBETA5"
colnames(FI1)[which(colnames(FI1)=="Clean_N_highBETA10")]="Clean.N_highBETA10"
FI1$Harmo.N_unknownImputeWithNoInfo = NA
FI1$Clean.N_unknownImpute = NA
table( colnames(FI1) %in% ee )
FI1 = FI1[,ee]


# compare Gaelle vs Katie/Laura for HbA1cadjFGlu
HA1$File = paste(HA1$File,".gz",sep="")
HA12 = rbind(HA1,HA2)
table(HA12$File %in% HA3$File)
table(HA12$File == HA3$File)
HA12 = HA12[order(HA12$File),]
HA3 = HA3[order(HA3$File),]
table(HA12$File == HA3$File)

var=NULL
for ( v in ee[-c(1)] )
{
 tt = table(HA12[,v]==HA3[,v],exclude=c())
 ttTR = as.numeric(tt["TRUE"] ) ; if(is.na(ttTR)) {ttTR=0}
 ttFA = as.numeric(tt["FALSE"]) ; if(is.na(ttFA)) {ttFA=0}
 ttNA = as.numeric(tt[which(is.na(names(tt)))])
 if ( ttTR!=128 ) {cat( "\n",v,": " , ttFA ,"FALSE +",ttNA,"NA\n")}
 if ( ttFA>0   ) {var=c(var,v)}
 #if ( ttNA>0   ) { if(sum(is.na(HA12[,v])!=is.na(HA3[,v]),na.rm=T)>0) {var=c(var,v)}}
 rm(tt,ttTR,ttFA,ttNA)
} ; rm(v)

par(mfrow=c(2,3))
for ( v in var[1] )
{
 plot( HA12[,v] , HA3[,v] , xlab="Katie/Laura" , ylab="Gaelle" , col=ifelse(HA3$CoVolunteer=="Laura",yes="green3",no="blue") , main=v ) ; abline(0,1,col="grey")
} ; rm(v)



for ( v in var )
{
 cat(v,"\n")
 tt = cbind( HA12[which( HA12[,v] != HA3[,v]),1] ,
             HA12[which( HA12[,v] != HA3[,v]),v] ,
             HA3[which( HA12[,v] != HA3[,v]),v])
 print(tt)
 cat("\n\n")
 rm(tt)
} ; rm(v)



# compare Katie vs Laura for 2hrGluadjBMI
HG1$File = paste(HG1$File,".gz",sep="")
table(HG1$File %in% HG2$File)
table(HG1$File == HG2$File)

var=NULL
for ( v in ee[-c(1)] )
{
 tt = table(HG1[,v]==HG2[,v],exclude=c())
 ttTR = as.numeric(tt["TRUE"] ) ; if(is.na(ttTR)) {ttTR=0}
 ttFA = as.numeric(tt["FALSE"]) ; if(is.na(ttFA)) {ttFA=0}
 ttNA = as.numeric(tt[which(is.na(names(tt)))])
 if ( ttTR!=76 ) {cat( "\n",v,": " , ttFA ,"FALSE +",ttNA,"NA\n")}
 if ( ttFA>0   ) {var=c(var,v)}
 #if ( ttNA>0   ) { if(sum(is.na(HA12[,v])!=is.na(HA3[,v]),na.rm=T)>0) {var=c(var,v)}}
 rm(tt,ttTR,ttFA,ttNA)
} ; rm(v)


for ( v in var[grep("N_",var)] )
{
 cat(v,"\n")
 tt = cbind( HG1[which( HG1[,v] != HG2[,v]),1] ,
             HG1[which( HG1[,v] != HG2[,v]),v] ,
             HG2[which( HG1[,v] != HG2[,v]),v])
 print(tt)
 cat("\n\n")
 rm(tt)
} ; rm(v)


# To take into account
summALL = rbind( FG1, FI1, FI2, HA1, HA2, HG2, HU1)
dim(summALL)
table(duplicated(summALL$File))
summALL$FileName = sub(".txt","",sub(".gz","",summALL$File))
table(duplicated(summALL$FileName))
summALL$FileName = sub("CM_SW","CM.SW",summALL$FileName)
summALL$FileName = sub("_modifiedEW","",summALL$FileName)

# check if all there
harmonized = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/HarmonizedFiles_FREEZE1.txt" ,as.is=T,h=T,sep="\t")
cleaned    = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/CleanedFiles20150522.txt" ,as.is=T,h=T,sep="\t")

harmonized$FileName = sub(".txt.gz","",harmonized$File)
cleaned$FileName = sub(".txt","",sub(".gz","",cleaned$File))
cleaned$FileName = sub("CM_SW","CM.SW",cleaned$FileName)

table( cleaned$FileName %in% harmonized$FileName )
table( harmonized$FileName %in% cleaned$FileName )
table( harmonized$FileName %in% cleaned$FileName , harmonized$Trait )

table( summALL$FileName %in% harmonized$FileName )
table( summALL$FileName %in% cleaned$FileName , grepl("FInsadjBMI",summALL$FileName) )


dd1 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download",full.names=T)
dd2 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned",full.names=T) ; dd2 = dd2[grep("gz",dd2)]
#dd3 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/CLEANED",full.names=T)
dd4 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/FILES/CLEANED",full.names=T) ; dd4 = dd4[grep("gz",dd4)]
ff0 = c(dd1,dd2,dd4) ; rm(dd1,dd2,dd4)
ff0_file = sapply( ff0 , FUN=function(x){ x1 = strsplit(x,"/",fixed=T)[[1]] ; return(x1[length(x1)]) } )
ff0_fileName = sub(".txt","",sub(".gz","",ff0_file))
ff0_fileName = sub("CM_SW","CM.SW",ff0_fileName)
ff0_fileName = sub("_modifiedEW","",ff0_fileName)
ff1 = sapply( ff0 , FUN=function(x){
                                     x1 = strsplit(x,"/",fixed=T)[[1]]
                                     x2 = strsplit( x1[length(x1)] ,"_",fixed=T)[[1]]
                                     a1 = x2[1]
                                     if(a1=="EA") {a2="EUR"}
                                     if(a1=="AA") {a2="AFR"}
                                     if(a1=="EAA") {a2="ASN"}
                                     if(a1=="IAA") {a2="SAS"}
                                     if(a1=="HA") {a2="AMR"}
                                     t1 = x2[4]
                                     if( t1=="FGluadjBMI" ) { t2="FG" }
                                     if( t1=="FInsadjBMI" ) { t2="FI" }
                                     if( t1=="2hrGluadjBMI" ) { t2="2hGlu" }
                                     if( t1 %in% c("HbA1c","HbA1cadjFGlu") ) { t2="HbA1c" }
                                     c1 = x2[3]
                                     if( c1 %in% c("A","AX"))  { c2=1 } else { c2=0 }
                                     return( c( CHIP=x2[2] , ANCESTRY=a2 , TRAIT=t2 , AUTOSOME_IN=c2 ) )
                                   } )

table(ff1["CHIP",])
table(ff1["ANCESTRY",])
table(ff1["TRAIT",])
table(ff1["AUTOSOME_IN",])

table(ff1["TRAIT",],ff1["ANCESTRY",],ff1["CHIP",],ff1["AUTOSOME_IN",])


table( summALL$FileName %in% ff0_fileName )
table( ff0_fileName %in% summALL$FileName )

# work on all summaries
tp = sapply( summALL$FileName , FUN=function(x){ x1=strsplit(x,"_")[[1]]
                                                        if( x1[5]=="InvNorm" ) {tt="InvNorm";ss=x1[6]} else {tt="Raw";ss=x1[5]}
                                                        return(c(Ancestry=x1[1],Chip=x1[2],Chr=x1[3],Trait=x1[4],Transf=tt,Study=ss))
                                                      } )

summALL$Ancestry = tp["Ancestry",]
summALL$Chip     = tp["Chip",]
summALL$Chr      = tp["Chr",]
summALL$Trait    = tp["Trait",]
summALL$Transf   = tp["Transf",]
summALL$Study    = tp["Study",]
rm(tp)

table(summALL$Ancestry)
table(summALL$Chip)
table(summALL$Chr)
table(summALL$Trait)
table(summALL$Transf)




# 001 - Inconsistencies between file name chromosome and actual chromosomes in the file
table( summALL$Chr %in% c("A","AX") , summALL$Harmo.N_A>0 )
table( summALL$Chr %in% c("A") , summALL$Harmo.N_X==0 & summALL$Harmo.N_XY==0 )
table( summALL$Chr %in% c("AX","X","XF","XM") , summALL$Harmo.N_X>0 | summALL$Harmo.N_XY>0 ) # 4 files / 2 studies
table( summALL$Chr %in% c("X","XF","XM") , summALL$Harmo.N_A==0 )

summALL[which( summALL$Chr!="A" & summALL$Harmo.N_X==0 & summALL$Harmo.N_XY==0 ),c("FileName")]


# 002 -No genotyped variants
table( summALL$Harmo.N_genotyped==0 )
summALL[which( summALL$Harmo.N_genotyped==0 ), c("FileName","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute") ]
dim(unique(summALL[which( summALL$Harmo.N_genotyped==0 ), c("Study","Trait","Chip","Ancestry","Chr","Transf") ]))
dim(unique(summALL[which( summALL$Harmo.N_genotyped==0 ), c("Study","Trait","Ancestry") ]))
tt1 = unique(summALL[which( summALL$Harmo.N_genotyped==0 ), c("Study","Trait","Ancestry") ]) ; tt1[order(tt1$Study,tt1$Ancestry,tt1$Trait),] ; rm(tt1)

saN = unique(paste( summALL$Study,summALL$Ancestry,sep=":" )[which( summALL$Harmo.N_genotyped==0 )])
saP = unique(paste( summALL$Study,summALL$Ancestry,sep=":" )[which( summALL$Harmo.N_genotyped>0 )])
table( saN %in% saP )
saN[which(saN %in% saP)]

tt1 = summALL[which( paste(summALL$Study,summALL$Ancestry,sep=":") %in% saN ), c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute") ] ; tt1[order(tt1$Study,tt1$Ancestry,tt1$Trait),] ; rm(tt1)



# 003 - Unknown imputed variants
table( summALL$Harmo.N_unknownImpute>0 , summALL$Harmo.N_genotyped==0  )

summALL[which( summALL$Harmo.N_unknownImpute>0 & summALL$Harmo.N_genotyped>0 ), c("FileName","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_A","Harmo.N_X","Harmo.N_XY") ]


# 004 - check number/proportion of dropped variants
table( summALL$Harmo.N_SNPs+summALL$Harmo.N_indels - summALL$Harmo.N_A-summALL$Harmo.N_X-summALL$Harmo.N_XY ==0 )
table( summALL$Harmo.N_SNPs+summALL$Harmo.N_indels - summALL$Harmo.N_imputed-summALL$Harmo.N_genotyped-summALL$Harmo.N_unknownImpute ==0 )

table( summALL$Clean.N_SNPs+summALL$Clean.N_indels - summALL$Clean.N_A-summALL$Clean.N_X-summALL$Clean.N_XY ==0 )
table( summALL$Clean.N_SNPs+summALL$Clean.N_indels - summALL$Clean.N_imputed-summALL$Clean.N_genotyped-summALL$Clean.N_unknownImpute ==0 )

ww=which(is.na(summALL$Clean.N_unknownImpute))
summALL$Clean.N_unknownImpute[ww] = (summALL$Clean.N_SNPs+summALL$Clean.N_indels - summALL$Clean.N_imputed-summALL$Clean.N_genotyped)[ww]
rm(ww)
table( summALL$Clean.N_SNPs+summALL$Clean.N_indels - summALL$Clean.N_imputed-summALL$Clean.N_genotyped-summALL$Clean.N_unknownImpute ==0 )


summALL$N_DROPPED = summALL$Harmo.N_SNPs+summALL$Harmo.N_indels - summALL$Clean.N_SNPs-summALL$Clean.N_indels
summALL$P_DROPPED = summALL$N_DROPPED/(summALL$Harmo.N_SNPs+summALL$Harmo.N_indels)

plot( summALL$N_DROPPED , summALL$P_DROPPED )

plot( summALL$Harmo.N_SNPs+summALL$Harmo.N_indels , summALL$Clean.N_SNPs+summALL$Clean.N_indels )


summALL[which(summALL$Study=="CHS" & summALL$Ancestry=="AA"),c("Ancestry","Chip","Chr","Trait","Transf","Study","Harmo.N_SNPs","Harmo.N_indels","Clean.N_SNPs","Clean.N_indels","N_DROPPED","P_DROPPED",
                                      "Harmo.N_lowMAC","Harmo.N_highSE","Harmo.N_lowINFO","Harmo.N_missingBETAorSE","Harmo.N_genotypedLowHWE","Harmo.N_genotypedLowCALLRATE",
                                      "Harmo.N_imputedWithNoInfo","Harmo.N_unknownImputeWithNoInfo")]


EASYrep_CENT[which(EASYrep_CENT$Study=="CHS" & EASYrep_CENT$Ancestry=="AA"),c("Ancestry","Chip","Chr","Trait","Transf","Study","numSNPsIn","Clean.N_SNPs","Clean.N_indels","AFCHECK.numOutlier")]



summALL[which(summALL$Study=="CFS" & summALL$Ancestry=="EA"),c("Ancestry","Chip","Chr","Trait","Transf","Study","Harmo.N_SNPs","Harmo.N_indels","Clean.N_SNPs","Clean.N_indels","N_DROPPED","P_DROPPED",
                                      "Harmo.N_lowMAC","Harmo.N_highSE","Harmo.N_lowINFO","Harmo.N_missingBETAorSE","Harmo.N_genotypedLowHWE","Harmo.N_genotypedLowCALLRATE",
                                      "Harmo.N_imputedWithNoInfo","Harmo.N_unknownImputeWithNoInfo")]


EASYrep_CENT[which(EASYrep_CENT$Study=="CFS" & EASYrep_CENT$Ancestry=="EA"),c("Ancestry","Chip","Chr","Trait","Transf","Study","numSNPsIn","Clean.N_SNPs","Clean.N_indels","AFCHECK.numOutlier")]



# 1- imputed variants with no info
table( summALL$Harmo.N_imputedWithNoInfo>0 )

tt1 = summALL[ which(summALL$Harmo.N_imputedWithNoInfo>0) , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_imputedWithNoInfo","Harmo.INFO_NA") ] ; tt1[order(tt1$Study,tt1$Ancestry,tt1$Trait),] ; rm(tt1)



# 2 - Unknown IMPUTED status with missing INFO
table( summALL$Harmo.N_unknownImputeWithNoInfo>0 )
summALL[which( summALL$Harmo.N_unknownImputeWithNoInfo>0 ), c("FileName","Ancestry","Chip","Chr","Transf","Study","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_A","Harmo.N_X","Harmo.N_XY","Harmo.N_imputedWithNoInfo","Harmo.N_unknownImputeWithNoInfo","Harmo.INFO_NA") ]
  ## this is PROCARDIS


# 3 - Genotyped variants are a mix of available and missing HWE_P_QTL
table( summALL$Harmo.N_genotypedWithNoHWE>0 & summALL$Harmo.N_genotypedWithNoHWE<summALL$Harmo.N_genotyped )

tt1=summALL[ which(summALL$Harmo.N_genotypedWithNoHWE>0 & summALL$Harmo.N_genotypedWithNoHWE<summALL$Harmo.N_genotyped)  , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedWithNoHWE","Harmo.HWE_P_QTL_NA","Harmo.N_genotypedLowHWE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)

table( summALL$Harmo.N_genotypedWithNoHWE==summALL$Harmo.N_genotyped )
unique(summALL[ which(summALL$Harmo.N_genotypedWithNoHWE==summALL$Harmo.N_genotyped)  , "Study" ])


# 4 - Genotyped variants are a mix of available and missing CALL_RATE_QTL
table( summALL$Harmo.N_genotypedWithNoCALLRATE>0 & summALL$Harmo.N_genotypedWithNoCALLRATE<summALL$Harmo.N_genotyped )

tt1=summALL[ which(summALL$Harmo.N_genotypedWithNoCALLRATE>0 & summALL$Harmo.N_genotypedWithNoCALLRATE<summALL$Harmo.N_genotyped)  , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedWithNoCALLRATE","Harmo.CALL_RATE_QTL_NA","Harmo.N_genotypedLowCALLRATE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)

table( summALL$Harmo.N_genotypedWithNoCALLRATE==summALL$Harmo.N_genotyped )
unique(summALL[ which(summALL$Harmo.N_genotypedWithNoCALLRATE==summALL$Harmo.N_genotyped)  , "Study" ])


# 5 - Genotyped variants with low HWE_P_QTL (<10-6)
table( summALL$Harmo.N_genotypedLowHWE>0 )

tt1=summALL[ which(summALL$Harmo.N_genotypedLowHWE>0)  , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedWithNoHWE","Harmo.N_genotypedLowHWE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)


# 6 - Genotyped variants with low CALL RATE (<95%)
table( summALL$Harmo.N_genotypedLowCALLRATE>0 )

tt1=summALL[ which(summALL$Harmo.N_genotypedLowCALLRATE>0)  , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_genotypedLowCALLRATE","Harmo.N_genotypedWithNoCALLRATE") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)

# 7 - Missing SE or BETA
table( summALL$Harmo.SE_NA>0 )
table( summALL$Harmo.BETA_NA>0 )

table( summALL$Harmo.SE_NA>0 | summALL$Harmo.BETA_NA>0 )


tt1=summALL[ which(summALL$Harmo.SE_NA>0 | summALL$Harmo.BETA_NA>0)  , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.BETA_NA","Harmo.SE_NA") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)


# 8 - 9 - Missing MAC (ie missing EAF_QTL or N_QTL)
table( summALL$Harmo.MAC_NA>0 )


table( summALL$Harmo.N_QTL_NA>0 , summALL$Harmo.EAF_QTL_NA>0 )
table( summALL$Harmo.N_QTL_NA>0 , summALL$Clean.N_QTL_NA>0 )
table( summALL$Harmo.EAF_QTL_NA>0 , summALL$Clean.EAF_QTL_NA>0 )

tt1=summALL[ which(summALL$Harmo.N_QTL_NA>0 | summALL$Harmo.EAF_QTL_NA>0)  , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.N_A","Harmo.N_X","Harmo.N_XY","Harmo.N_QTL_NA","Clean.N_QTL_NA","Harmo.EAF_QTL_NA","Clean.EAF_QTL_NA") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)



# Missing or unexpected PVAL
table( summALL$Harmo.PVAL_NA>0 )
table( summALL$Harmo.PVAL_min<0 )
table( summALL$Harmo.PVAL_max>1 )

tt1=summALL[ which( summALL$Harmo.PVAL_min<0 | summALL$Harmo.PVAL_max>1)  , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.PVAL_NA","Harmo.PVAL_min","Harmo.PVAL_max") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)

table( summALL$Harmo.SE_min<0 )

tt1=summALL[ which(summALL$Harmo.SE_min<0 )  , c("FileName","Study","Trait","Ancestry","Harmo.N_imputed","Harmo.N_genotyped","Harmo.N_unknownImpute","Harmo.PVAL_NA","Harmo.PVAL_min","Harmo.PVAL_max","Harmo.SE_min") ] ; tt1[order(tt1$Study,tt1$Ancestry),] ; rm(tt1)



# variants with low sample size
table( summALL$Clean.MinN<=30 )





###################
## RUN EasyQC for plots across ALL cleaned files
#################

## Allele Frequency from 1000G downloaded from EasyQC website to /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles
wget http://homepages.uni-regensburg.de/~wit59712/easyqc/1000g/allelefreq.1000G_EUR_p1v3.impute_legends.noMono.noDup.noX.v2.gz
wget http://homepages.uni-regensburg.de/~wit59712/easyqc/1000g/allelefreq.1000G_AFR_p1v3.impute_legends.noMono.noDup.noX.v2.gz
wget http://homepages.uni-regensburg.de/~wit59712/easyqc/1000g/allelefreq.1000G_AMR_p1v3.impute_legends.noMono.noDup.noX.v2.gz
wget http://homepages.uni-regensburg.de/~wit59712/easyqc/1000g/allelefreq.1000G_ASN_p1v3.impute_legends.noMono.noDup.noX.v2.gz

bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_AFR.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_AFR.err -R"select[mem>100] rusage[mem=100]" -M100 "zcat ReferenceFiles/allelefreq.1000G_AFR_p1v3.impute_legends.noMono.noDup.noX.v2.gz | awk '{ split(\$1,a,\":\") ; if(length(a)==1) {print \"MarkerName\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==2) {print a[1]\":\"a[2]\":SNP\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==3) {print a[1]\":\"a[2]\":INDEL\t\"\$2\"\t\"\$3\"\t\"\$4} }' | bgzip -c > ReferenceFiles/allelefreq.1000G_AFR_p1v3.impute_legends.noMono.noDup.noX.v2.MarkerName.gz"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_AMR.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_AMR.err -R"select[mem>100] rusage[mem=100]" -M100 "zcat ReferenceFiles/allelefreq.1000G_AMR_p1v3.impute_legends.noMono.noDup.noX.v2.gz | awk '{ split(\$1,a,\":\") ; if(length(a)==1) {print \"MarkerName\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==2) {print a[1]\":\"a[2]\":SNP\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==3) {print a[1]\":\"a[2]\":INDEL\t\"\$2\"\t\"\$3\"\t\"\$4} }' | bgzip -c > ReferenceFiles/allelefreq.1000G_AMR_p1v3.impute_legends.noMono.noDup.noX.v2.MarkerName.gz"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_ASN.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_ASN.err -R"select[mem>100] rusage[mem=100]" -M100 "zcat ReferenceFiles/allelefreq.1000G_ASN_p1v3.impute_legends.noMono.noDup.noX.v2.gz | awk '{ split(\$1,a,\":\") ; if(length(a)==1) {print \"MarkerName\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==2) {print a[1]\":\"a[2]\":SNP\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==3) {print a[1]\":\"a[2]\":INDEL\t\"\$2\"\t\"\$3\"\t\"\$4} }' | bgzip -c > ReferenceFiles/allelefreq.1000G_ASN_p1v3.impute_legends.noMono.noDup.noX.v2.MarkerName.gz"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_EUR.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_EUR.err -R"select[mem>100] rusage[mem=100]" -M100 "zcat ReferenceFiles/allelefreq.1000G_EUR_p1v3.impute_legends.noMono.noDup.noX.v2.gz | awk '{ split(\$1,a,\":\") ; if(length(a)==1) {print \"MarkerName\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==2) {print a[1]\":\"a[2]\":SNP\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==3) {print a[1]\":\"a[2]\":INDEL\t\"\$2\"\t\"\$3\"\t\"\$4} }' | bgzip -c > ReferenceFiles/allelefreq.1000G_EUR_p1v3.impute_legends.noMono.noDup.noX.v2.MarkerName.gz"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_SAS.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_SAS.err -R"select[mem>100] rusage[mem=100]" -M100 "zcat ReferenceFiles/1000GP_p3v5_legends_rbind.noDup.MAFgt01.noCnv.noCnAll.noX.TW.afref.SAS.txt.gz | awk '{ split(\$1,a,\":\") ; if(length(a)==1) {print \"MarkerName\ta0\ta1\teaf\"} ; if(length(a)==2) {print a[1]\":\"a[2]\":SNP\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==3) {print a[1]\":\"a[2]\":INDEL\t\"\$2\"\t\"\$3\"\t\"\$4} }' | bgzip -c > ReferenceFiles/1000GP_p3v5_legends_rbind.noDup.MAFgt01.noCnv.noCnAll.noX.TW.afref.SAS.MarkerName.gz"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_MXL.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/bsub_MXL.err -R"select[mem>100] rusage[mem=100]" -M100 "zcat ReferenceFiles/1000G_p1v3.MXL.FiltPass.NoDup.MAFgt01.NoX.AF_rbind.txt.gz | awk '{ split(\$1,a,\":\") ; if(length(a)==1) {print \"MarkerName\ta0\ta1\teaf\"} ; if(length(a)==2) {print a[1]\":\"a[2]\":SNP\t\"\$2\"\t\"\$3\"\t\"\$4} ; if(length(a)==3) {print a[1]\":\"a[2]\":INDEL\t\"\$2\"\t\"\$3\"\t\"\$4} }' | bgzip -c > ReferenceFiles/1000G_p1v3.MXL.FiltPass.NoDup.MAFgt01.NoX.AF_rbind.MarkerName.gz"

grep Success ReferenceFiles/bsub_*

## Known loci for each trait circulated by Sara for Exome-Chip QC --> make the names matching our MarkerName (chr:pos:SNP)
fg = read.table("/nfs/users/nfs_g/gm10/MAGICexomeQC_NOV2014/MAGIC_QC_known_SNPs_FG.txt",h=T,as.is=T,sep="\t")
fi = read.table("/nfs/users/nfs_g/gm10/MAGICexomeQC_NOV2014/MAGIC_QC_known_SNPs_FI.txt",h=T,as.is=T,sep="\t")
h2 = read.table("/nfs/users/nfs_g/gm10/MAGICexomeQC_NOV2014/MAGIC_QC_known_SNPs_2hGlu.txt",h=T,as.is=T,sep="\t")
hb = read.table("/nfs/users/nfs_g/gm10/MAGICexomeQC_NOV2014/MAGIC_QC_known_SNPs_HbA1c.txt",h=T,as.is=T,sep="\t")
hb = hb[which(hb$chrpos!=""),]

fg$MarkerName = paste( fg$chrpos , "SNP" , sep=":" )
fi$MarkerName = paste( fi$chrpos , "SNP" , sep=":" )
h2$MarkerName = paste( h2$chrpos , "SNP" , sep=":" )
hb$MarkerName = paste( hb$chrpos , "SNP" , sep=":" )

write.table( fg[,c("MarkerName","raising_allele","other_allele","raising_allele_freq","beta")] , "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_FG_for1000Gimputation.txt" , quote=F, row.names=F , sep="\t" )
write.table( fi[,c("MarkerName","raising_allele","other_allele","raising_allele_freq","beta")] , "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_FI_for1000Gimputation.txt" , quote=F, row.names=F , sep="\t" )
write.table( h2[,c("MarkerName","raising_allele","other_allele","raising_allele_freq","beta")] , "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_2hGlu_for1000Gimputation.txt" , quote=F, row.names=F , sep="\t" )
write.table( hb[,c("MarkerName","raising_allele","other_allele","raising_allele_freq","beta")] , "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_HbA1c_for1000Gimputation.txt" , quote=F, row.names=F , sep="\t" )


## Download cleaned files from sftp
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/FGluadjBMI/Raw/CLEANED*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/FGluadjBMI/InvNorm/CLEANED*   /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/FInsadjBMI/Raw/*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/FInsadjBMI/InvNorm/*   /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/2hrGluadjBMI/Raw/*     /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/2hrGluadjBMI/InvNorm/* /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/HbA1c/Raw/*            /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/HbA1c/InvNorm/*        /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
#wait Katie -- Take mine# get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/HbA1cadjFGlu/Raw/*     /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
#wait Katie -- Take mine# get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/HbA1cadjFGlu/InvNorm/* /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/

get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/FGluadjBMI/Raw/CLEANED*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/FGluadjBMI/InvNorm/CLEANED*   /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/FInsadjBMI/Raw/*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/FInsadjBMI/InvNorm/*   /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/2hrGluadjBMI/Raw/*     /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/2hrGluadjBMI/InvNorm/* /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/HbA1c/Raw/*            /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/HbA1c/InvNorm/*        /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
#wait Katie -- Take mine# get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/HbA1cadjFGlu/Raw/*     /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
#wait Katie -- Take mine# get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/HbA1cadjFGlu/InvNorm/* /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/

# AASC files I get wrong, take those from Katie fron the sftp
get GWAS-1000G/HbA1cadjFGlu/Raw/EAA_GWAS_AX_HbA1cadjFGlu_AASC_MINIMACH_25102014_YT.txt.gz             /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
get GWAS-1000G/HbA1cadjFGlu/InvNorm/EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_AASC_MINIMACH_25102014_YT.txt.gz /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/


get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/HbA1cadjFGlu/Raw/*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download2/
get /cleaned_gluinsrelatedtraits/Cleaned/GWAS-1000G/HbA1cadjFGlu/InvNorm/*   /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download2/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/HbA1cadjFGlu/Raw/*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download2/
get /cleaned_gluinsrelatedtraits/Cleaned/Mchip-1000G/HbA1cadjFGlu/InvNorm/*   /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download2/

get /cleaned_gluinsrelatedtraits/FREEZE1_Cleaned_Ready/GWAS-1000G/HbA1cadjFGlu/Raw/*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download2/
get /cleaned_gluinsrelatedtraits/FREEZE1_Cleaned_Ready/GWAS-1000G/HbA1cadjFGlu/InvNorm/*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download2/
get /cleaned_gluinsrelatedtraits/FREEZE1_Cleaned_Ready/Mchip-1000G/HbA1cadjFGlu/Raw/*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download2/
get /cleaned_gluinsrelatedtraits/FREEZE1_Cleaned_Ready/Mchip-1000G/HbA1cadjFGlu/InvNorm/*       /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download2/



# try a MCHIP file for lambda and QQ plot
dd = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/EA_MCHIP_A_HbA1cadjFGlu_Health2006_SNPTEST_16122014_NG.txt.gz",h=T,as.is=T,na.strings=".")
rr = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/Metabochip_QT_SNPs_withMarkerName.txt",h=T,as.is=T)
dd$QTSNP = ifelse(dd$MarkerName  %in% rr$MarkerName , yes=1 , no=0 )
table( dd$QTSNP , dd$EAF_QTL>0.01 & dd$EAF_QTL<0.99 )
median(qchisq(dd$PVAL[which(dd$QTSNP==1 & dd$EAF_QTL>0.01 & dd$EAF_QTL<0.99 )],df=1,lower.tail=F),na.rm=T)/qchisq(0.5,df=1) # ---> OK with EasyQC  report

qqO=qchisq(dd$PVAL[which(dd$QTSNP==1 & dd$EAF_QTL>0.01 & dd$EAF_QTL<0.99 )],df=1,lower.tail=F)
qqO=qchisq(dd$PVAL[which(dd$QTSNP==1 )],df=1,lower.tail=F)
qqO=qqO[order(qqO)]
qqE=qchisq(seq(from=0,to=1,length.out=(length(qqO)+1))[-c(1)],df=1,lower.tail=F)
qqE=qqE[order(qqE)]

plot( qqE , qqO ) ; abline(0,1,col="grey")



####### /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_test.ecf
### Set parameters and input files
DEFINE --strMissing . --strSeparator TAB --pathOut /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results --acolIn MarkerName;EFFECT_ALLELE;NON_EFFECT_ALLELE;N_QTL;EAF_QTL;BETA;SE;PVAL;IMPUTED;INFO --acolInClasses character;character;character;numeric;numeric;numeric;numeric;numeric;numeric;numeric
### Start EasyQC
START EASYQC
# add/edit some columns
ADDCOL  --rcdAddCol  pmin(EAF_QTL*N_QTL*2,(1-EAF_QTL)*N_QTL*2,na.rm=TRUE)                      --colOut MAC
ADDCOL  --rcdAddCol  sapply( MarkerName , FUN=function(x){ return(strsplit(x,":")[[1]][1]) } ) --colOut CHR2
ADDCOL  --rcdAddCol  ifelse( EAF_QTL>0.01 & EAF_QTL<0.99 , yes=PVAL , no=NA ) --colOut PVALCOMM
# general counts
GETNUM --rcdGetNum (grepl("SNP",MarkerName))   --strGetNumName Clean.N_SNPs
GETNUM --rcdGetNum (grepl("INDEL",MarkerName) | grepl("MERGED_DEL",MarkerName)) --strGetNumName Clean.N_indels
GETNUM --rcdGetNum (CHR2 %in% 1:22)            --strGetNumName Clean.N_A
GETNUM --rcdGetNum (CHR2=="X")                 --strGetNumName Clean.N_X
GETNUM --rcdGetNum (CHR2=="XY")                --strGetNumName Clean.N_XY
GETNUM --rcdGetNum (IMPUTED==1)                --strGetNumName Clean.N_imputed
GETNUM --rcdGetNum (IMPUTED==0)                --strGetNumName Clean.N_genotyped
GETNUM --rcdGetNum (is.na(IMPUTED))            --strGetNumName Clean.N_unknownImpute
# Check of cleaning criteria
GETNUM --rcdGetNum (MAC<3)                           --strGetNumName Clean.N_lowMAC
GETNUM --rcdGetNum (SE>=10)                          --strGetNumName Clean.N_highSE
GETNUM --rcdGetNum (INFO<0.4)                        --strGetNumName Clean.N_lowINFO
GETNUM --rcdGetNum (is.na(BETA) | is.na(SE))         --strGetNumName Clean.N_missingBETAorSE
GETNUM --rcdGetNum (IMPUTED==1 & is.na(INFO))        --strGetNumName Clean.N_imputedWithNoInfo
GETNUM --rcdGetNum (is.na(IMPUTED) & is.na(INFO))    --strGetNumName Clean.N_unknownImputeWithNoInfo
# counts for further checks
GETNUM --rcdGetNum (EAF_QTL==0 | EAF_QTL==1)           --strGetNumName Clean.N_mono
GETNUM --rcdGetNum (is.na(N_QTL))                      --strGetNumName Clean.N_QTL_NA
# counts for further checks of mine not in the guidelines
GETNUM --rcdGetNum (is.na(EAF_QTL))      --strGetNumName Clean.EAF_QTL_NA_getnum
# summary on cleaned files
GETNUM --rcdGetNum (abs(BETA)>xx)   --strGetNumName Clean.N_highBETA
# More
CALCULATE --rcdCalc max(N_QTL,na.rm=T)   --strCalcName Clean.SampleSize
CALCULATE --rcdCalc median(SE)           --strCalcName Clean.MedianSE
GETNUM --rcdGetNum ((PVAL)>1)            --strGetNumName Clean.N_PVAL1
# stats
EVALSTAT --colStat MAC           --strTag Clean
EVALSTAT --colStat INFO          --strTag Clean
EVALSTAT --colStat BETA          --strTag Clean
EVALSTAT --colStat SE            --strTag Clean
EVALSTAT --colStat PVAL          --strTag Clean
EVALSTAT --colStat EAF_QTL       --strTag Clean
# Lambda
GC --colPval PVALCOMM --blnSuppressCorrection 1
#!! CAUTION for Lambda: for MCHIP  --fileGcSnps /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/Metabochip_QT_SNPs_withMarkerName.txt --colGcSnpsMarker MarkerName --colInMarker MarkerName
# stat and outliers for boxplots of BETA by MAF bins
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL<0.01 | EAF_QTL>0.99)],plot=F)$stat[1,1] --strCalcName Clean.BETA.RARE.stat1
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL<0.01 | EAF_QTL>0.99)],plot=F)$stat[2,1] --strCalcName Clean.BETA.RARE.stat2
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL<0.01 | EAF_QTL>0.99)],plot=F)$stat[3,1] --strCalcName Clean.BETA.RARE.stat3
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL<0.01 | EAF_QTL>0.99)],plot=F)$stat[4,1] --strCalcName Clean.BETA.RARE.stat4
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL<0.01 | EAF_QTL>0.99)],plot=F)$stat[5,1] --strCalcName Clean.BETA.RARE.stat5
CALCULATE --rcdCalc min(BETA[which(EAF_QTL<0.01 | EAF_QTL>0.99)])                      --strCalcName Clean.BETA.RARE.min
CALCULATE --rcdCalc max(BETA[which(EAF_QTL<0.01 | EAF_QTL>0.99)])                      --strCalcName Clean.BETA.RARE.max
CALCULATE --rcdCalc boxplot(BETA[which((EAF_QTL>=0.01 & EAF_QTL<0.05) | (EAF_QTL<=0.99 & EAF_QTL>0.95))],plot=F)$stat[1,1] --strCalcName Clean.BETA.LF.stat1
CALCULATE --rcdCalc boxplot(BETA[which((EAF_QTL>=0.01 & EAF_QTL<0.05) | (EAF_QTL<=0.99 & EAF_QTL>0.95))],plot=F)$stat[2,1] --strCalcName Clean.BETA.LF.stat2
CALCULATE --rcdCalc boxplot(BETA[which((EAF_QTL>=0.01 & EAF_QTL<0.05) | (EAF_QTL<=0.99 & EAF_QTL>0.95))],plot=F)$stat[3,1] --strCalcName Clean.BETA.LF.stat3
CALCULATE --rcdCalc boxplot(BETA[which((EAF_QTL>=0.01 & EAF_QTL<0.05) | (EAF_QTL<=0.99 & EAF_QTL>0.95))],plot=F)$stat[4,1] --strCalcName Clean.BETA.LF.stat4
CALCULATE --rcdCalc boxplot(BETA[which((EAF_QTL>=0.01 & EAF_QTL<0.05) | (EAF_QTL<=0.99 & EAF_QTL>0.95))],plot=F)$stat[5,1] --strCalcName Clean.BETA.LF.stat5
CALCULATE --rcdCalc min(BETA[which((EAF_QTL>=0.01 & EAF_QTL<0.05) | (EAF_QTL<=0.99 & EAF_QTL>0.95))])                      --strCalcName Clean.BETA.LF.min
CALCULATE --rcdCalc max(BETA[which((EAF_QTL>=0.01 & EAF_QTL<0.05) | (EAF_QTL<=0.99 & EAF_QTL>0.95))])                      --strCalcName Clean.BETA.LF.max
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL>=0.05 & EAF_QTL<=0.95)],plot=F)$stat[1,1] --strCalcName Clean.BETA.COMM.stat1
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL>=0.05 & EAF_QTL<=0.95)],plot=F)$stat[2,1] --strCalcName Clean.BETA.COMM.stat2
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL>=0.05 & EAF_QTL<=0.95)],plot=F)$stat[3,1] --strCalcName Clean.BETA.COMM.stat3
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL>=0.05 & EAF_QTL<=0.95)],plot=F)$stat[4,1] --strCalcName Clean.BETA.COMM.stat4
CALCULATE --rcdCalc boxplot(BETA[which(EAF_QTL>=0.05 & EAF_QTL<=0.95)],plot=F)$stat[5,1] --strCalcName Clean.BETA.COMM.stat5
CALCULATE --rcdCalc min(BETA[which(EAF_QTL>=0.05 & EAF_QTL<=0.95)])                      --strCalcName Clean.BETA.COMM.min
CALCULATE --rcdCalc max(BETA[which(EAF_QTL>=0.05 & EAF_QTL<=0.95)])                      --strCalcName Clean.BETA.COMM.max
# Plots
PZPLOT --colBeta BETA  --colSe SE  --colPval PVAL
EXTRACTSNPS --colInMarker MarkerName
            --fileRef
            --colRefMarker MarkerName
            --strTag KnownSnps
#MERGE --fileRef /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/Metabochip_QT_SNPs_withMarkerName.txt --colRefMarker MarkerName --colInMarker MarkerName --acolIn trait;MarkerName --acolInClasses character;character
QQPLOT --acolQQPlot PVALCOMM
AFCHECK --colInMarker MarkerName
        --colInA1 EFFECT_ALLELE
        --colInA2 NON_EFFECT_ALLELE
        --colInFreq EAF_QTL
        --fileRef
        --colRefMarker MarkerName
        --colRefA1 a0
        --colRefA2 a1
        --colRefFreq eaf
        --blnPlotAll 1
### Stop EasyQC
STOP EASYQC
####################################################################################
#        --fileRef /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/allelefreq.1000G_EUR_p1v3.impute_legends.noMono.noDup.noX.v2.MarkerName.gz
#            --fileRef /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_FG_for1000Gimputation.txt



####### /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/c_template.ecf
### Set parameters and input files
DEFINE --strMissing . --strSeparator TAB --pathOut /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_c_results --acolIn MarkerName;EFFECT_ALLELE;NON_EFFECT_ALLELE;N_QTL;EAF_QTL;BETA;SE;PVAL;IMPUTED;INFO --acolInClasses character;character;character;numeric;numeric;numeric;numeric;numeric;numeric;numeric
### Start EasyQC
START EASYQC
# add/edit some columns
ADDCOL  --rcdAddCol  pmin(EAF_QTL,(1-EAF_QTL),na.rm=TRUE) --colOut MAF
# More
CALCULATE --rcdCalc max(N_QTL,na.rm=T)            --strCalcName Clean.SampleSize
CALCULATE --rcdCalc median(SE)                    --strCalcName Clean.MedianSE
CALCULATE --rcdCalc median(1/sqrt(2*MAF*(1-MAF))) --strCalcName Clean.c
### Stop EasyQC
STOP EASYQC
####################################################################################




## prepare EasyQc scripts
ls -lh /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download/
ls -lh /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned/*.gz
ls -lh /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/CLEANED/


tt = system("cat /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_Template.ecf",intern=T)
  ## WARNING! change the template!
  ##   1 - AFCHECK at the end
  ##   2 - Lambda & QQ on PVALCOMM
  ##   3 - highBETA needs to depend on the trait
ttc = system("cat /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/c_template.ecf",intern=T)

dd1 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download",full.names=T)
dd2 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned",full.names=T) ; dd2 = dd2[grep("gz",dd2)]
#dd3 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/CLEANED",full.names=T)
dd4 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/FILES/CLEANED/Raw",full.names=T) ; dd4 = dd4[grep("gz",dd4)]
dd5 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/FILES/CLEANED/InvNorm",full.names=T) ; dd4 = dd4[grep("gz",dd4)]
ff0 = c(dd1,dd2,dd4,dd5) ; rm(dd1,dd2,dd4,dd5)
ff0_file = sapply( ff0 , FUN=function(x){ x1 = strsplit(x,"/",fixed=T)[[1]] ; return(x1[length(x1)]) } )
ff0_file = sub("CLEANED.","",ff0_file)
ff1 = sapply( ff0_file , FUN=function(x){
                                     x2 = strsplit( x ,"_",fixed=T)[[1]]
                                     a1 = x2[1]
                                     if(a1=="EA") {a2="EUR"}
                                     if(a1=="AA") {a2="AFR"}
                                     if(a1=="EAA") {a2="ASN"}
                                     if(a1=="IAA") {a2="SAS"}
                                     if(a1=="HA") {a2="MXL"}
                                     t1 = x2[4]
                                     if( t1=="FGluadjBMI" ) { t2="FG" }
                                     if( t1=="FInsadjBMI" ) { t2="FI" }
                                     if( t1=="2hrGluadjBMI" ) { t2="2hGlu" }
                                     if( t1 %in% c("HbA1c","HbA1cadjFGlu") ) { t2="HbA1c" }
                                     c1 = x2[3]
                                     if( c1 %in% c("A","AX"))  { c2=1 } else { c2=0 }
                                     return( c( CHIP=x2[2] , ANCESTRY=a2 , TRAIT=t2 , AUTOSOME_IN=c2 ) )
                                   } )

table(ff1["CHIP",])
table(ff1["ANCESTRY",])
table(ff1["TRAIT",])
table(ff1["AUTOSOME_IN",])

table(ff1["TRAIT",],ff1["ANCESTRY",],ff1["CHIP",],ff1["AUTOSOME_IN",])
table(ff1["TRAIT",],ff1["ANCESTRY",],ff1["CHIP",])
table(ff1["CHIP",],ff1["AUTOSOME_IN",])


#chip="GWAS" ; ancestry="EUR" ; trait="FI" ; div=2
TODO = ifelse(!(ff0_file %in% EASYrep_CENT$fileInShortName),yes=1,no=0)
FUN1 = function( chip , ancestry , trait , autosome_in , div=1 )
{
 ww0=which( ff1["CHIP",]==chip & ff1["ANCESTRY",]==ancestry & ff1["TRAIT",]==trait & ff1["AUTOSOME_IN",]==autosome_in & TODO==1 )
 if (ancestry %in% c("EUR","AFR","ASN","AMR") ) { aafile = paste("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/allelefreq.1000G_",ancestry,"_p1v3.impute_legends.noMono.noDup.noX.v2.MarkerName.gz" ,sep="") }
 if (ancestry=="SAS" ) { aafile="/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/1000GP_p3v5_legends_rbind.noDup.MAFgt01.noCnv.noCnAll.noX.TW.afref.SAS.MarkerName.gz" }
 if (ancestry=="MXL" ) { aafile="/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/1000G_p1v3.MXL.FiltPass.NoDup.MAFgt01.NoX.AF_rbind.MarkerName.gz" }
 b = ceiling( length(ww0)/div )
 for ( i in 1:div )
 {
  ww1 = ww0[ ((i-1)*b+1):min((i*b),length(ww0)) ]
  #zz = NULL
  #for ( j in ww1 )
  #{
  # zz = c( zz , as.numeric(strsplit(system(paste("ls -l",ff0[j]),intern=T)," ",fixed=T)[[1]][5]) )
  #} ; rm(j)
  #mzz = max(zz)/1000000 ; rm(zz)
  ee=c( tt[1:2]
      , paste( "EASYIN --fileIn" , ff0[ww1] )
      , tt[3:70]
      , paste( tt[71] , " /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_",trait,"_for1000Gimputation.txt" ,sep="")
      , tt[72:79]
      , paste( tt[80] , aafile )
   , tt[81:88]
      )
  if( chip=="MCHIP" )
  {
    wwP=grep("GC --colPval",ee) ; ee[wwP] = paste( ee[wwP] , sub("#!! CAUTION for Lambda: for MCHIP","",ee[wwP+1]) ,sep="" ) ; rm(wwP)
    wwQ=grep("QQPLOT",ee) ; ee[wwQ-1]=sub("#","",ee[wwQ-1]) ; ee[wwQ]=paste(ee[wwQ]," --arcdExclude is.na(trait)") ; rm(wwQ)
    ## !! in case MCHIP file for X only ---> what to do with GC and QQ plot because no X SNPs in QT SNP file!
  }
  # highBETA needs to depend on the trait
  wwB=grep("N_highBETA",ee)
  if( trait=="FG"    ) { ee[wwB] = sub("xx","3.4",ee[wwB]) }
  if( trait=="FI"    ) { ee[wwB] = sub("xx","3"  ,ee[wwB]) }
  if( trait=="2hGlu" ) { ee[wwB] = sub("xx","8.7",ee[wwB]) }
  if( trait=="HbA1c" ) { ee[wwB] = sub("xx","2.4",ee[wwB]) }
  rm(wwB)
  # for X only files, remove the AFCHECK function!
  if(autosome_in==0) { wwA=grep("AFCHECK",ee) ; ee=ee[-c(wwA:(wwA+9))] ; rm(wwA) }
  # write script
  write.table( ee , paste("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_",chip,"_",ancestry,"_",trait,"_late_",autosome_in,"_",i,".ecf",sep="") , quote=F , col.names=F , row.names=F )
  rm(ww1,ee)
 }
}


FUN2 = function( chip , autosome_in , div=1 )
{
 ww0=which( ff1["CHIP",]==chip & ff1["AUTOSOME_IN",]==autosome_in )
 b = ceiling( length(ww0)/div )
 for ( i in 1:div )
 {
  ww1 = ww0[ ((i-1)*b+1):min((i*b),length(ww0)) ]
  ee=c( ttc[1:2]
      , paste( "EASYIN --fileIn" , ff0[ww1] )
      , ttc[3:12]
      )
  # write script
  write.table( ee , paste("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/c_",chip,"_",autosome_in,"_",i,".ecf",sep="") , quote=F , col.names=F , row.names=F )
  rm(ww1,ee)
 }
}



# done# FUN1( chip="GWAS" , ancestry="AFR" , trait="FG" , autosome_in=1 , div=2 )

# GWAS - EUR - FG done for  Peggy/Ching-Ti files only!!!  --> will  need to run Thibaud/Reedik ones later!
FUN1( chip="GWAS" , ancestry="EUR" , trait="FG" , autosome_in=1 , div=10 )
FUN1( chip="GWAS" , ancestry="EUR" , trait="FG" , autosome_in=0 , div=2 )

# late FG file from Thibaud/Reedik
FUN1( chip="GWAS" , ancestry="EUR" , trait="FG" , autosome_in=0 , div=2 )
FUN1( chip="GWAS" , ancestry="ASN" , trait="FG" , autosome_in=0 , div=1 )
FUN1( chip="GWAS" , ancestry="EUR" , trait="FG" , autosome_in=1 , div=3 )
FUN1( chip="GWAS" , ancestry="ASN" , trait="FG" , autosome_in=1 , div=3 )
FUN1( chip="GWAS" , ancestry="MXL" , trait="FG" , autosome_in=1 , div=2 )
FUN1( chip="GWAS" , ancestry="SAS" , trait="FG" , autosome_in=1 , div=2 )
FUN1( chip="MCHIP" , ancestry="ASN" , trait="FG" , autosome_in=1 , div=1 )
FUN1( chip="MCHIP" , ancestry="EUR" , trait="FG" , autosome_in=1 , div=2 )



FUN1( chip="GWAS" , ancestry="MXL" , trait="FI" , autosome_in=1 , div=2 )
FUN1( chip="GWAS" , ancestry="MXL" , trait="2hGlu" , autosome_in=1 , div=2 )
FUN1( chip="GWAS" , ancestry="MXL" , trait="HbA1c" , autosome_in=1 , div=2 )



# done# FUN1( chip="GWAS" , ancestry="AFR" , trait="FI" , autosome_in=1 , div=2 )
# done# FUN1( chip="GWAS" , ancestry="ASN" , trait="FI" , autosome_in=1 , div=3 )
# done# FUN1( chip="GWAS" , ancestry="EUR" , trait="FI" , autosome_in=1 , div=10 )
# done# FUN1( chip="GWAS" , ancestry="SAS" , trait="FI" , autosome_in=1 , div=2 )
FUN1( chip="GWAS" , ancestry="AMR" , trait="FI" , autosome_in=1 )
# done# FUN1( chip="GWAS" , ancestry="AFR" , trait="FI" , autosome_in=0 , div=1 )
# done# FUN1( chip="GWAS" , ancestry="ASN" , trait="FI" , autosome_in=0 , div=1 )
# done# FUN1( chip="GWAS" , ancestry="EUR" , trait="FI" , autosome_in=0 , div=1 )
FUN1( chip="GWAS" , ancestry="AMR" , trait="FI" , autosome_in=0 )

# done# FUN1( chip="GWAS" , ancestry="AFR" , trait="HbA1c" , autosome_in=1 , div=2 )
# done# FUN1( chip="GWAS" , ancestry="ASN" , trait="HbA1c" , autosome_in=1 , div=5 )
# done# FUN1( chip="GWAS" , ancestry="EUR" , trait="HbA1c" , autosome_in=1 , div=10 )
# done# FUN1( chip="GWAS" , ancestry="SAS" , trait="HbA1c" , autosome_in=1 , div=3 )
FUN1( chip="GWAS" , ancestry="AMR" , trait="HbA1c" , autosome_in=1 , div=2 )
# done# FUN1( chip="GWAS" , ancestry="AFR" , trait="HbA1c" , autosome_in=0 , div=1 )
# done# FUN1( chip="GWAS" , ancestry="ASN" , trait="HbA1c" , autosome_in=0 , div=1 )
# done# FUN1( chip="GWAS" , ancestry="EUR" , trait="HbA1c" , autosome_in=0 , div=1 )
FUN1( chip="GWAS" , ancestry="AMR" , trait="HbA1c" , autosome_in=0 , div=1 )

# done# FUN1( chip="MCHIP" , ancestry="ASN" , trait="FI" , autosome_in=1 , div=1 )
# done# FUN1( chip="MCHIP" , ancestry="EUR" , trait="FI" , autosome_in=1 , div=3 )

# done# FUN1( chip="MCHIP" , ancestry="ASN" , trait="HbA1c" , autosome_in=1 , div=1 )
# done# FUN1( chip="MCHIP" , ancestry="EUR" , trait="HbA1c" , autosome_in=1 , div=3 )


# done# FUN1( chip="GWAS" , ancestry="EUR" , trait="2hGlu" , autosome_in=0 , div=1 )
# done# FUN1( chip="GWAS" , ancestry="EUR" , trait="2hGlu" , autosome_in=1 , div=4 )
# done# FUN1( chip="GWAS" , ancestry="AFR" , trait="2hGlu" , autosome_in=1 , div=1 )
# done# FUN1( chip="GWAS" , ancestry="ASN" , trait="2hGlu" , autosome_in=1 , div=1 )
# done# FUN1( chip="MCHIP" , ancestry="EUR" , trait="2hGlu" , autosome_in=1 , div=2 )

#for f in `ls QC2_EasyQC_scripts/CENTRALLY_*FG*` ; do echo $f ; grep EASYIN $f | wc ; done



FUN2( chip="GWAS" , autosome_in=0 , div=10 )
FUN2( chip="GWAS" , autosome_in=1 , div=65 )
FUN2( chip="MCHIP" , autosome_in=1 , div=10 )


####################################################################################

rr=rnorm(10000)
ss = boxplot(rr,plot=F)$stats[,1]

c( quantile(rr,0.25,type=2)-1.5*(quantile(rr,0.75,type=2)-quantile(rr,0.25,type=2)),
   quantile(rr,0.25,type=2),
   median(rr),
   quantile(rr,0.75,type=2),
   quantile(rr,0.75,type=2)+1.5*(quantile(rr,0.75,type=2)-quantile(rr,0.25,type=2)) )

wu=quantile(rr,0.75,type=2)+1.5*(quantile(rr,0.75,type=2)-quantile(rr,0.25,type=2))
max(rr[which(rank(abs(wu-rr))<=2 & sign(wu-rr)==1)])

wl=quantile(rr,0.25,type=2)-1.5*(quantile(rr,0.75,type=2)-quantile(rr,0.25,type=2))
min(rr[which(rank(abs(rr-wl))<=2 & sign(rr-wl)==1)])

out=boxplot(rr,plot=F)$out
boxplot(ss,ylim=c(min(out),max(out))) ; points(rep(1,length(out)),out)

bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_15.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_15.err -R"select[mem>15000]  rusage[mem=15000]"  -M15000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_MCHIP_ASN_FI_1.ecf    < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_16.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_16.err -R"select[mem>15000]  rusage[mem=15000]"  -M15000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_MCHIP_ASN_HbA1c_1.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_17.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_17.err -R"select[mem>15000]  rusage[mem=15000]"  -M15000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_MCHIP_EUR_FI_1.ecf    < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_18.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_18.err -R"select[mem>15000]  rusage[mem=15000]"  -M15000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_MCHIP_EUR_HbA1c_1.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_19.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_19.err -R"select[mem>16000]  rusage[mem=16000]"  -M16000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_MCHIP_EUR_HbA1c_2.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"




for f in `ls QC2_EasyQC_scripts/CENTRALLY_*.ecf` ; do echo $f ; grep EASYIN $f | wc ; done
for f in `ls QC2_EasyQC_scripts/bsub_CENT_*.out` ; do echo $f ; grep Processing $f | wc ; done
grep "Succesfully finished EasyX" QC2_EasyQC_scripts/bsub_CENT_*| wc
grep Success QC2_EasyQC_scripts/bsub_CENT_GWAS* | wc
grep "Max Memory" QC2_EasyQC_scripts/bsub_CENT_MCHIP*
grep "Max Memory" QC2_EasyQC_scripts/bsub_CENT_GWAS*


#cd QC2_EasyQC_scripts/
#i=0 ; for s in `ls -1 CENTRALLY_GWAS_*.ecf` ; do ((i+=1)) ; bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=37 ; for s in `ls -1 CENTRALLY_GWAS_SAS*.ecf` ; do ((i+=1)) ; bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=0 ; for s in `ls -1 CENTRALLY_MCHIP_*.ecf`; do ((i+=1)) ; bsub -q normal  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_MCHIP_$i.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_MCHIP_$i.err -R"select[mem>16000]  rusage[mem=16000]"  -M16000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=43 ; for s in `ls -1 CENTRALLY_GWAS_AFR_FG_1*.ecf` ; do ((i+=1)) ; bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=45 ; for s in `ls -1 CENTRALLY_GWAS_*_2hGlu*.ecf`  ; do ((i+=1)) ; bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=52 ; for s in `ls -1 CENTRALLY_MCHIP_*_2hGlu*.ecf` ; do ((i+=1)) ; bsub -q normal  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.err  -R"select[mem>16000]  rusage[mem=16000]"  -M16000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=54 ; for s in `ls -1 CENTRALLY_GWAS_EUR_FG*.ecf`   ; do ((i+=1)) ; bsub -q normal  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=66 ; for s in `ls -1 CENTRALLY_GWAS_MXL*.ecf`   ; do ((i+=1)) ; bsub -q normal  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=68 ; for s in `ls -1 CENTRALLY_GWAS_MXL_FI*.ecf`   ; do ((i+=1)) ; bsub -q normal  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=0 ; for s in `ls -1 CENTRALLY_AFforIAAvsEUR*.ecf`   ; do ((i+=1)) ; bsub -q hugemem  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_IAAvsEUR_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_IAAvsEUR_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=0 ; for s in `ls -1 CENTRALLY_MCHIP_*late*.ecf`; do ((i+=1)) ; bsub -q normal  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_MCHIP_late_$i.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_MCHIP_late_$i.err -R"select[mem>16000]  rusage[mem=16000]"  -M16000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=0 ; for s in `ls -1 CENTRALLY_GWAS_*late*.ecf` ; do ((i+=1)) ; bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_late_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_late_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#cd ../
#bsub -q hugemem  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_12.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_12.err -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_GWAS_ASN_HbA1c_1_3.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
#bsub -q hugemem  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_37.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_37.err -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_GWAS_AFR_FI_1_1b.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
#bsub -q hugemem  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_43.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_43.err -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_GWAS_AFR_FI_1_1c.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
#bsub -q hugemem  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_73.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_GWAS_73.err -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_GWAS_MXL_FI_1_1b.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"
#bsub -q hugemem  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_CRCvsASN.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENT_CRCvsASN.err -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args CENTRALLY_AFforCRCvsASN.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"

for f in `ls QC2_EasyQC_scripts/bsub_CENT_GWAS_69*.out`  ; do echo $f ; grep Running $f | sed 's/Running script  \/lustre\/scratch113\/projects\/t35_magic\/TRANSETHNIC_1000G\/gaelle\/QC2_EasyQC_scripts\///g' ; grep ".gz" $f | grep "+  /" | wc ; grep Processing $f | wc ; done
for f in `ls QC2_EasyQC_scripts/bsub_CENT_GWAS_7*.out`  ; do echo $f ; grep Running $f | sed 's/Running script  \/lustre\/scratch113\/projects\/t35_magic\/TRANSETHNIC_1000G\/gaelle\/QC2_EasyQC_scripts\///g' ; grep ".gz" $f | grep "+  /" | wc ; grep Processing $f | wc ; done
for f in `ls QC2_EasyQC_scripts/bsub_CENT_MCHIP*.out` ; do echo $f ; grep Running $f | sed 's/Running script  \/lustre\/scratch113\/projects\/t35_magic\/TRANSETHNIC_1000G\/gaelle\/QC2_EasyQC_scripts\///g' ; grep ".gz" $f | grep "+  /" | wc ; grep Processing $f | wc ; done


grep "Succesfully finished EasyX" QC2_EasyQC_scripts/bsub_CENT_GWAS_6*
grep Success QC2_EasyQC_scripts/bsub_CENT_GWAS_6*



#cd QC2_EasyQC_scripts/
#i=0  ; for s in `ls -1 c_MCHIP_*.ecf`  ; do ((i+=1)) ; bsub -q normal  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_c_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_c_$i.err  -R"select[mem>16000]  rusage[mem=16000]"  -M16000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=10 ; for s in `ls -1 c_GWAS_0_*.ecf` ; do ((i+=1)) ; bsub -q normal  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_c_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_c_$i.err  -R"select[mem>16000]  rusage[mem=16000]"  -M16000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=20 ; for s in `ls -1 c_GWAS_1_*.ecf` ; do ((i+=1)) ; bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_c_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_c_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#cd ../
bsub -q hugemem  -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_c_60.out -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_c_60.err -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args c_GWAS_1_46.ecf < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R"

grep "Succesfully finished EasyX" QC2_EasyQC_scripts/bsub_c_*
for f in `ls QC2_EasyQC_scripts/bsub_c_*.out`  ; do echo $f ; grep Running $f | sed 's/Running script  \/lustre\/scratch113\/projects\/t35_magic\/TRANSETHNIC_1000G\/gaelle\/QC2_EasyQC_scripts\///g' ; grep ".gz" $f | grep "+  /" | wc ; grep Processing $f | wc ; done




grep Success QC2_EasyQC_scripts/bsub_CENT_GWAS* | wc
bjobs | grep hugemem | wc
bjobs | grep hugemem | grep RUN | wc


ls -1 QC2_EasyQC_scripts/CENTRALLY_*.ecf |wc
ls -1 QC2_EasyQC_scripts/CENTRALLY_*_0_*.ecf | wc
ls -1 QC2_CENTRAL_results/CENTRALLY_*AFCHECK.png | wc
ls -1 QC2_CENTRAL_results/CENTRALLY_*PZ-PLOTS.png | wc
ls -1 QC2_CENTRAL_results/CENTRALLY_*.rep | wc
ls -1 QC2_CENTRAL_results/*.qq.png | wc



qqPlots = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results",full.names=T) ; qqPlots = qqPlots[grep("qq.png",qqPlots)]
qqPlots_file = sub("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/","",sub(".qq.png","",qqPlots))

table( qqPlots_file %in% ff0_file)
table( ff0_file %in% qqPlots_file)
table(ff1["TRAIT",],ff1["ANCESTRY",],ff0_file %in% qqPlots_file)



## EasyQC reports from CENTRAL EasyQC for all cleaned files

repFiles = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results",full.names=T) ; repFiles = repFiles[grep("rep",repFiles)]

AFCHECKcolumns = c("AFCHECK.NotInRef","AFCHECK.NotInIn","AFCHECK.Checked","AFCHECK.StrandChange","AFCHECK.AlleleMatch","AFCHECK.AlleleChange",
                   "AFCHECK.AlleleMismatch","AFCHECK.AlleleInMissing","AFCHECK.AlleleInInvalid","AFCHECK.StrandInInvalid","AFCHECK.AlleleRefMissing",
                   "AFCHECK.AlleleRefInvalid","AFCHECK.StrandRefInvalid","AFCHECK.cor_eaf.ref_EAF_QTL","AFCHECK.numOutlier")

EASYrep_CENT = NULL ; MERGE_QT_SNPs=NULL ; i=0
for ( ff in repFiles)
{
 i=i+1
 cat(i,".. ")
 tt = read.table(ff,h=T,as.is=T)
 #if (!("KnownSnpsnumExtractMissing" %in% colnames(tt))) { tt$KnownSnpsnumExtractMissing=0 }
 if( grepl("_0_",ff) ) { tt=cbind( tt , matrix( NA , ncol=length(AFCHECKcolumns) , nrow=nrow(tt) ,dimnames=list(NULL,AFCHECKcolumns) ) ) }
 tt$Job=sub("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/","",sub(".rep","",ff))
 tt$JobFileNum = 1:nrow(tt)
 if( grepl("MCHIP",ff) ) { MERGE_QT_SNPs=rbind( MERGE_QT_SNPs , tt[,c("fileInShortName","NotInRef","NotInIn","Job","JobFileNum")] ) ; tt$NotInRef=NULL ; tt$NotInIn=NULL }
 EASYrep_CENT = rbind( EASYrep_CENT , tt )
 rm(tt)
} ; rm(ff,i,AFCHECKcolumns) ; cat("Done.\n")

summary(MERGE_QT_SNPs$NotInIn)
summary(MERGE_QT_SNPs$NotInRef)

table(duplicated(EASYrep_CENT$fileInShortName))
table(is.na(EASYrep_CENT$numSNPsOut))

table( EASYrep_CENT$fileInShortName %in% ff0_file )
table( ff0_file %in% EASYrep_CENT$fileInShortName )
table(ff1["TRAIT",],ff1["ANCESTRY",],ff0_file %in% EASYrep_CENT$fileInShortName)
table(ff1["TRAIT",],ff1["ANCESTRY",],ff0_file %in% EASYrep_CENT$fileInShortName[which(duplicated(EASYrep_CENT$fileInShortName))] )

table( duplicated(EASYrep_CENT$fileInShortName) , duplicated(EASYrep_CENT[,-c(117,118)]) )
table( duplicated(EASYrep_CENT$fileInShortName) , is.na(EASYrep_CENT$numSNPsOut) )
EASYrep_CENT[ which(duplicated(EASYrep_CENT$fileInShortName)) , c(117,118) ]

table(is.na(EASYrep_CENT$numSNPsOut))
table(grepl("CLEANED",EASYrep_CENT$fileInShortName),grepl("late",EASYrep_CENT$Job))

EASYrep_CENT = EASYrep_CENT[ -which(is.na(EASYrep_CENT$numSNPsOut)) , ]
EASYrep_CENT = EASYrep_CENT[ -which(duplicated(EASYrep_CENT$fileInShortName)) , ]
EASYrep_CENT$fileInShortName = sub("CLEANED.","",EASYrep_CENT$fileInShortName)

tp = sapply( EASYrep_CENT$fileInShortName , FUN=function(x){ x1=strsplit(x,"_")[[1]]
                                                        if( x1[5]=="InvNorm" ) {tt="InvNorm";ss=x1[6]} else {tt="Raw";ss=x1[5]}
                                                        return(c(Ancestry=x1[1],Chip=x1[2],Chr=x1[3],Trait=x1[4],Transf=tt,Study=ss))
                                                      } )

EASYrep_CENT$Ancestry = tp["Ancestry",]
EASYrep_CENT$Chip     = tp["Chip",]
EASYrep_CENT$Chr      = tp["Chr",]
EASYrep_CENT$Trait    = tp["Trait",]
EASYrep_CENT$Transf   = tp["Transf",]
EASYrep_CENT$Study    = tp["Study",]
rm(tp,MERGE_QT_SNPs,repFiles)

table(EASYrep_CENT$Ancestry)
table(EASYrep_CENT$Chip)
table(EASYrep_CENT$Chr)
table(EASYrep_CENT$Trait)
table(EASYrep_CENT$Transf)

save( EASYrep_CENT , file="/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/EASYrep_CENT.RData" )

load("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/EASYrep_CENT.RData" )
## EasyQC reports from computation of c with EasyQC for all cleaned files (c is the constant used in the sqrt(N) vs 1/med(SE) plot as recommended in the EasyQC paper)
repFiles = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_c_results",full.names=T) ; repFiles = repFiles[grep("rep",repFiles)]

EASYrep_c = NULL ; i=0
for ( ff in repFiles)
{
 i=i+1
 cat(i,".. ")
 tt = read.table(ff,h=T,as.is=T)
 tt$Job=sub("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_c_results/","",sub(".rep","",ff))
 tt$JobFileNum = 1:nrow(tt)
 EASYrep_c = rbind( EASYrep_c , tt )
 rm(tt)
} ; rm(ff,i,repFiles) ; cat("Done.\n")

EASYrep_c$fileInShortName = sub("CLEANED.","",EASYrep_c$fileInShortName)

tp = sapply( EASYrep_c$fileInShortName , FUN=function(x){ x1=strsplit(x,"_")[[1]]
                                                        if( x1[5]=="InvNorm" ) {tt="InvNorm";ss=x1[6]} else {tt="Raw";ss=x1[5]}
                                                        return(c(Ancestry=x1[1],Chip=x1[2],Chr=x1[3],Trait=x1[4],Transf=tt,Study=ss))
                                                      } )

EASYrep_c$Ancestry = tp["Ancestry",]
EASYrep_c$Chip     = tp["Chip",]
EASYrep_c$Chr      = tp["Chr",]
EASYrep_c$Trait    = tp["Trait",]
EASYrep_c$Transf   = tp["Transf",]
EASYrep_c$Study    = tp["Study",]
rm(tp)



table( EASYrep_c$fileInShortName %in% EASYrep_CENT$fileInShortName )

EASYrep_check = merge( EASYrep_CENT , EASYrep_c , by="fileInShortName" )
summary( EASYrep_check$Clean.SampleSize.x - EASYrep_check$Clean.SampleSize.y )
summary( EASYrep_check$Clean.MedianSE.x - EASYrep_check$Clean.MedianSE.y )
rm(EASYrep_check)

EASYrep_table = merge( EASYrep_CENT[,c("fileInShortName","Ancestry","Chip","Chr","Trait","Transf","Study","Clean.SampleSize","Clean.MedianSE")] ,
                       EASYrep_c[,c("fileInShortName","Clean.c")] ,
                       by="fileInShortName" )
EASYrep_table$sqrtN = sqrt(EASYrep_table$Clean.SampleSize)
EASYrep_table$one_over_medianSE = 1/(EASYrep_table$Clean.MedianSE)
EASYrep_table$c_over_medianSE = (EASYrep_table$Clean.c)/(EASYrep_table$Clean.MedianSE)
write.table(EASYrep_table,"/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ValuesForNvsSEplot.txt",quote=F,row.names=F,sep="\t")

## AF check
plot(EASYrep_CENT$AFCHECK.cor_eaf.ref_EAF_QTL , EASYrep_CENT$AFCHECK.numOutlier)
abline(h=1000,col="orange")

table( EASYrep_CENT$AFCHECK.numOutlier==0 )
table( EASYrep_CENT$AFCHECK.numOutlier<50 )

EASYrep_CENT$AFCHECK.numOutlier[(EASYrep_CENT$AFCHECK.numOutlier)<50]
EASYrep_CENT$AFCHECK.numOutlier[(EASYrep_CENT$AFCHECK.numOutlier)>50]
tt1=EASYrep_CENT[(EASYrep_CENT$AFCHECK.numOutlier)>1000,c("Study","Ancestry","Trait","Transf","Chip","Chr","AFCHECK.numOutlier")];tt1[order(tt1$Study,tt1$Ancestry),];rm(tt1)

binsAFO = ifelse( EASYrep_CENT$AFCHECK.numOutlier==0 , yes="0" ,
                  no=ifelse( EASYrep_CENT$AFCHECK.numOutlier<1000 , yes="<1,000" ,
                             no=ifelse( EASYrep_CENT$AFCHECK.numOutlier<10000 , yes="<10,000" , no=">10,000" )) )
table(binsAFO)
table(binsAFO,EASYrep_CENT$Ancestry)


AF = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results",full.names=T) ; AF = AF[grep("AFCHECK",AF)]
length(AF)
head(AF)
AF2 = sub("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/","",sub(".multi.AFCHECK.png","",AF))
repFiles2 = sub("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/","",sub(".rep","",repFiles))
table(repFiles2 %in% AF2 , grepl("_0_",repFiles2)) # chrX only files don't have AFCHECK plot




## sqrt(N) vs 1/med(SE) -- by trait and raw/invnorm
png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_N_vs_SE_850.png",width=500,heigh=1250)
par(mfcol=c(5,2))
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="FGluadjBMI" & EASYrep_CENT$Transf=="Raw")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="FGluadjBMI" & EASYrep_CENT$Transf=="Raw")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="Raw FGluadjBMI" )
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="FInsadjBMI" & EASYrep_CENT$Transf=="Raw")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="FInsadjBMI" & EASYrep_CENT$Transf=="Raw")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="Raw FInsadjBMI" )
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="2hrGluadjBMI" & EASYrep_CENT$Transf=="Raw")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="2hrGluadjBMI" & EASYrep_CENT$Transf=="Raw")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="Raw 2hrGluadjBMI"  )
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="HbA1c" & EASYrep_CENT$Transf=="Raw")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="HbA1c" & EASYrep_CENT$Transf=="Raw")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="Raw HbA1c"  )
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="HbA1cadjFGlu" & EASYrep_CENT$Transf=="Raw")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="HbA1cadjFGlu" & EASYrep_CENT$Transf=="Raw")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="Raw HbA1cadjFGlu"  )

plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="FGluadjBMI" & EASYrep_CENT$Transf=="InvNorm")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="FGluadjBMI" & EASYrep_CENT$Transf=="InvNorm")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="InvNorm FGluadjBMI"  )
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="FInsadjBMI" & EASYrep_CENT$Transf=="InvNorm")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="FInsadjBMI" & EASYrep_CENT$Transf=="InvNorm")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="InvNorm FInsadjBMI"  )
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="2hrGluadjBMI" & EASYrep_CENT$Transf=="InvNorm")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="2hrGluadjBMI" & EASYrep_CENT$Transf=="InvNorm")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="InvNorm 2hrGluadjBMI"  )
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="HbA1c" & EASYrep_CENT$Transf=="InvNorm")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="HbA1c" & EASYrep_CENT$Transf=="InvNorm")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="InvNorm HbA1c"  )
plot( sqrt(EASYrep_CENT$Clean.SampleSize[which(EASYrep_CENT$Trait=="HbA1cadjFGlu" & EASYrep_CENT$Transf=="InvNorm")]) , 1/(EASYrep_CENT$Clean.MedianSE[which(EASYrep_CENT$Trait=="HbA1cadjFGlu" & EASYrep_CENT$Transf=="InvNorm")]) , xlab="sqrt(N)" , ylab="1/med(SE)" , main="InvNorm HbA1cadjFGlu"  )
dev.off()

sqrtN = sqrt(EASYrep_CENT$Clean.SampleSize)
mSEinv = 1/(EASYrep_CENT$Clean.MedianSE)

trait =  "HbA1cadjFGlu" # FGluadjBMI,FInsadjBMI,2hrGluadjBMI,HbA1c,HbA1cadjFGlu
transf = "InvNorm" # Raw,InvNorm
ww = which(EASYrep_CENT$Trait==trait & EASYrep_CENT$Transf==transf)
plot( sqrtN[ww] , mSEinv[ww] , xlab="sqrt(N)" , ylab="1/med(SE)" , main=paste(trait,transf) )
ss=which( EASYrep_CENT$Study[ww]=="WHI" )
points( sqrtN[ww][ss] , mSEinv[ww][ss] , col="red",pch=20 ) ; text( sqrtN[ww][ss] , mSEinv[ww][ss] , "WHI" , pos=3 , offset=0.5 , col="red" )

x = 58
y = 13
abline(v=x,h=y,col="grey")
sum( sqrtN[ww]>x  & mSEinv[ww]<y )
ww2 = which(sqrtN[ww]>x  & mSEinv[ww]<y)
cbind( EASYrep_CENT[ww,][ww2,c("fileInShortName","Study","Trait","Ancestry","Transf","Clean.SampleSize","Clean.MedianSE")] ,  sqrtN[ww][ww2] , mSEinv[ww][ww2] )

x2 = 40
y2 = 30
abline(v=x2,h=y2,col="orange")
ww22 = which(sqrtN[ww]<x2  & mSEinv[ww]>y2)
cbind( EASYrep_CENT[ww,][ww22,c("fileInShortName","Study","Trait","Ancestry","Transf","Clean.SampleSize","Clean.MedianSE")] ,  sqrtN[ww][ww22] , mSEinv[ww][ww22] )



# with c
summary(EASYrep_c$Clean.c)

tapply( EASYrep_c$Clean.c , INDEX=as.factor(paste(EASYrep_c$Chip,EASYrep_c$Ancestry,sep="_")) , FUN=summary )


sqrtN_c = sqrt(EASYrep_c$Clean.SampleSize)
cmSEinv = (EASYrep_c$Clean.c)/(EASYrep_c$Clean.MedianSE)

ww_FG_R = which(EASYrep_c$Trait=="FGluadjBMI" & EASYrep_c$Transf=="Raw")
ww_FG_I = which(EASYrep_c$Trait=="FGluadjBMI" & EASYrep_c$Transf=="InvNorm")

ww_FI_R = which(EASYrep_c$Trait=="FInsadjBMI" & EASYrep_c$Transf=="Raw")
ww_FI_I = which(EASYrep_c$Trait=="FInsadjBMI" & EASYrep_c$Transf=="InvNorm")

ww_2H_R = which(EASYrep_c$Trait=="2hrGluadjBMI" & EASYrep_c$Transf=="Raw")
ww_2H_I = which(EASYrep_c$Trait=="2hrGluadjBMI" & EASYrep_c$Transf=="InvNorm")

ww_HU_R = which(EASYrep_c$Trait=="HbA1c" & EASYrep_c$Transf=="Raw")
ww_HU_I = which(EASYrep_c$Trait=="HbA1c" & EASYrep_c$Transf=="InvNorm")

ww_HA_R = which(EASYrep_c$Trait=="HbA1cadjFGlu" & EASYrep_c$Transf=="Raw")
ww_HA_I = which(EASYrep_c$Trait=="HbA1cadjFGlu" & EASYrep_c$Transf=="InvNorm")

png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_N_vs_cSE.png",width=500,heigh=1250)
par(mfcol=c(5,2))
plot( sqrtN_c[ww_FG_R] , cmSEinv[ww_FG_R] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="Raw FGluadjBMI" )
plot( sqrtN_c[ww_FI_R] , cmSEinv[ww_FI_R] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="Raw FInsadjBMI" )
plot( sqrtN_c[ww_2H_R] , cmSEinv[ww_2H_R] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="Raw 2hrGluadjBMI" )
plot( sqrtN_c[ww_HU_R] , cmSEinv[ww_HU_R] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="Raw HbA1c" )
plot( sqrtN_c[ww_HA_R] , cmSEinv[ww_HA_R] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="Raw HbA1cadjFGlu" )

plot( sqrtN_c[ww_FG_I] , cmSEinv[ww_FG_I] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="InvNorm FGluadjBMI" )
plot( sqrtN_c[ww_FI_I] , cmSEinv[ww_FI_I] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="InvNorm FInsadjBMI" )
plot( sqrtN_c[ww_2H_I] , cmSEinv[ww_2H_I] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="InvNorm 2hrGluadjBMI" )
plot( sqrtN_c[ww_HU_I] , cmSEinv[ww_HU_I] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="InvNorm HbA1c" )
plot( sqrtN_c[ww_HA_I] , cmSEinv[ww_HA_I] , xlab="sqrt(N)" , ylab="c/med(SE)" , main="InvNorm HbA1cadjFGlu" )
dev.off()

ww=ww_HA_I

x = 13
y = 34
abline(v=x,h=y,col="grey")
ww2 = which(sqrtN_c[ww]>x  & cmSEinv[ww]<y)
cbind( EASYrep_c[ww,][ww2,c("fileInShortName","Study","Trait","Ancestry","Transf","Clean.SampleSize","Clean.MedianSE")] ,  sqrtN_c[ww][ww2] , cmSEinv[ww][ww2] )

x2 = 70
y2 = 70
abline(v=x2,h=y2,col="orange")
ww22 = which(sqrtN_c[ww]<x2  & cmSEinv[ww]>y2)
cbind( EASYrep_c[ww,][ww22,c("fileInShortName","Study","Trait","Ancestry","Transf","Clean.SampleSize","Clean.MedianSE")] ,  sqrtN_c[ww][ww22] , cmSEinv[ww][ww22] )


## known loci plots -- only raw analyses -- by trait and ancestry

H2 = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_2hGlu_for1000Gimputation.txt",as.is=T,h=T)
FG = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_FG_for1000Gimputation.txt",as.is=T,h=T)
FI = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_FI_for1000Gimputation.txt",as.is=T,h=T)
Hb = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/ReferenceFiles/MAGIC_QC_known_SNPs_HbA1c_for1000Gimputation.txt",as.is=T,h=T)

table(substr(H2$MarkerName,1,2))
table(substr(FG$MarkerName,1,2))
table(substr(FI$MarkerName,1,2))
table(substr(Hb$MarkerName,1,2))

wwFG = which(EASYrep_CENT$Trait=="FGluadjBMI" & EASYrep_CENT$Transf=="Raw")
nrow(FG)
table( EASYrep_CENT$KnownSnpsnumExtractMissing[wwFG] , EASYrep_CENT$Chr[wwFG] )

wwFI = which(EASYrep_CENT$Trait=="FInsadjBMI" & EASYrep_CENT$Transf=="Raw")
nrow(FI)
table( EASYrep_CENT$KnownSnpsnumExtractMissing[wwFI] , EASYrep_CENT$Chr[wwFI] )

wwH2 = which(EASYrep_CENT$Trait=="2hrGluadjBMI" & EASYrep_CENT$Transf=="Raw")
nrow(H2)
table( EASYrep_CENT$KnownSnpsnumExtractMissing[wwH2] , EASYrep_CENT$Chr[wwH2] )

wwHU = which(EASYrep_CENT$Trait=="HbA1c" & EASYrep_CENT$Transf=="Raw")
nrow(Hb)
table( EASYrep_CENT$KnownSnpsnumExtractMissing[wwHU] , EASYrep_CENT$Chr[wwHU] )

wwHA = which(EASYrep_CENT$Trait=="HbA1cadjFGlu" & EASYrep_CENT$Transf=="Raw")
nrow(Hb)
table( EASYrep_CENT$KnownSnpsnumExtractMissing[wwHA] , EASYrep_CENT$Chr[wwHA] )



FUNknownLoci = function(ancestry,trait)
{
  summaryKnownLociPerFile = NULL
  AllKnownLoci = NULL
  ww = which(EASYrep_CENT$Ancestry==ancestry & EASYrep_CENT$Trait==trait & EASYrep_CENT$Transf=="Raw"  & EASYrep_CENT$Chr %in% c("A","AX"))
  if(length(ww)>0)
  {
  for ( i in ww )
  {
    tp1=read.table(paste("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/",EASYrep_CENT$fileInShortName[i],".KnownSnps.extracted",sep=""),h=T,as.is=T,colClasses=c("character","character","character","numeric","numeric","character","character","numeric","numeric","numeric","numeric","numeric","numeric","numeric","numeric","numeric","numeric"))
    wwK = which( (tp1$raising_allele==tp1$EFFECT_ALLELE & tp1$other_allele==tp1$NON_EFFECT_ALLELE)
               | (tp1$raising_allele==tp1$NON_EFFECT_ALLELE & tp1$other_allele==tp1$EFFECT_ALLELE) )
    if (length(wwK)>1)
    {
     tp2 = tp1[wwK,1:12]
     tp2$Study = EASYrep_CENT$Study[i]
     tp2$Ancestry = EASYrep_CENT$Ancestry[i]
     tp2$Chip = EASYrep_CENT$Chip[i]
     tp2$Chr = EASYrep_CENT$Chr[i]
     wwF = which( tp2$other_allele==tp2$EFFECT_ALLELE & tp2$raising_allele==tp2$NON_EFFECT_ALLELE )
     if (length(wwF)>1) { tp2$BETA[wwF]=(-tp2$BETA)[wwF] ; tp2$EAF_QTL[wwF]=(1-tp2$EAF_QTL)[wwF] }
     AllKnownLoci = rbind( AllKnownLoci , tp2 )
     wwS = which(tp2$PVAL<0.05) ; if (length(wwS)>0) { SampleSizeLociMedian=median(tp2$N_QTL[wwS]) } else { SampleSizeLociMedian=NA }
     summaryKnownLociPerFile = rbind( summaryKnownLociPerFile , c(fileInShortName=EASYrep_CENT$fileInShortName[i] ,
                                                                  Study = EASYrep_CENT$Study[i],
                                                                  Ancestry = EASYrep_CENT$Ancestry[i],
                                                                  Chip = EASYrep_CENT$Chip[i],
                                                                  Chr = EASYrep_CENT$Chr[i],
                                                                  Trait = EASYrep_CENT$Trait[i],
                                                                  Transf = EASYrep_CENT$Transf[i],
                                                                  SampleSizeALL = EASYrep_CENT$Clean.SampleSize[i] ,
                                                                  Nloci = nrow(tp2) ,
                                                                  NlociSignif = length(wwS) ,
                                                                  NlociSignifSameDir = sum( tp2$PVAL<0.05 & sign(tp2$BETA)==sign(tp2$beta) ) ,
                                                                  SampleSizeLociMedian = SampleSizeLociMedian
                                                                  ) )
      rm(tp2,wwF,SampleSizeLociMedian,wwS)
    } else { summaryKnownLociPerFile = rbind( summaryKnownLociPerFile , c(fileInShortName=EASYrep_CENT$fileInShortName[i] ,
                                                                          Study = EASYrep_CENT$Study[i],
                                                                          Ancestry = EASYrep_CENT$Ancestry[i],
                                                                          Chip = EASYrep_CENT$Chip[i],
                                                                          Chr = EASYrep_CENT$Chr[i],
                                                                          Trait = EASYrep_CENT$Trait[i],
                                                                          Transf = EASYrep_CENT$Transf[i],
                                                                          SampleSizeALL = EASYrep_CENT$Clean.SampleSize[i] ,
                                                                          Nloci = 0 ,
                                                                          NlociSignif = 0 ,
                                                                          NlociSignifSameDir = 0,
                                                                          SampleSizeLociMedian = NA
                                                                          ) )
           }
    rm(wwK,tp1)
  } ; rm(i)
  summaryKnownLociPerFile = as.data.frame(summaryKnownLociPerFile,stringsAsFactors=F)
  for ( i in 8:12 ) { summaryKnownLociPerFile[,i]=as.numeric(summaryKnownLociPerFile[,i]) } ; rm(i)
  }
  return(list(AllKnownLoci,summaryKnownLociPerFile))
}

KnownLoci_FG_EA = FUNknownLoci(ancestry="EA",trait="FGluadjBMI")
KnownLoci_FG_AA = FUNknownLoci(ancestry="AA",trait="FGluadjBMI")

KnownLoci_FI_EA = FUNknownLoci(ancestry="EA",trait="FInsadjBMI")
KnownLoci_FI_AA = FUNknownLoci(ancestry="AA",trait="FInsadjBMI")
KnownLoci_FI_EAA = FUNknownLoci(ancestry="EAA",trait="FInsadjBMI")
KnownLoci_FI_IAA = FUNknownLoci(ancestry="IAA",trait="FInsadjBMI")

KnownLoci_HU_EA = FUNknownLoci(ancestry="EA",trait="HbA1c")
KnownLoci_HU_AA = FUNknownLoci(ancestry="AA",trait="HbA1c")
KnownLoci_HU_EAA = FUNknownLoci(ancestry="EAA",trait="HbA1c")
KnownLoci_HU_IAA = FUNknownLoci(ancestry="IAA",trait="HbA1c")

KnownLoci_HA_EA = FUNknownLoci(ancestry="EA",trait="HbA1cadjFGlu")
KnownLoci_HA_AA = FUNknownLoci(ancestry="AA",trait="HbA1cadjFGlu")
KnownLoci_HA_EAA = FUNknownLoci(ancestry="EAA",trait="HbA1cadjFGlu")
KnownLoci_HA_IAA = FUNknownLoci(ancestry="IAA",trait="HbA1cadjFGlu")

KnownLoci_H2_EA = FUNknownLoci(ancestry="EA",trait="2hrGluadjBMI")
KnownLoci_H2_AA = FUNknownLoci(ancestry="AA",trait="2hrGluadjBMI")
KnownLoci_H2_EAA = FUNknownLoci(ancestry="EAA",trait="2hrGluadjBMI")


plot( KnownLoci_FG_EA[[2]]$SampleSizeALL , KnownLoci_FG_EA[[2]]$SampleSizeLociMedian ) ; abline(0,1,col="grey")
plot( KnownLoci_FG_EA[[2]]$NlociSignif , KnownLoci_FG_EA[[2]]$NlociSignifSameDir/KnownLoci_FG_EA[[2]]$NlociSignif )

summaryKnownLoci=NULL
par( mfrow=c(5,5) )
for ( aa in c("AA","EA","EAA","IAA","HA") )
{
  for ( tt in c("FGluadjBMI","FInsadjBMI","2hrGluadjBMI","HbA1c","HbA1cadjFGlu") )
  {
    cat("---->",aa,tt,"\n")
    tp = FUNknownLoci(ancestry=aa,trait=tt)[[2]]
    if (!is.null(tp))
    {
     wwDN = which((tp$SampleSizeALL-tp$SampleSizeLociMedian)>1) ; if(length(wwDN)>0) { cat(length(wwDN),"studies with different overall and known loci sample sizes.\n") } ; rm(wwDN)
     ww0L = which(tp$NlociSignif==0) ; if(length(ww0L)>0) { cat(length(ww0L),"studies with no any significant known loci.\n") } ; rm(ww0L)
     ww1L = which(tp$NlociSignif>0)
     if( length(ww1L)>0 )
     {
       cat(length(ww1L),"study(ies) to plot.\n")
       plot( tp$SampleSizeLociMedian , tp$NlociSignifSameDir/tp$NlociSignif , main=paste(tt,"-",aa) , xlab="Median N" , ylab="% of concordant effect direction" , ylim=c(0,1) )
     } else { plot( 1 , 1 , main=paste(tt,"-",aa) , xaxt="n" , yaxt="n" , xlab="" , ylab="" , type="n" ) ; text(1,1,"No study with significant loci") }
     rm(ww1L)
     summaryKnownLoci = rbind( summaryKnownLoci , tp )
    } else { cat("No Study.\n") ; plot( 1 , 1 , main=paste(tt,"-",aa) , xaxt="n" , yaxt="n" , xlab="" , ylab="" , type="n" ) ; text(1,1,"No study") }
    rm(tp)
  } ; rm(tt)
} ; rm(aa)

length( which((summaryKnownLoci$SampleSizeALL-summaryKnownLoci$SampleSizeLociMedian)>1) )

summaryKnownLoci[ which((summaryKnownLoci$SampleSizeALL-summaryKnownLoci$SampleSizeLociMedian)>1) ,]

table(summaryKnownLoci$Ancestry,summaryKnownLoci$Trait)
summaryKnownLoci = rbind( summaryKnownLoci ,
                          c(NA,NA,"EAA",NA,NA,"FGluadjBMI",rep(NA,6)),
                          c(NA,NA,"IAA",NA,NA,"FGluadjBMI",rep(NA,6)),
                          c(NA,NA,"HA",NA,NA,"FGluadjBMI",rep(NA,6)),
                          c(NA,NA,"IAA",NA,NA,"2hrGluadjBMI",rep(NA,6))
                        )

tp1 = data.frame( fileInShortName = rep(NA,4) ,
                  Study = rep(NA,4) ,
                  Ancestry = c("EAA","IAA","HA","IAA") ,
                  Chip = rep(NA,4) ,
                  Chr = rep(NA,4) ,
                  Trait = c("FGluadjBMI","FGluadjBMI","FGluadjBMI","2hrGluadjBMI") ,
                  Transf = rep(NA,4) ,
                  SampleSizeALL = rep(NA,4) ,
                  Nloci = rep(NA,4) ,
                  NlociSignif = rep(NA,4) ,
                  NlociSignifSameDir = rep(NA,4) ,
                  SampleSizeLociMedian = rep(NA,4) ,
                  stringsAsFactors=F )
for ( i in 1:7 ) { tp1[,i]=as.character(tp1[,i]) } ; rm(i)
for ( i in 8:12 ) { tp1[,i]=as.numeric(tp1[,i]) } ; rm(i)
tp2 = rbind( summaryKnownLoci , tp1 )

sp <- ggplot(tp2, aes(x=SampleSizeLociMedian, y=NlociSignifSameDir/NlociSignif)) + geom_point(shape=1)
sp + facet_grid(Ancestry ~ Trait)



# Fumi code
library(rmeta)

data=rbind(  FUNknownLoci(ancestry="AA",trait="FGluadjBMI")[[1]]
            ,FUNknownLoci(ancestry="EA",trait="FGluadjBMI")[[1]]
          )

data=rbind(  FUNknownLoci(ancestry="AA",trait="FInsadjBMI")[[1]]
            ,FUNknownLoci(ancestry="EA",trait="FInsadjBMI")[[1]]
            ,FUNknownLoci(ancestry="EAA",trait="FInsadjBMI")[[1]]
            ,FUNknownLoci(ancestry="IAA",trait="FInsadjBMI")[[1]]
            ,FUNknownLoci(ancestry="HA",trait="FInsadjBMI")[[1]]
          )


data=rbind(  FUNknownLoci(ancestry="AA",trait="2hrGluadjBMI")[[1]]
            ,FUNknownLoci(ancestry="EA",trait="2hrGluadjBMI")[[1]]
            ,FUNknownLoci(ancestry="EAA",trait="2hrGluadjBMI")[[1]]
            ,FUNknownLoci(ancestry="HA",trait="2hrGluadjBMI")[[1]]
          )

data=rbind(  FUNknownLoci(ancestry="AA",trait="HbA1c")[[1]]
            ,FUNknownLoci(ancestry="EA",trait="HbA1c")[[1]]
            ,FUNknownLoci(ancestry="EAA",trait="HbA1c")[[1]]
            ,FUNknownLoci(ancestry="IAA",trait="HbA1c")[[1]]
            ,FUNknownLoci(ancestry="HA",trait="HbA1c")[[1]]
          )




data$SOURCE = paste(data$Ancestry,data$Chip,data$Study,sep="_")

summary(data$PVAL)
#data$BETA[which(data$PVAL>0.05)]=99

snps = union(data$MarkerName, c())
length(snps)
cohorts = union(data$SOURCE, c())
length(cohorts)

summaryeffect = c()
summarySE = c()
summaryP = c()
nstudies = c()
heterogeneityP = c()
for (snp in snps) {
        print(snp);
        m = meta.summaries(data$BETA[data$MarkerName==snp], data$SE[data$MarkerName==snp]);
        summaryeffect = c(summaryeffect, m$summary)
        summarySE = c(summarySE, m$se.summary)
        summaryP = c(summaryP, pnorm(-abs(m$test[1]))*2)
        nstudies = c(nstudies, m$het[2]+1)
        if (m$het[2] > 0) {
                heterogeneityP = c(heterogeneityP, m$het[3])
        } else {
                heterogeneityP = c(heterogeneityP, NA)
        }
}


stderrs = c();
samplesizeMax=c()
for (cohort in cohorts) {
        stderrs = c(stderrs, median(data$SE[data$SOURCE==cohort]))
        samplesizeMax = c(samplesizeMax,max(data$N_QTL[data$SOURCE==cohort]))
}

known = unique(data[,1:5])
table( known$MarkerName==snps )

plot( known$beta , summaryeffect , xlab="Expected effect for known loci" , ylab="combined effect from studies for known loci" ) ; abline(0,1,col="grey")

cohortsO = cohorts[order( sapply(cohorts,FUN=function(x){return(strsplit(x,"_")[[1]][1])}) , samplesizeMax , decreasing=T )]

table( sapply(cohorts,FUN=function(x){return(strsplit(x,"_")[[1]][1])})  )

mm = cbind( sort(known$beta),
                summaryeffect[order(known$beta)],
                    outer( snps[order(known$beta)],
                               cohortsO,
                               FUN=Vectorize(function(x,y){ ww=which(data$MarkerName==x & data$SOURCE==y) ; if (length(ww)==1) {return(ifelse(data$PVAL[ww]<0.05,yes=data$BETA[ww],no=99))} else {return(NA)} })
                         )
                  )
mmN = as.numeric(mm) ; mmN=mmN[which(mmN<99)]
summary(as.numeric(mm))
summary(mmN)

png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_BETAheatmap_FG.png",width=1000,heigh=1000)
png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_BETAheatmap_FI.png",width=1000,heigh=1500)
png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_BETAheatmap_2HrG.png",width=1000,heigh=1500)
png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_BETAheatmap_HbA1c.png",width=1000,heigh=1500)

nn = 30 # FG

par(xaxt="n")
par(yaxt="n")
par(mar = c(10, 4, 4, 20) + 0.1)
image(
        1:length(snps),
        1:(length(cohorts)+2),
        mm,
        col=c(heat.colors(nn),"grey75"), #topo.colors
        breaks=c( seq( from=min(mmN)-0.001 , to=max(mmN)+0.001 , length.out=(nn+1)  ) , 100 ) ,
        xlab="",
        ylab=""
)
par(xaxt="s")
par(yaxt="s")

par(las=2)
axis(side=1,
        labels=snps[order(known$beta)],
        at=c(1:length(snps)))
axis(side=4,
        labels=c("Expected effect","Combined effect", cohortsO),
        at=c(1:(length(cohorts)+2)))
par(las=0)
par(mar = c(5, 4, 4, 2) + 0.1)
#abline(h=50+2.5) # FG
#abline(h=c(6,8+6,13+8+6,62+13+8+6)+2.5) # FI
#abline(h=c(5,1+5,25+1+5,2+25+1+5)+2.5) # 2hrGlu
abline(h=c(7,5+7,15+5+7,39+15+5+7)+2.5) # HbA1c


dev.off()

par(xaxt="n")
par(yaxt="n")
par(mar = c(1, 6, 1, 1) + 0.1)
image(1,1:(nn+2),matrix(c(1:(nn+2)),nrow=1),col=c("grey75","white",heat.colors(nn)),xlab="",ylab="")
par(xaxt="s")
par(yaxt="s")
title(ylab="Beta legend",line=4)
par(las=2)
axis(side=2,
        labels=c(round(seq( from=min(mmN)-0.001 , to=max(mmN)+0.001 , length.out=(nn+1)  ) ,4)),
        at=c(  3:(nn+3)-0.5 ) )
text( c(1,1) , c(1,2) , c("p>0.05","Missing") )
par(las=0)
par(mar = c(5, 4, 4, 2) + 0.1)




threshold = 0.01
sumofsquareddifflength_allSNPs = c()
sumofsquareddiffp_allSNPs = c()
sumofsquareddiff = c()
sumofsquareddifflength = c()
sumofsquareddiffp = c()
sumofsquareddiff_fromzero = c()
sumofsquareddiff_opposite = c()
for (cohort in cohortsO) {
        datatemp = data[data$SOURCE==cohort & data$PVAL<0.05, ];
        datatemp = datatemp[match(snps, datatemp$MarkerName), ];
        x = na.omit( ((datatemp$BETA - summaryeffect) / datatemp$SE)^2 );
        sumofsquareddifflength_allSNPs = c(sumofsquareddifflength_allSNPs, length(x));
        sumofsquareddiffp_allSNPs = c(sumofsquareddiffp_allSNPs,
                pchisq(sum(x), length(x), lower.tail=F));
        x = na.omit( ( ((datatemp$BETA - summaryeffect) / datatemp$SE)^2 )[summaryP < threshold] );
        sumofsquareddiff = c(sumofsquareddiff, sum(x));
        sumofsquareddifflength = c(sumofsquareddifflength, length(x));
        sumofsquareddiffp = c(sumofsquareddiffp,
                pchisq(sum(x), length(x), lower.tail=F));
        x = na.omit( ( (datatemp$BETA / datatemp$SE)^2 )[summaryP < threshold] );
        sumofsquareddiff_fromzero = c(sumofsquareddiff_fromzero, sum(x));
        x = na.omit( ( ((datatemp$BETA + summaryeffect) / datatemp$SE)^2 )[summaryP < threshold] );
        sumofsquareddiff_opposite = c(sumofsquareddiff_opposite, sum(x));
}

png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_BETAstat_FG.png",width=1000,heigh=1000)
png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_BETAstat_FI.png",width=1000,heigh=1000)
png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_BETAstat_2HrG.png",width=1000,heigh=1000)
png("/nfs/users/nfs_g/gm10/MAGICtransethnic1000gimputation/Plots_BETAstat_HbA1c.png",width=1000,heigh=1000)

par(yaxt="n")
par(mar = c(5, 4, 4, 30) + 0.1)
matplot(cbind(sumofsquareddiff, # D
        sumofsquareddiff_fromzero,  # E
        sumofsquareddiff_opposite), # F
        1:length(cohorts),
        pch=c("D","E","F"),
        xlab="Discordance statistic",
        ylab="")
abline(h=1:length(cohorts),lty=2,col="grey")
for (i in 1:length(cohorts)) {
        lines(c(qchisq(0.005/length(cohorts), sumofsquareddifflength[i]),
                        qchisq(1 - 0.005/length(cohorts), sumofsquareddifflength[i])),
                c(i,i),
                col="black")
}
par(yaxt="s")
par(las=2)
axis(side=4,
        labels=cohortsO,
        at=c(1:length(cohorts)))
par(las=0)
par(mar = c(5, 4, 4, 2) + 0.1)

#abline(h=50.5)
#abline(h=c(6,8+6,13+8+6,62+13+8+6)+0.5) # FI
#abline(h=c(5,1+5,25+1+5,2+25+1+5)+0.5) # 2hrGlu
abline(h=c(7,5+7,15+5+7,39+15+5+7)+0.5) # HbA1c

dev.off()


## Lambda

hist( EASYrep_CENT$Lambda.PVALCOMM.GC , breaks=100 , xlab="Lambda on common variants (MAF>1%)") ; abline(v=c(1.2,0.9),col="orange")

EASYrep_CENT[which(EASYrep_CENT$Lambda.PVALCOMM.GC<0.9 | EASYrep_CENT$Lambda.PVALCOMM.GC>1.2 ),c("fileInShortName","Study","Trait","Ancestry","Transf","Chr","Lambda.PVALCOMM.GC")]
EASYrep_CENT[which(EASYrep_CENT$Study=="niddm" ),c("Study","Trait","Ancestry","Transf","Chr","Lambda.PVALCOMM.GC")]


montage QC2_CENTRAL_results_QQplots/*WHI*        -tile 2x2 -geometry 1000x1000 QC2_CENTRAL_results_QQplots_conc/WHI.qq.png
montage QC2_CENTRAL_results_QQplots/AA*GeneSTAR* -tile 2x2 -geometry 1000x1000 QC2_CENTRAL_results_QQplots_conc/GeneSTAR_AA.qq.png
montage QC2_CENTRAL_results_QQplots/AA*CFS*      -tile 2x3 -geometry 1000x1000 QC2_CENTRAL_results_QQplots_conc/CFS_AA.qq.png
montage QC2_CENTRAL_results_QQplots/AA*CHS*      -tile 2x3 -geometry 1000x1000 QC2_CENTRAL_results_QQplots_conc/CHS_AA.qq.png
montage QC2_CENTRAL_results_QQplots/AA_GWAS_AX_HbA1c_*        -tile 2x3 -geometry 1000x1000 QC2_CENTRAL_results_QQplots_conc/AA_HbA1c.qq.png
montage QC2_CENTRAL_results_QQplots/AA_GWAS_AX_HbA1cadjFGlu_* -tile 2x3 -geometry 1000x1000 QC2_CENTRAL_results_QQplots_conc/AA_HbA1cadjFGlu.qq.png
montage QC2_CENTRAL_results_QQplots/*GenerationS* -tile 2x2 -geometry 600x600 QC2_CENTRAL_results_QQplots_conc/GenerationScotland.qq.png
montage QC2_CENTRAL_results_QQplots/*GOYA-Male-CASES* -tile 2x5 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/GOYA-Male-CASES.qq.png
montage QC2_CENTRAL_results_QQplots/*GOYA-Male-CONTROLS* -tile 2x5 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/GOYA-Male-CONTROLS.qq.png
montage QC2_CENTRAL_results_QQplots/*LifeLines* -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/LifeLines.qq.png
montage QC2_CENTRAL_results_QQplots/*_A_*_NTR_* -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/NTR_A.qq.png
montage QC2_CENTRAL_results_QQplots/*_XF_*_NTR_* -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/NTR_XF.qq.png
montage QC2_CENTRAL_results_QQplots/*_XM_*_NTR_* -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/NTR_XM.qq.png

montage QC2_CENTRAL_results_QQplots/*LOLIPOP-EW610* -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/LOLIPOP-EW610.qq.png
montage QC2_CENTRAL_results_QQplots/*LOLIPOP-EWA* -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/LOLIPOP-EWA.qq.png
montage QC2_CENTRAL_results_QQplots/*LOLIPOP-EWP* -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/LOLIPOP-EWP.qq.png

montage QC2_CENTRAL_results_QQplots/*COPSAC* -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/COPSAC2000.qq.png


montage QC2_CENTRAL_results_QQplots/EA*CFS*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/CFS_EA.qq.png
montage QC2_CENTRAL_results_QQplots/*Fenland-OMICS*      -tile 2x5 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/Fenland-OMICS.qq.png
montage QC2_CENTRAL_results_QQplots/*PROCARDIS*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/PROCARDIS.qq.png
montage QC2_CENTRAL_results_QQplots/*DGI*      -tile 2x5 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/DGI.qq.png
montage QC2_CENTRAL_results_QQplots/*InterAct*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/InterAct.qq.png
montage QC2_CENTRAL_results_QQplots/*WGHS*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/WGHS.qq.png
montage QC2_CENTRAL_results_QQplots/*FHS*      -tile 2x5 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/FHS.qq.png
montage QC2_CENTRAL_results_QQplots/*ALSPACm*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/ALSPACmothers.qq.png
montage QC2_CENTRAL_results_QQplots/*ALSPACc*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/ALSPACchildren.qq.png
montage QC2_CENTRAL_results_QQplots/*MICROS*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/MICROS.qq.png

montage QC2_CENTRAL_results_QQplots/EA_GWAS_A_FGluadjBMI_InvNorm_MEGA_SNPTEST_10102014_RLI.txt.gz.qq.png QC2_CENTRAL_results_QQplots/EA_MCHIP_A_FInsadjBMI_IMPROVE_MACH2QTL_25092013_MDF.txt.gz.qq.png  -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/RoundingProblem.qq.png

montage QC2_CENTRAL_results_QQplots/*_A_*SORBS*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/SORBS_A.qq.png
montage QC2_CENTRAL_results_QQplots/*_XF_*SORBS*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/SORBS_XF.qq.png
montage QC2_CENTRAL_results_QQplots/*_XM_*SORBS*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/SORBS_XM.qq.png


montage QC2_CENTRAL_results_QQplots/*GoDARTS*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/GoDARTS.qq.png
montage QC2_CENTRAL_results_QQplots/*NFBC1986*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/NFBC1986.qq.png

montage QC2_CENTRAL_results_QQplots/*VanderbiltLargeShanghai1*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/VanderbiltLargeShanghai1.qq.png
montage QC2_CENTRAL_results_QQplots/*VanderbiltLargeShanghai2*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/VanderbiltLargeShanghai2.qq.png
montage QC2_CENTRAL_results_QQplots/*VanderbiltGWAS6548*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/VanderbiltGWAS6548.qq.png


montage QC2_CENTRAL_results_QQplots/*SP2-610*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/SP2-610.qq.png
montage QC2_CENTRAL_results_QQplots/*SCES*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/SCES.qq.png
montage QC2_CENTRAL_results_QQplots/*SiMES*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/SiMES.qq.png
montage QC2_CENTRAL_results_QQplots/*CLHNS*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/CLHNS.qq.png
montage QC2_CENTRAL_results_QQplots/*NHAPC*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/NHAPC.qq.png
montage QC2_CENTRAL_results_QQplots/*KARE*      -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/KARE.qq.png
montage QC2_CENTRAL_results_QQplots/*NAGAH*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/NAGAHAMA.qq.png
montage QC2_CENTRAL_results_QQplots/*TWSC*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/TWSC.qq.png
montage QC2_CENTRAL_results_QQplots/AA*MESA*      -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/MESA_AA.qq.png
montage QC2_CENTRAL_results_QQplots/EA_*MESA*      -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/MESA_EA.qq.png
montage QC2_CENTRAL_results_QQplots/EAA*MESA*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/MESA_EAA.qq.png
montage QC2_CENTRAL_results_QQplots/HA*MESA*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/MESA_HA.qq.png
montage QC2_CENTRAL_results_QQplots/*_A_*CAGE-KING-Quad*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/CAGE-KING-Quad_A.qq.png
montage QC2_CENTRAL_results_QQplots/*_XF_*CAGE-KING-Quad*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/CAGE-KING-Quad_XF.qq.png
montage QC2_CENTRAL_results_QQplots/*_XM_*CAGE-KING-Quad*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/CAGE-KING-Quad_XM.qq.png


montage QC2_CENTRAL_results_QQplots/*IRAS_*      -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/IRAS.qq.png
montage QC2_CENTRAL_results_QQplots/*niddm*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/niddm.qq.png
montage QC2_CENTRAL_results_QQplots/*htn*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/htn.qq.png
montage QC2_CENTRAL_results_QQplots/*macad*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/macad.qq.png
montage QC2_CENTRAL_results_QQplots/*_SOL_*      -tile 2x4 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/SOL.qq.png


montage QC2_CENTRAL_results_QQplots/AA*BioMe*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/BioMe_AA.qq.png
montage QC2_CENTRAL_results_QQplots/EA*BioMe*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/BioMe_EA.qq.png
montage QC2_CENTRAL_results_QQplots/HA*BioMe*      -tile 2x2 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/BioMe_HA.qq.png


montage QC2_CENTRAL_results_QQplots/*_RHS_*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/RHS.qq.png
montage QC2_CENTRAL_results_QQplots/*RHS-610*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/RHS-610.qq.png

montage QC2_CENTRAL_results_QQplots/*LOLIPOP-IA317*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/LOLIPOP-IA317.qq.png
montage QC2_CENTRAL_results_QQplots/*LOLIPOP-IAP*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/LOLIPOP-IAP.qq.png
montage QC2_CENTRAL_results_QQplots/*LOLIPOP-OmniEE*      -tile 2x3 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/LOLIPOP-OmniEE.qq.png
montage QC2_CENTRAL_results_QQplots/*SINDI*      -tile 2x1 -geometry 400x400 QC2_CENTRAL_results_QQplots_conc/SINDI.qq.png




## PZ plots
PZ = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results",full.names=T) ; PZ = PZ[grep("PZ-PLOTS",PZ)]
length(PZ)
head(PZ)
PZ2 = sub("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/","",sub(".multi.PZ-PLOTS.png","",PZ))
repFiles2 = sub("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/","",sub(".rep","",repFiles))
table(repFiles2 %in% PZ2)
which(!(repFiles2 %in% PZ2))
repFiles2[23:24]

grep CENTRALLY_GWAS_EUR_FG_11.ecf QC2_EasyQC_scripts/bsub_CENT_GWAS_* # QC2_EasyQC_scripts/bsub_CENT_GWAS_22.out
grep CENTRALLY_GWAS_EUR_FG_12.ecf QC2_EasyQC_scripts/bsub_CENT_GWAS_* # QC2_EasyQC_scripts/bsub_CENT_GWAS_23.out




## list study with problems
load( "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/EASYrep_CENT.RData" )
cleaned    = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/CleanedFiles20150703.txt" ,as.is=T,h=T,sep="\t")
table(EASYrep_CENT$fileInShortName %in% sub("CLEANED.","",cleaned$File))
EASYrep_CENT$fileInShortName[ which(!(EASYrep_CENT$fileInShortName %in% sub("CLEANED.","",cleaned$File))) ]
cleaned$File[grep("HbA1cadjFGlu_InvNorm_KARE",cleaned$File)]

bad = c("ERF4","RAINE","RAINE-Y20","KORA","FHS","WGHS","DGI","BioMe","MICROS","GeneSTAR","EPIHEALTH","TWINGENE","PELOTAS","LifeLines","TRAILSpop","SIGMA","CRC",
        "ASCOT-SC","ASCOT-UK","CHS","CoLaus","IMPROVE","PROCARDIS","SCARFSHEEP","TAICHI","PROSPER","SOL","LOLIPOP-EW610","LOLIPOP-EWA","LOLIPOP-EWP","LOLIPOP-IA317",
        "LOLIPOP-IA610","LOLIPOP-IAP","LOLIPOP-OmniEE","HANDLS","WHI","CFS","NAGAHAMA","GOYA-Male-CASES","GOYA-Male-CONTROLS","RISC","INCHIANTI","BC1936","D2D2007.DPS",
        "D2D2007.DPS.DRSEXTRA.FUSIONS2","DIAGEN","FUSION","METSIM","TROMSO","MESA","CLHNS","NHAPC","SORBS","htn","IRAS","IRASFS","macad","niddm","LLS","TWINSUK","AMISH",
        "TWSC","Health2006","Inter99","AASC","CAGE-KING-Omni","CAGE-KING-Quad","CAGEGWAS1","RHS","RHS-610K","COPSAC2000","ALSPACchildren","ALSPACmothers","GoDARTS","NFBC1966",
        "NFBC1986","SCES","SiMES","SINDI","SP2-1M","SP2-610","AGES","MEGA","EGCUT","PIVUS","ULSAM","LEIPZIG-adults","LEIPZIG-kids","Ely","NSHD","EPIC","Fenland-OMICS","Fenland",
        "InterAct","VanderbiltGWAS6548","VanderbiltLargeShanghai1","VanderbiltLargeShanghai2","NTR","ORCADES","KARE","LURIC","GENOA","DES","FrenchYoungControl","OBA","OBYcase",
        "HELICMANOLIS","HELICPomak","CROATIA-Korcula","CROATIA-Split","CROATIA-Vis","GenerationScotland")

table(unique(EASYrep_CENT$Study) %in% bad)
unique(EASYrep_CENT$Study)[which(!(unique(EASYrep_CENT$Study) %in% bad))]

good = c(
## ? "FHS",
"AGES",
"ASCOT-SC",
"ASCOT-UK",
"BC1936",
"BioMe",
"CAGE-KING-Omni",
"CAGE-KING-Quad",
"CAGEGWAS1",
"CHS",
"D2D2007.DPS",
"D2D2007.DPS.DRSEXTRA.FUSIONS2",
"DIAGEN",
"EGCUT",
"Ely",
"FamHS",
"Fenland",
"Fenland-OMICS",
"FUSION",
"GenerationScotland",
"GeneSTAR",
"HANDLS",
"htn",
"INCHIANTI",
"InterAct",
"IRAS",
"IRASFS",
"KARE",
"LifeLines",
"LOLIPOP-EW610",
"LOLIPOP-EWA",
"LOLIPOP-EWP",
"LOLIPOP-IA317",
"LOLIPOP-IA610",
"LOLIPOP-IAP",
"LOLIPOP-OmniEE",
"LURIC",
"macad",
"MESA",
"NFBC1966",
"NHAPC",
"niddm",
"NSHD",
"ORCADES",
"PIVUS",
"PREVEND",
"SardiNIA",
"SORBS",
"TROMSO",
"TWINGENE",
"ULSAM",
"VanderbiltGWAS6548",
"VanderbiltLargeShanghai1",
"VanderbiltLargeShanghai2"
)
length(good)
length(unique(good))


"CRC" %in% good
table(grepl("RHS",good))

table(good %in% EASYrep_CENT$Study)

sum(EASYrep_CENT$Study %in% good)
sum(cleaned$Study %in% good)

goodFiles = EASYrep_CENT[ which(EASYrep_CENT$Study %in% good
                                | EASYrep_CENT$Study %in% c("RS1","RS2","RS3","ERF4") # they re-uploaded FI but FG is fine
                               ),
                          c("fileInShortName","Study","Ancestry","Trait","Transf","Chr","Chip")]
cleaned$File2 = sub("CLEANED.","",cleaned$File)
cleaned$File2[which(cleaned$File=="EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_KARE_MACH2QTL_31102014_YJK.gz")]="EAA_GWAS_AX_HbA1cadjFGlu_InvNorm_KARE_MACH2QTL_31102014_YJK.txt.gz"
table( goodFiles$fileInShortName %in% cleaned$File2 )
goodFiles = merge( goodFiles , cleaned[,c("File2","Path")] , by.x="fileInShortName" , by.y="File2")

write.table( cbind(paste( "rename /cleaned_gluinsrelatedtraits/Cleaned/" , goodFiles$Path , " /cleaned_gluinsrelatedtraits/FREEZE1_Cleaned_Ready/" , goodFiles$Path , sep="" )) , "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/MovingFiles.txt" , quote=F , row.names=F , col.names=F )



good2 = EASYrep_CENT[which(EASYrep_CENT$Study %in% good),]

unique( good2[,c("Study","Ancestry","Trait","Transf","Chr","Chip")] )

good2$Anc_Trait = paste( good2$Ancestry , good2$Trait , good2$Transf , sep="_" )
wwA = which(good2$Chr %in% c("A","AX"))
Nsamples = (tapply( good2$Clean.SampleSize[wwA] , INDEX=as.factor(good2$Anc_Trait[wwA]) , FUN=sum ))
Nstudies = (tapply( good2$Clean.SampleSize[wwA] , INDEX=as.factor(good2$Anc_Trait[wwA]) , FUN=length ))
table(names(Nsamples)==names(Nstudies))

N = data.frame( Anc_Trait_Transf=names(Nsamples) , Nsamples=Nsamples , Nstudies=Nstudies , stringsAsFactors=F )
row.names(N)=NULL
tp = sapply( N$Anc_Trait_Transf , FUN=function(x){return(strsplit(x,"_")[[1]])} )
N$Ancestry = tp[1,]
N$Trait = tp[2,]
N$Transf = tp[3,]
rm(tp,Nsamples,Nstudies)

Nsamples=NULL
Nstudies=NULL
for ( tt in unique(N$Trait) )
{
 tpSamples=NULL
 tpStudies=NULL
 for ( aa in unique(N$Ancestry) )
 {
  ww = which( N$Ancestry==aa & N$Trait==tt & N$Transf=="InvNorm" )
  if( length(ww)==1 )
  {
   tpSamples = c( tpSamples , N$Nsamples[ww] ) ; names(tpSamples)[length(tpSamples)]=aa
   tpStudies = c( tpStudies , N$Nstudies[ww] ) ; names(tpStudies)[length(tpStudies)]=aa
  } else {
           tpSamples = c( tpSamples , 0 ) ; names(tpSamples)[length(tpSamples)]=aa
           tpStudies = c( tpStudies , 0 ) ; names(tpStudies)[length(tpStudies)]=aa
         }
  rm(ww)
 } ; rm(aa)
 Nsamples=cbind( Nsamples , tpSamples ) ; colnames(Nsamples)[ncol(Nsamples)]=tt
 Nstudies=cbind( Nstudies , tpStudies ) ; colnames(Nstudies)[ncol(Nstudies)]=tt
 rm(tpSamples,tpStudies)
} ; rm(tt)
Nstudies
Nsamples

cleaned$Path = paste("/cleaned_gluinsrelatedtraits/Cleaned/",cleaned$Path,sep="")

table(cleaned$Transf,cleaned$Trans_Folder)
cleaned$Transf = cleaned$Trans_Folder

forMA = cleaned[ which(cleaned$Study %in% good), c("Path","File","Study","Ancestry","Trait","Transf","Chr") ]
table(forMA$Ancestry[which(forMA$Transf=="InvNorm" & forMA$Chr %in% c("A","AX"))],forMA$Trait[which(forMA$Transf=="InvNorm" & forMA$Chr %in% c("A","AX"))])

forMA = forMA[ which(  forMA$Trait %in% c("FGluadjBMI","FInsadjBMI")
                     | forMA$Ancestry=="EA"
                     | (forMA$Trait=="2hrGluadjBMI" & forMA$Ancestry=="HA")
                     | ( forMA$Trait %in% c("HbA1c","HbA1cadjFGlu") & forMA$Ancestry=="IAA" )  ) ,]
table(forMA$Ancestry[which(forMA$Transf=="InvNorm" & forMA$Chr %in% c("A","AX"))],forMA$Trait[which(forMA$Transf=="InvNorm" & forMA$Chr %in% c("A","AX"))])

table(forMA$Ancestry[which(forMA$Transf=="Raw")],forMA$Chr[which(forMA$Transf=="Raw")])

write.table( forMA , "/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FilesForFirstPassMA_20150703.txt" , quote=F , row.names=F , sep="\t" )




####################################################################################
## RERUN EasyQC PZ plots on common variants only for a subset of studies
####################################################################################

####### /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_PZ_template.ecf
### Set parameters and input files
DEFINE --strMissing . --strSeparator TAB --pathOut /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_PZ_results --acolIn MarkerName;EFFECT_ALLELE;NON_EFFECT_ALLELE;N_QTL;EAF_QTL;BETA;SE;PVAL;IMPUTED;INFO --acolInClasses character;character;character;numeric;numeric;numeric;numeric;numeric;numeric;numeric
### Start EasyQC
START EASYQC
# add/edit some columns
ADDCOL  --rcdAddCol  ifelse( EAF_QTL>0.01 & EAF_QTL<0.99 , yes=PVAL , no=NA ) --colOut PVALCOMM
PZPLOT --colBeta BETA  --colSe SE  --colPval PVALCOMM
### Stop EasyQC
STOP EASYQC
#######


####### /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_PZ2_template.ecf
### Set parameters and input files
DEFINE --strMissing . --strSeparator TAB --pathOut /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_PZ2_results --acolIn MarkerName;EFFECT_ALLELE;NON_EFFECT_ALLELE;N_QTL;EAF_QTL;BETA;SE;PVAL;IMPUTED;INFO --acolInClasses character;character;character;numeric;numeric;numeric;numeric;numeric;numeric;numeric
### Start EasyQC
START EASYQC
# add/edit some columns
ADDCOL  --rcdAddCol  ifelse( EAF_QTL>0.05 & EAF_QTL<0.95 & INFO>=0.7 , yes=PVAL , no=NA ) --colOut PVALCOMMGOOD
PZPLOT --colBeta BETA  --colSe SE  --colPval PVALCOMMGOOD
### Stop EasyQC
STOP EASYQC
#######



ss = c("BioMe","MESA","SP2-1M","SP2-610","NHAPC","SCES","SiMES","TWSC",
       "LURIC","BC1936","DGI","Fenland","KORA","LLS","ULSAM","MEGA",
       "LOLIPOP-EW610","LOLIPOP-EWA","LOLIPOP-EWP","LOLIPOP-IA317","LOLIPOP-IA610","LOLIPOP-IAP","LOLIPOP-OmniEE",
       "ALSPACchildren","ALSPACmothers","EGCUT","EPIHEALTH","HELICPomak","NFBC1966",
       "DES","OBA","OBYcase","PIVUS","PROSPER","TRAILSpop","COPSAC2000","InterAct",
       "IRAS","SIGMA","PELOTAS","RHS","RHS-610K","SINDI",
       "TAICHI","Inter99","LEIPZIG-adults","LEIPZIG-kids","GoDARTS","Health2006","NFBC1986",
       "TWINSUK","IMPROVE")

table( ss %in% EASYrep_CENT$Study)
sum(EASYrep_CENT$Study %in% ss)

EASYrep_CENT$fileInShortName[which(EASYrep_CENT$Study %in% ss)]


dd1 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_download",full.names=T)
dd2 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_cleaned",full.names=T) ; dd2 = dd2[grep("gz",dd2)]
#dd3 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/CLEANED",full.names=T)
dd4 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/FILES/CLEANED/Raw",full.names=T) ; dd4 = dd4[grep("gz",dd4)]
dd5 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/QC2/FILES/CLEANED/InvNorm",full.names=T) ; dd4 = dd4[grep("gz",dd4)]
ff0 = c(dd1,dd2,dd4,dd5) ; rm(dd1,dd2,dd4,dd5)
ff0_file = sapply( ff0 , FUN=function(x){ x1 = strsplit(x,"/",fixed=T)[[1]] ; return(x1[length(x1)]) } )
ff0_file = sub("CLEANED.","",ff0_file)
ff1 = sapply( ff0_file , FUN=function(x){
                                     x2 = strsplit( x ,"_",fixed=T)[[1]]
                                     a1 = x2[1]
                                     if(a1=="EA") {a2="EUR"}
                                     if(a1=="AA") {a2="AFR"}
                                     if(a1=="EAA") {a2="ASN"}
                                     if(a1=="IAA") {a2="SAS"}
                                     if(a1=="HA") {a2="MXL"}
                                     t1 = x2[4]
                                     if( t1=="FGluadjBMI" ) { t2="FG" }
                                     if( t1=="FInsadjBMI" ) { t2="FI" }
                                     if( t1=="2hrGluadjBMI" ) { t2="2hGlu" }
                                     if( t1 %in% c("HbA1c","HbA1cadjFGlu") ) { t2="HbA1c" }
                                     c1 = x2[3]
                                     if( c1 %in% c("A","AX"))  { c2=1 } else { c2=0 }
                                     return( c( CHIP=x2[2] , ANCESTRY=a2 , TRAIT=t2 , AUTOSOME_IN=c2 ) )
                                   } )

table(ff1["CHIP",])
table(ff1["ANCESTRY",])
table(ff1["TRAIT",])
table(ff1["AUTOSOME_IN",])

table(ff1["TRAIT",],ff1["ANCESTRY",],ff1["CHIP",],ff1["AUTOSOME_IN",])
table(ff1["TRAIT",],ff1["ANCESTRY",],ff1["CHIP",])




#tt = system("cat /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_PZ_template.ecf",intern=T)
tt = system("cat /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_PZ2_template.ecf",intern=T)

FUN1 = function( ssx )
{
 ww0=which( EASYrep_CENT$Study==ssx )
 chip = unique(EASYrep_CENT$Chip[ww0])
  ee=c( tt[1:2]
      , paste( "EASYIN --fileIn" , ff0[ which(ff0_file %in% EASYrep_CENT$fileInShortName[ww0]) ])
      , tt[3:10]
      )
  # write script
  #write.table( ee , paste("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_PZ_",chip,"_",ssx,".ecf",sep="") , quote=F , col.names=F , row.names=F )
  write.table( ee , paste("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/CENTRALLY_PZ2_",chip,"_",ssx,".ecf",sep="") , quote=F , col.names=F , row.names=F )
  rm(ww0,ee,chip)
}


for ( ssi in ss) { FUN1( ssi ) } ; rm(ssi)

cd QC2_EasyQC_scripts/
#i=0 ; for s in `ls -1 CENTRALLY_PZ_GWAS_*.ecf` ; do ((i+=1)) ; bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENTPZ_GWAS_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENTPZ_GWAS_$i.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
#i=43 ; for s in `ls -1 CENTRALLY_PZ_MCHIP_*.ecf` ; do ((i+=1)) ; bsub -q normal -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENTPZ_MCHIP_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENTPZ_MCHIP_$i.err  -R"select[mem>16000]  rusage[mem=16000]"  -M16000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
i=0  ; for s in `ls -1 CENTRALLY_PZ2_GWAS_*.ecf`  ; do ((i+=1)) ; bsub -q hugemem -G t35_magic -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENTPZ2_GWAS_$i.out   -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENTPZ2_GWAS_$i.err   -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
i=44 ; for s in `ls -1 CENTRALLY_PZ2_MCHIP_*.ecf` ; do ((i+=1)) ; bsub -q normal -G t35_magic  -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENTPZ2_MCHIP_$i.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/bsub_CENTPZ2_MCHIP_$i.err  -R"select[mem>16000]  rusage[mem=16000]"  -M16000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_EasyQC_scripts/RunEasyQC.R" ; done
cd ../


grep "Succesfully finished EasyX" QC2_EasyQC_scripts/bsub_CENTPZ2_*| wc

for f in `ls QC2_EasyQC_scripts/bsub_CENTPZ2_GWAS_*.out` ; do echo $f ; grep Running $f | sed 's/Running script  \/lustre\/scratch113\/projects\/t35_magic\/TRANSETHNIC_1000G\/gaelle\/QC2_EasyQC_scripts\///g' ; grep ".gz" $f | grep "+  /" | wc ; grep Processing $f | wc ; done
for f in `ls QC2_EasyQC_scripts/bsub_CENTPZ2_MCHIP*.out` ; do echo $f ; grep Running $f | sed 's/Running script  \/lustre\/scratch113\/projects\/t35_magic\/TRANSETHNIC_1000G\/gaelle\/QC2_EasyQC_scripts\///g' ; grep ".gz" $f | grep "+  /" | wc ; grep Processing $f | wc ; done




grep Success QC2_EasyQC_scripts/bsub_CENTPZ_* | wc
bjobs | grep hugemem | wc
bjobs | grep hugemem | grep RUN | wc
bjobs | grep normal | wc
bjobs | grep normal | grep RUN | wc


ls -1 QC2_EasyQC_scripts/CENTRALLY_PZ_*.ecf |wc
ls -1 QC2_CENTRAL_PZ_results/*PZ-PLOTS.png | wc





####################################################################################
## Extract missing HWE variants
####################################################################################


cd QC1_DONE/
for f in `ls *D2D2007.DPS*`; do zcat $f | awk '{ if(NR==1 || ($11=="." && $16==0)) {print $1"\t"$2"\t"$3"\t"$4"\t"$5}}' | bgzip -c > ../ExtractMissingHWEvariants/$f; done
for f in `ls *DIAGEN*`; do zcat $f | awk '{ if(NR==1 || ($11=="." && $16==0)) {print $1"\t"$2"\t"$3"\t"$4"\t"$5}}' | bgzip -c > ../ExtractMissingHWEvariants/$f; done
for f in `ls *FUSION*`; do zcat $f | awk '{ if(NR==1 || ($11=="." && $16==0)) {print $1"\t"$2"\t"$3"\t"$4"\t"$5}}' | bgzip -c > ../ExtractMissingHWEvariants/$f; done
for f in `ls *METSIM*`; do zcat $f | awk '{ if(NR==1 || ($11=="." && $16==0)) {print $1"\t"$2"\t"$3"\t"$4"\t"$5}}' | bgzip -c > ../ExtractMissingHWEvariants/$f; done
for f in `ls *TROMSO*`; do zcat $f | awk '{ if(NR==1 || ($11=="." && $16==0)) {print $1"\t"$2"\t"$3"\t"$4"\t"$5}}' | bgzip -c > ../ExtractMissingHWEvariants/$f; done
cd ../

cd ExtractMissingHWEvariants/
for f in `ls *`; do echo $f; zcat $f | wc; done
cd ../


####################################################################################
## Extract missing INFO variants for CFS
####################################################################################


cd QC1_DONE/
for f in `ls EA_*CFS*`; do zcat $f | awk '{ if(NR==1 || ($18==".")) {print $1"\t"$2"\t"$3"\t"$4"\t"$5}}' | bgzip -c > ../ExtractMissingINFOvariants/$f; done
cd ../

cd ExtractMissingINFOvariants/
for f in `ls *`; do echo $f; zcat $f | wc; done
cd ../
cd ExtractMissingINFOvariants/
for f in `ls *`; do echo $f; zcat $f | awk '{split($1,a,":") ;  if( length(a)==2) {print $0} }'; done
cd ../



####################################################################################
## Check outlier frequencies for SOL
####################################################################################

SOL = read.table("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/QC2_CENTRAL_results/HA_GWAS_AX_FInsadjBMI_SOL_MIXEDMODEL_20022015_TL.AFCHECK.outlier.txt",as.is=T,h=T)
plot( SOL$eaf.ref , SOL$EAF_QTL )
table( SOL$EFFECT_ALLELE==SOL$a0.ref )
table( SOL$NON_EFFECT_ALLELE==SOL$a1.ref)



####################################################################################
## Clean some last files from FREEZE2
####################################################################################

cd /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/

mv files_fromJianAn/*LIVING-BIOBANK-MERCK-CHIN* files_Harmonised_Formatted/

bsub -q yesterday -G t35_magic  -o bsub/bsub_ARIC_EA_FG_i_reformat.out  -e bsub/bsub_ARIC_EA_FG_i_reformat.err  -R"select[mem>100]  rusage[mem=100]"  -M100  "zcat files_fromJianAn/EA_GWAS_A_FGluadjBMI_InvNorm_ARIC_IMPUTE_170215_ML.txt.gz  | awk '{ if (NR>1) {\$7=\$8+\$9+\$10} ; print \$1\"\t\"\$2\"\t\"\$3\"\t\"\$4\"\t\"\$5\"\t\"\$6\"\t\"\$7\"\t\"\$8\"\t\"\$9\"\t\"\$10\"\t\"\$11\"\t\"\$12\"\t\"\$13\"\t\"\$14\"\t\"\$15\"\t\"\$16\"\t\"\$17\"\t\"\$18\"\t\"\$19}' | bgzip -c > files_Harmonised_Formatted/EA_GWAS_A_FGluadjBMI_InvNorm_ARIC_IMPUTE_170215_ML.txt.gz"

cp /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/CROATIA-VIS/NEW_FORMATTED/EA_GWAS_X_FGluadjBMI_InvNorm_CROATIA-Vis_PROBABEL_25012016_TB.txt.gz files_Harmonised_Formatted/
cp /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/CROATIA-VIS/NEW_FORMATTED/EA_GWAS_X_FInsadjBMI_CROATIA-Vis_PROBABEL_25012016_TB.txt.gz         files_Harmonised_Formatted/
cp /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/ellie/CROATIA-VIS/NEW_FORMATTED/EA_GWAS_X_FInsadjBMI_InvNorm_CROATIA-Vis_PROBABEL_25012016_TB.txt.gz files_Harmonised_Formatted/



tt = system("cat /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/template.ecf",intern=T)

dd1 = dir("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/files_Harmonised_Formatted")


FUN1 = function( ff )
{
  ee=c( tt[1:2]
      , paste( "EASYIN --fileIn /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/files_Harmonised_Formatted/" , ff , sep="")
      , tt[3:length(tt)]
      )
  write.table( ee , paste("/lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/CLEANING_",ff,".ecf",sep="") , quote=F , col.names=F , row.names=F )
  rm(ee)
}


sapply( dd1 , FUN=FUN1 )



#cd /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/
#for s in `ls -1 CLEANING_*.ecf` ; do bsub -q hugemem -G t35_magic  -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/RunEasyQC.R" ; done
#for s in `ls -1 CLEANING_*ARIC*.ecf` ; do bsub -q hugemem -G t35_magic  -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/RunEasyQC.R" ; done
#for s in `ls -1 CLEANING_*GOYA*.ecf` ; do bsub -q hugemem -G t35_magic  -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/RunEasyQC.R" ; done
#for s in `ls -1 CLEANING_*CRC*.ecf` ; do bsub -q hugemem -G t35_magic  -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/RunEasyQC.R" ; done
#for s in `ls -1 CLEANING_*NEO*.ecf` ; do bsub -q hugemem -G t35_magic  -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/RunEasyQC.R" ; done
#for s in `ls -1 CLEANING_*LURIC*.ecf` ; do bsub -q hugemem -G t35_magic  -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/RunEasyQC.R" ; done
cd /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/
for s in `ls -1 CLEANING_*2hrGluadjBMI_LURIC*.ecf` ; do bsub -q hugemem -G t35_magic  -o /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.out  -e /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_$s.err  -R"select[mem>40000]  rusage[mem=40000]"  -M40000  "/software/R-3.0.0/bin/R --no-save --args $s < /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/scripts/RunEasyQC.R" ; done
cd /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/

tail -n1 /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_CLEANING_*2hrGluadjBMI_LURIC*.out
grep Success /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_CLEANING_*2hrGluadjBMI_LURIC*
grep "Succesfully finished EasyX" /lustre/scratch113/projects/t35_magic/TRANSETHNIC_1000G/gaelle/FREEZE2_ToClean/bsub/bsub_CLEANING_*2hrGluadjBMI_LURIC*


ls -1 FREEZE2_ToClean/outputs/*.gz
mv FREEZE2_ToClean/outputs/*.gz FREEZE2_download/
