# MAGIC-1000G

QC-step-1.R is the code used to reformat summary statistics from cohorts and harmonize variant name in centralized quality control step.

QC-step-2.sh is the code applying EasyQC to clean the data in centralized quality control step.

single-ancestry_meta-analyses.R is the code applying METAL to conduct the signal-ancestry meta-analyses.

trans-ancetry_meta-analyses.R is the code applying MANTRA to conduct the trans-ancestry meta-analyses.

variance-explained.txt is the code used to conduct the trait variance explained analysis.

prs-calculation.R is the code applying PRScs to generate the European ancestry-derived polygenic scores.

prs-test.R is the code used to test the transferability of European ancestry-derived polygenic scores across ancestries.

fine-mapping.py is the code applying FINEMAP to conduct the fine-mapping analysis.

hba1c_classification.R is the code used to conduct HbA1c signal classification.

hba1c_grs_t2d.R is the code used to analyse HbA1c genetic risk scores (GRSs) and type 2 diabetes (T2D) risk.

depict.sh is the code applying DEPICT to conduct tissue and gene set enrichment analysis.

The code for Epigenomic landscape of trait-associated variants can be found here https://github.com/ParkerLab/magic_final.
